---
title: "Silverside Genomics - Summary statistics"
author: "Arne Jacobs"
output: 
  md_document:
    toc: true
highlight: pygments
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
names(knitr::knit_engines$get())
```

# Plotting locations and phenotypic trait variation for M. menidia sample sites:

Load R-packages:
```{r, eval=F, include=F}
#Plotting:
library(tidyverse)
library(ggfortify)
library(RColorBrewer)
library(wesanderson)
library(cowplot)
library(ggridges)
library(reshape2)
library(CMplot)
library(ggtree)
library(patchwork)

#Analysis
library(windowscanr)
library(corrr)
library(vegan)
library(conStruct)
library(ade4)
library(gradientForest)
library(matrixcalc)
```


# Create SAF and SFS files for inferring summary statistics

Create folded saf files for each population based on the filtered SNP dataset:
```{bash, eval=F}
nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
JIGA_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_jiga_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_LRSC.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
LRSC_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_LRSC_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCNC.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MCNC_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MCNC_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_OINC.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
OINC_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_OINC_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_CTVA.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
CTVA_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_CTVA_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
PANY_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_PANY_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_DXMA.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
DXMA_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_DXMA_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MDME.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MDME_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MDME_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MBNS.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MBNS_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MBNS_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_HXNS.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
HXNS_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_HXNS_fold.nohup &

nohup ./angsd_gl_saf_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MAQU.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MAQU_filtsnps_fold \
/workdir/arne/results/snp_datasets/Mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MAQU_fold.nohup &
```


Create unfolded saf files for each population based on the filtered SNP dataset (for Fst):
```{bash, eval=F}
nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
JIGA_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_jiga.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_LRSC.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
LRSC_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_LRSC.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCNC.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MCNC_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MCNC.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_OINC.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
OINC_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_OINC.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_CTVA.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
CTVA_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_CTVA.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
PANY_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_PANY.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCCT.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MCCT_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MCCT.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_DXMA.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
DXMA_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_DXMA.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MDME.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MDME_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MDME.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MBNS.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MBNS_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MBNS.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_HXNS.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
HXNS_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_HXNS.nohup &

nohup sh ./angsd_gl_saf_bypop_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MAQU.txt \
/workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta \
/workdir/arne/results/snp_datasets/ \
MAQU_fst_filtsnps \
/workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt \
> /workdir/arne/output_logfiles/angsd_pop_saf_mme_MAQU.nohup &
```


Create folded saf files for each population based on all sites:
```{bash, eval=F}
nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt JIGA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 24 16 200 > /workdir/arne/output_logfiles/jiga_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_LRSC.txt LRSC /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 25 16 200 > /workdir/arne/output_logfiles/LRSC_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCNC.txt MCNC /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 25 16 200 > /workdir/arne/output_logfiles/MCNC_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_OINC.txt OINC /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 24 16 200 > /workdir/arne/output_logfiles/OINC_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_CTVA.txt CTVA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 23 15 200 > /workdir/arne/output_logfiles/CTVA_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt PANY /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 24 16 200 > /workdir/arne/output_logfiles/pany_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCCT.txt MCCT /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 25 16 200 > /workdir/arne/output_logfiles/MCCT_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_DXMA.txt DXMA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 25 16 200 > /workdir/arne/output_logfiles/DXMA_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MDME.txt MDME /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 25 16 200 > /workdir/arne/output_logfiles/MDME_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MBNS.txt MBNS /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 25 16 200 > /workdir/arne/output_logfiles/MBNS_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_HXNS.txt HXNS /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 23 15 200 > /workdir/arne/output_logfiles/HXNS_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MAQU.txt MAQU /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_folded 21 14 200 > /workdir/arne/output_logfiles/MAQU_folded_sfs_allsites_nohup.log &
```

Estimate folded saf and sfs for all sites for PANY and JIGA using GATK GL model. Compare theta results for estimates from samtools and gatk GL models
```{bash, eval=F}
nohup sh ./angsd_gatkGL_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt PANY /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_gatk_folded 24 16 200 > /workdir/arne/output_logfiles/pany_gatk_folded_sfs_allsites_nohup.log &

nohup sh ./angsd_gatkGL_sfs_allsites_folded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt JIGA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/ _allsites_gatk_folded 24 16 200 > /workdir/arne/output_logfiles/jiga_gatk_folded_sfs_allsites_nohup.log &
```






## Perform LD analyses with ngsLD: LD decay and LD pruning

Get genotype-likelihood file (glf format) for OINC / PANY / MDME: (intermediate AF for different inversions):
```{bash, eval = F}
nohup sh ./ansgd_get_glf.sh /workdir/arne/BamFiles_clean/ /workdir/arne/sampleinfo/BamFiles_clean_list_OINC.txt /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt /workdir/arne/results/snp_datasets/ GLF3_filtsnps_mme_oinc > /workdir/arne/output_logfiles/GLF3_Angsd_mme_filtSNPs_oinc.nohup &

nohup sh ./ansgd_get_glf.sh /workdir/arne/BamFiles_clean/ /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt /workdir/arne/results/snp_datasets/ GLF3_filtsnps_mme_pany > /workdir/arne/output_logfiles/GLF3_Angsd_mme_filtSNPs_pany.nohup &

nohup sh ./ansgd_get_glf.sh /workdir/arne/BamFiles_clean/ /workdir/arne/sampleinfo/BamFiles_clean_list_MDME.txt /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt /workdir/arne/results/snp_datasets/ GLF3_filtsnps_mme_mdme > /workdir/arne/output_logfiles/GLF3_Angsd_mme_filtSNPs_mdme.nohup &
```

Run ngsLD for each population:
```{bash, eval = F}
nohup sh run_ngsLD.sh /workdir/arne/results/snp_datasets/ /workdir/arne/results/ld_analyses/ GLF3_filtsnps_mme_oinc GLF3_filtsnps_mme_oinc.glf.pos 47 7689845 OINC 0.01 5 > /workdir/arne/output_logfiles/oinc_ngsLD_estimates_nohup.log &

nohup sh run_ngsLD.sh /workdir/arne/results/snp_datasets/ /workdir/arne/results/ld_analyses/ GLF3_filtsnps_mme_pany GLF3_filtsnps_mme_pany.glf.pos 48 7689845 PANY 0.01 5 > /workdir/arne/output_logfiles/pany_ngsLD_estimates_nohup.log &

#Select columns 1,4,7,8,9,10,11
awk '{print $1,$4,$7,$8,$9,$10,$11}' OINC_ngsLD_output.ld > OINC_ngsLD_sub_output.ld
nohup awk '{print $1,$4,$7,$8,$9,$10,$11}' PANY_ngsLD_output > PANY_ngsLD_sub_output.ld &
awk '{print $1,$4,$7,$8,$9,$10,$11}' MDME_ngsLD_output.ld > MDME_ngsLD_sub_output.ld
```





### LD-pruning:

Split files into individual chromosomes:
```{bash, eval=F}
#Split column 1 and 2 (then 3) into two columns, 1 with chr and one with position:
nohup awk '{sub(/\:/," ",$1)};1' PANY_ngsLD_sub_output.ld > PANY_ngsLD_sub_output_split1.ld &
nohup awk '{sub(/\:/," ",$3)};1' PANY_ngsLD_sub_output_split1.ld > PANY_ngsLD_sub_output_split.ld &

#Create one file with all chromosomes but without chromosome names (col.1 and col.3)
nohup awk '{ print $2, $4, $5, $6, $7, $8, $9}' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_sub_output_split_nochrnames.ld &


#Create one file per chromosome:
nohup grep -E "^Mme_chr01" PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_split_chr01.ld &
nohup awk '{ print $2, $4, $5, $6, $7, $8, $9}' PANY_ngsLD_split_chr01.ld > PANY_ngsLD_split2_chr01.ld &

nohup grep -E "^Mme_chr01" PANY_ngsLD_sub_output.ld > PANY_ngsLD_chr01.ld &


nohup awk -F "Mme_chr01" PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr01.ld &
nohup awk '{ if ($1 == Mme_chr02) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr02.ld &
nohup awk '{ if ($1 == Mme_chr03) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr03.ld &
nohup awk '{ if ($1 == Mme_chr04) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr04.ld &
nohup awk '{ if ($1 == Mme_chr05) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr05.ld &
nohup awk '{ if ($1 == Mme_chr06) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr06.ld &
nohup awk '{ if ($1 == Mme_chr07) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr07.ld &
nohup awk '{ if ($1 == Mme_chr08) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr08.ld &
nohup awk '{ if ($1 == Mme_chr09) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr09.ld &
nohup awk '{ if ($1 == Mme_chr10) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr10.ld &
nohup awk '{ if ($1 == Mme_chr11) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr11.ld &
nohup awk '{ if ($1 == Mme_chr12) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr12.ld &
nohup awk '{ if ($1 == Mme_chr13) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr13.ld &
nohup awk '{ if ($1 == Mme_chr14) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr14.ld &
nohup awk '{ if ($1 == Mme_chr15) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr15.ld &
nohup awk '{ if ($1 == Mme_chr16) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr16.ld &
nohup awk '{ if ($1 == Mme_chr17) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr17.ld &
nohup awk '{ if ($1 == Mme_chr18) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr18.ld &
nohup awk '{ if ($1 == Mme_chr19) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr19.ld &
nohup awk '{ if ($1 == Mme_chr20) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr20.ld &
nohup awk '{ if ($1 == Mme_chr21) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr21.ld &
nohup awk '{ if ($1 == Mme_chr22) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr22.ld &
nohup awk '{ if ($1 == Mme_chr23) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr23.ld &
nohup awk '{ if ($1 == Mme_chr24) { print } }' PANY_ngsLD_sub_output_split.ld > PANY_ngsLD_chr24.ld &
```

```{bash, eval=F}
export PERL5LIB=/programs/PERL/lib/perl5
export PATH=/programs/ngsLD:/programs/ngsLD/scripts:$PATH

perl /workdir/arne/programs/ngsLD/scripts/prune_graph.pl --help

nohup perl /workdir/arne/programs/ngsLD/scripts/prune_graph.pl --in_file PANY_ngsLD_chr01.ld --weight_field 7 --min_weight 0.2 --max_kb_dist 5 --out pany_chr01_unlinked.id &
```

```{bash, eval=F}
nohup perl /workdir/arne/programs/ngsLD/scripts/prune_graph.pl --in_file PANY_ngsLD_sub_output.ld --max_kb_dist 5 --min_weight 0.2 --out pany_unlinked.id &
```




### Estimate LDdecay:

Sub sample full LD file (5 percent of all comparisons and) then run LD decay
```{bash, eval=F}
nohup perl -ne 'print if (rand() < .05)' PANY_ngsLD_sub_output_split_nochrnames.ld > PANY_ngsLD_output_5perc.ld &
```

```{r, eval=F}
Rscript --vanilla --slave /workdir/arne/programs/ngsLD/scripts/fit_LDdecay.R --help

nohup Rscript --vanilla --slave /workdir/arne/programs/ngsLD/scripts/fit_LDdecay.R --ld_files ngsLD_lddecay_testfiles.txt --header FALSE --col 3 --n_ind 48 --max_kb_dist 100 --out plot_pany_lddecay_allchr.pdf > pany_ld_decay_allchr_nohup.log &
```


Plot LDdecay on chr3 for PANY (check if script)
```{r, eval=F}
library(dplyr)
library(stringr)
library(ggplot2)

dfr <- read_delim("",sep="",header=F,check.names=F,stringsAsFactors=F)
colnames(dfr) <- c("dist","rsq")
dfr$distc <- cut(dfr$dist,breaks=seq(from=min(dfr$dist)-1,to=max(dfr$dist)+1,by=100000))
dfr1 <- dfr %>% group_by(distc) %>% summarise(mean=mean(rsq),median=median(rsq))
dfr1 <- dfr1 %>% mutate(start=as.integer(str_extract(str_replace_all(distc,"[\\(\\)\\[\\]]",""),"^[0-9-e+.]+")),
                        end=as.integer(str_extract(str_replace_all(distc,"[\\(\\)\\[\\]]",""),"[0-9-e+.]+$")),
                        mid=start+((end-start)/2))
                        
ggplot()+
 geom_point(data=dfr1,aes(x=start,y=mean),size=0.4,colour="grey20")+
 geom_line(data=dfr1,aes(x=start,y=mean),size=0.3,alpha=0.5,colour="grey40")+
 labs(x="Distance (Megabases)",y=expression(LD~(r^{2})))+
 scale_x_continuous(breaks=c(0,2*10^6,4*10^6,6*10^6,8*10^6),labels=c("0","2","4","6","8"))+
 theme_bw()
```



## Popuatlion genomic analyses.

Estimating within-population and between-population summary statistics using angsd and custom scripts:

Based on variable sites:
+ Pairwise genetic differentiation (Fst)
+ Population branch statistic for Northern Populations (PBS) -> e.g MAQU (Pop1), PANY(Pop2), JIGA(POP3)
+ Patterns of linkage disequilibrium decay (with ngsLD)
+ Pairwise allele frequency divergence (∆AF)

Based on all sites:
+ Nucleotide diversity (π) -> calculate pairwise ∆π
+ Tajmia's D  -> calculate pairwise ∆TajD
+ Absolute pairwise genetic divergence (dxy)

Minor allele frequency distributions:
```{r, eval=F}
JIGA.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_JIGA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
LRSC.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_LRSC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
CTVA.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_CTVA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
OINC.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_OINC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCNC.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCNC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
PANY.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_PANY_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCCT.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files//mme_MCCT_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
DXMA.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_DXMA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MDME.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MDME_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
HXNS.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_HXNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MBNS.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MBNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MAQU.maf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MAQU_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))

JIGA.maf.pop <- JIGA.maf
JIGA.maf.pop['pop'] = 'JIGA'
LRSC.maf.pop <- LRSC.maf
LRSC.maf.pop['pop'] = 'LRSC'
MCNC.maf.pop <- MCNC.maf
MCNC.maf.pop['pop'] = 'MCNC'
OINC.maf.pop <- OINC.maf
OINC.maf.pop['pop'] = 'OINC'
CTVA.maf.pop <- CTVA.maf
CTVA.maf.pop['pop'] = 'CTVA'
PANY.maf.pop <- PANY.maf
PANY.maf.pop['pop'] = 'PANY'
DXMA.maf.pop <- DXMA.maf
DXMA.maf.pop['pop'] = 'DXMA'
MDME.maf.pop <- MDME.maf
MDME.maf.pop['pop'] = 'MDME'
MBNS.maf.pop <- MBNS.maf
MBNS.maf.pop['pop'] = 'MBNS'
HXNS.maf.pop <- HXNS.maf
HXNS.maf.pop['pop'] = 'HXNS'
MAQU.maf.pop <- MAQU.maf
MAQU.maf.pop['pop'] = 'MAQU'

MAF.pop.df <- rbind(JIGA.maf.pop, LRSC.maf.pop, MCNC.maf.pop, OINC.maf.pop, CTVA.maf.pop,PANY.maf.pop,DXMA.maf.pop,MDME.maf.pop,MBNS.maf.pop,HXNS.maf.pop,MAQU.maf.pop)

MAF.pop.df$pop <- ordered(MAF.pop.df$pop, levels = c("JIGA", "LRSC", "MCNC", "OINC", "CTVA", "PANY", "DXMA", "MDME", "HXNS", "MBNS", "MAQU"))

ggplot(MAF.pop.df, aes(x = knownEM, y = pop)) + 
  geom_density_ridges(rel_min_height = 0.001)

ggplot(MAF.pop.df, aes(x = knownEM, y = pop, fill = pop)) + 
  geom_density_ridges() +
  scale_fill_manual(values = c(brewer.pal(n = 11, name = "RdBu"))) +
  ylab("Sampling site") +
  xlab("Global minor allele frequency") +
  theme(legend.position="none")
```

Derived allele frequency distributions:
```{r, eval=F}
JIGA.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/JIGA_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
LRSC.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_LRSC_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
CTVA.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_CTVA_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
OINC.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_OINC_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCNC.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCNC_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
PANY.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/PANY_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCCT.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files//mme_MCCT_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
DXMA.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_DXMA_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MDME.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MDME_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
HXNS.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_HXNS_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MBNS.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MBNS_filtsnps_daf.dafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MAQU.daf <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/MAQU_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))

hist(JIGA.daf$unknownEM)
hist(PANY.daf$unknownEM)
hist(MAQU.daf$unknownEM)


JIGA.daf.pop <- JIGA.daf
JIGA.daf.pop['pop'] = 'JIGA'
LRSC.daf.pop <- LRSC.daf
LRSC.daf.pop['pop'] = 'LRSC'
MCNC.daf.pop <- MCNC.daf
MCNC.daf.pop['pop'] = 'MCNC'
OINC.daf.pop <- OINC.daf
OINC.daf.pop['pop'] = 'OINC'
CTVA.daf.pop <- CTVA.daf
CTVA.daf.pop['pop'] = 'CTVA'
PANY.daf.pop <- PANY.daf
PANY.daf.pop['pop'] = 'PANY'
DXMA.daf.pop <- DXMA.daf
DXMA.daf.pop['pop'] = 'DXMA'
MDME.daf.pop <- MDME.daf
MDME.daf.pop['pop'] = 'MDME'
MBNS.daf.pop <- MBNS.daf
MBNS.daf.pop['pop'] = 'MBNS'
HXNS.daf.pop <- HXNS.daf
HXNS.daf.pop['pop'] = 'HXNS'
MAQU.daf.pop <- MAQU.daf
MAQU.daf.pop['pop'] = 'MAQU'

daf.pop.df <- rbind(JIGA.daf.pop, LRSC.daf.pop, MCNC.daf.pop, OINC.daf.pop, CTVA.daf.pop,PANY.daf.pop,DXMA.daf.pop,MDME.daf.pop,MBNS.daf.pop,HXNS.daf.pop,MAQU.daf.pop)

daf.pop.df$pop <- ordered(daf.pop.df$pop, levels = c("JIGA", "LRSC", "MCNC", "OINC", "CTVA", "PANY", "DXMA", "MDME", "HXNS", "MBNS", "MAQU"))

ggplot(daf.pop.df, aes(x = knownEM, y = pop)) + 
  geom_density_ridges(rel_min_height = 0.001)

ggplot(daf.pop.df, aes(x = knownEM, y = pop, fill = pop)) + 
  geom_density_ridges() +
  scale_fill_manual(values = c(brewer.pal(n = 11, name = "RdBu"))) +
  ylab("Sampling site") +
  xlab("Global minor allele frequency") +
  theme(legend.position="none")
```



Estimate Tajima's D and π in 10kb non-overlapping windows based on the SAF and folded SFS of all sites:
```{bash, eval=F}
#JIGA
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt JIGA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/JIGA_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/jiga_thetastats_folded_nohup.log & 

#LRSC
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_LRSC.txt LRSC /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/LRSC_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/LRSC_thetastats_folded_nohup.log & 

#MCNC
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCNC.txt MCNC /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/MCNC_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/MCNC_thetastats_folded_nohup.log & 

#OINC
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_OINC.txt OINC /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/OINC_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/OINC_thetastats_folded_nohup.log & 

#CTVA
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_CTVA.txt CTVA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/CTVA_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/CTVA_thetastats_folded_nohup.log & 

#PANY
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt PANY /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/PANY_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/PANY_thetastats_folded_nohup.log & 

#MMCT
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCCT.txt MCCT /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/MCCT_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/MCCT_thetastats_folded_nohup.log & 

#DXMA
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_DXMA.txt DXMA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/DXMA_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/DXMA_thetastats_folded_nohup.log & 

#MDME
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MDME.txt MDME /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/MDME_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/MDME_thetastats_folded_nohup.log & 

#MBNS
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MBNS.txt MBNS /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/MBNS_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/MBNS_thetastats_folded_nohup.log & 

#HXNS
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_HXNS.txt HXNS /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/HXNS_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/HXNS_thetastats_folded_nohup.log & 

#MAQU
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MAQU.txt MAQU /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/MAQU_allsites_folded.saf.idx /workdir/arne/results/summary_statistics/ _allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/MAQU_thetastats_folded_nohup.log & 
```

Correlation of mean genome-wide Tajima's D, theta and π vs latitude (for shared windows)
Get mean TD and π for all other pops:
```{r, eval= F}
pi.jiga <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/JIGA_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.jiga)[1] <- "region"
names(pi.jiga)[2] <- "chromo"
names(pi.jiga)[3] <- "midwin"
names(pi.jiga)[14] <- "nsites"
pi.jiga.filt <- na.omit(filter(pi.jiga, nsites >= 50))
pi.jiga.filt <- dplyr::filter(pi.jiga.filt, grepl('Mme_', chromo)) 
pi.jiga.filt$pi <- pi.jiga.filt$tP/pi.jiga.filt$nsites
jiga.theta <- mean((pi.jiga.filt$tW/pi.jiga.filt$nsites))
jiga.pi <- mean(pi.jiga.filt$pi)
jiga.TD <- mean(pi.jiga.filt$Tajima)

pi.lrsc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/LRSC_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.lrsc)[1] <- "region"
names(pi.lrsc)[2] <- "chromo"
names(pi.lrsc)[3] <- "midwin"
names(pi.lrsc)[14] <- "nsites"
pi.lrsc.filt <- na.omit(filter(pi.lrsc, nsites >= 50))
pi.lrsc.filt <- dplyr::filter(pi.lrsc.filt, grepl('Mme_', chromo)) 
pi.lrsc.filt$pi <- pi.lrsc.filt$tP/pi.lrsc.filt$nsites
lrsc.theta <- mean((pi.lrsc.filt$tW/pi.lrsc.filt$nsites))
lrsc.pi <- mean(pi.lrsc.filt$pi)
lrsc.TD <- mean(pi.lrsc.filt$Tajima)

pi.mcnc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MCNC_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.mcnc)[1] <- "region"
names(pi.mcnc)[2] <- "chromo"
names(pi.mcnc)[3] <- "midwin"
names(pi.mcnc)[14] <- "nsites"
pi.mcnc.filt <- na.omit(filter(pi.mcnc, nsites >= 50))
pi.mcnc.filt <- dplyr::filter(pi.mcnc.filt, grepl('Mme_', chromo)) 
pi.mcnc.filt$pi <- pi.mcnc.filt$tP/pi.mcnc.filt$nsites
mcnc.theta <- mean((pi.mcnc.filt$tW/pi.mcnc.filt$nsites))
mcnc.pi <- mean(pi.mcnc.filt$pi)
mcnc.TD <- mean(pi.mcnc.filt$Tajima)

pi.oinc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/OINC_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.oinc)[1] <- "region"
names(pi.oinc)[2] <- "chromo"
names(pi.oinc)[3] <- "midwin"
names(pi.oinc)[14] <- "nsites"
pi.oinc.filt <- na.omit(filter(pi.oinc, nsites >= 50))
pi.oinc.filt <- dplyr::filter(pi.oinc.filt, grepl('Mme_', chromo)) 
pi.oinc.filt$pi <- pi.oinc.filt$tP/pi.oinc.filt$nsites
oinc.theta <- mean((pi.oinc.filt$tW/pi.oinc.filt$nsites))
oinc.pi <- mean(pi.oinc.filt$pi)
oinc.TD <- mean(pi.oinc.filt$Tajima)

pi.ctva <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/CTVA_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.ctva)[1] <- "region"
names(pi.ctva)[2] <- "chromo"
names(pi.ctva)[3] <- "midwin"
names(pi.ctva)[14] <- "nsites"
pi.ctva.filt <- na.omit(filter(pi.ctva, nsites >= 50))
pi.ctva.filt <- dplyr::filter(pi.ctva.filt, grepl('Mme_', chromo)) 
pi.ctva.filt$pi <- pi.ctva.filt$tP/pi.ctva.filt$nsites
ctva.theta <- mean((pi.ctva.filt$tW/pi.ctva.filt$nsites))
ctva.pi <- mean(pi.ctva.filt$pi)
ctva.TD <- mean(pi.ctva.filt$Tajima)

pi.pany <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/PANY_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.pany)[1] <- "region"
names(pi.pany)[2] <- "chromo"
names(pi.pany)[3] <- "midwin"
names(pi.pany)[14] <- "nsites"
pi.pany.filt <- na.omit(filter(pi.pany, nsites >= 50))
pi.pany.filt <- dplyr::filter(pi.pany.filt, grepl('Mme_', chromo)) 
pi.pany.filt$pi <- pi.pany.filt$tP/pi.pany.filt$nsites
pany.theta <- mean((pi.pany.filt$tW/pi.pany.filt$nsites))
pany.pi <- mean(pi.pany.filt$pi)
pany.TD <- mean(pi.pany.filt$Tajima)

pi.mcct <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MCCT_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.mcct)[1] <- "region"
names(pi.mcct)[2] <- "chromo"
names(pi.mcct)[3] <- "midwin"
names(pi.mcct)[14] <- "nsites"
pi.mcct.filt <- na.omit(filter(pi.mcct, nsites >= 50))
pi.mcct.filt <- dplyr::filter(pi.mcct.filt, grepl('Mme_', chromo)) 
pi.mcct.filt$pi <- pi.mcct.filt$tP/pi.mcct.filt$nsites
mcct.theta <- mean((pi.mcct.filt$tW/pi.mcct.filt$nsites))
mcct.pi <- mean(pi.mcct.filt$pi)
mcct.TD <- mean(pi.mcct.filt$Tajima)


pi.dxma <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/DXMA_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.dxma)[1] <- "region"
names(pi.dxma)[2] <- "chromo"
names(pi.dxma)[3] <- "midwin"
names(pi.dxma)[14] <- "nsites"
pi.dxma.filt <- na.omit(filter(pi.dxma, nsites >= 50))
pi.dxma.filt <- dplyr::filter(pi.dxma.filt, grepl('Mme_', chromo)) 
pi.dxma.filt$pi <- pi.dxma.filt$tP/pi.dxma.filt$nsites
dxma.theta <- mean((pi.dxma.filt$tW/pi.dxma.filt$nsites))
dxma.pi <- mean(pi.dxma.filt$pi)
dxma.TD <- mean(pi.dxma.filt$Tajima)

pi.mdme <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MDME_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.mdme)[1] <- "region"
names(pi.mdme)[2] <- "chromo"
names(pi.mdme)[3] <- "midwin"
names(pi.mdme)[14] <- "nsites"
pi.mdme.filt <- na.omit(filter(pi.mdme, nsites >= 50))
pi.mdme.filt <- dplyr::filter(pi.mdme.filt, grepl('Mme_', chromo)) 
pi.mdme.filt$pi <- pi.mdme.filt$tP/pi.mdme.filt$nsites
mdme.theta <- mean((pi.mdme.filt$tW/pi.mdme.filt$nsites))
mdme.pi <- mean(pi.mdme.filt$pi)
mdme.TD <- mean(pi.mdme.filt$Tajima)

pi.mbns <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MBNS_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.mbns)[1] <- "region"
names(pi.mbns)[2] <- "chromo"
names(pi.mbns)[3] <- "midwin"
names(pi.mbns)[14] <- "nsites"
pi.mbns.filt <- na.omit(filter(pi.mbns, nsites >= 50))
pi.mbns.filt <- dplyr::filter(pi.mbns.filt, grepl('Mme_', chromo)) 
pi.mbns.filt$pi <- pi.mbns.filt$tP/pi.mbns.filt$nsites
mbns.theta <- mean((pi.mbns.filt$tW/pi.mbns.filt$nsites))
mbns.pi <- mean(pi.mbns.filt$pi)
mbns.TD <- mean(pi.mbns.filt$Tajima)

pi.hxns <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/HXNS_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.hxns)[1] <- "region"
names(pi.hxns)[2] <- "chromo"
names(pi.hxns)[3] <- "midwin"
names(pi.hxns)[14] <- "nsites"
pi.hxns.filt <- na.omit(filter(pi.hxns, nsites >= 50))
pi.hxns.filt <- dplyr::filter(pi.hxns.filt, grepl('Mme_', chromo)) 
pi.hxns.filt$pi <- pi.hxns.filt$tP/pi.hxns.filt$nsites
hxns.theta <- mean((pi.hxns.filt$tW/pi.hxns.filt$nsites))
hxns.pi <- mean(pi.hxns.filt$pi)
hxns.TD <- mean(pi.hxns.filt$Tajima)

pi.maqu <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MAQU_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.maqu)[1] <- "region"
names(pi.maqu)[2] <- "chromo"
names(pi.maqu)[3] <- "midwin"
names(pi.maqu)[14] <- "nsites"
pi.maqu.filt <- na.omit(filter(pi.maqu, nsites >= 50))
pi.maqu.filt <- dplyr::filter(pi.maqu.filt, grepl('Mme_', chromo)) 
pi.maqu.filt$pi <- pi.maqu.filt$tP/pi.maqu.filt$nsites
maqu.theta <- mean((pi.maqu.filt$tW/pi.maqu.filt$nsites))
maqu.pi <- mean(pi.maqu.filt$pi)
maqu.TD <- mean(pi.maqu.filt$Tajima)
```

Merge all estimates with latitude in table
```{r, eval= F}
theta <- as.data.frame(rbind(jiga.theta,
                             lrsc.theta,
                             mcnc.theta,
                             oinc.theta,
                             ctva.theta,
                             pany.theta,
                             mcct.theta,
                             dxma.theta,
                             mdme.theta,
                             mbns.theta,
                             hxns.theta,
                             maqu.theta))

TD <- as.data.frame(rbind(jiga.TD,
                             lrsc.TD,
                             mcnc.TD,
                             oinc.TD,
                             ctva.TD,
                             pany.TD,
                             mcct.TD,
                             dxma.TD,
                             mdme.TD,
                             mbns.TD,
                             hxns.TD,
                             maqu.TD))

pi <- as.data.frame(rbind(jiga.pi,
                             lrsc.pi,
                             mcnc.pi,
                             oinc.pi,
                             ctva.pi,
                             pany.pi,
                             mcct.pi,
                             dxma.pi,
                             mdme.pi,
                             mbns.pi,
                             hxns.pi,
                             maqu.pi))

sumstats.mean <- as.data.frame(cbind(theta, TD, pi))
names(sumstats.mean) <- c("theta","TD","pi")

lat <- c(31.0, 33.9,34.7,35.8,37.9,40.8,41.3,42.0,44.4,45.3,44.6,47.4)
batch <- c("A", "B","B","B","B","A","C","B","B","A","B","A")
sumstats.mean.lat.df <- as_tibble(cbind(sumstats.mean, lat, batch))
```

Analyse correlations of summary statistics with latitude:

Watterson's theta:
```{r, eval= F}
summary(lm(theta ~ lat, data = sumstats.mean.lat.df))
ggplot(data = sumstats.mean.lat.df, aes(x=lat, y=theta)) +
  geom_point() +
  geom_smooth(method = "lm")

sumstats.mean.lat.df$lat2 <- (sumstats.mean.lat.df$lat)^2
summary(lm(theta ~ lat + lat2, data = sumstats.mean.lat.df))
theta.lat.quad <- ggplot(data = sumstats.mean.lat.df, aes(x=lat, y=theta)) +
  stat_smooth(method = "lm", formula = y ~ poly(x, 2), size = 1, colour = "gray20") +
  geom_point(size = 2) +
  annotate(geom="text", x=35, y=0.01, label="R2 = 0.69, P = 0.0020",
              color="black") +
  ylab("Watterson's theta") +
  xlab("latitude") +
  theme_cowplot()
theta.lat.quad
```

Tajima's D:
```{r, eval= F}
summary(lm(TD ~ lat, data = sumstats.mean.lat.df))
ggplot(data = sumstats.mean.lat.df, aes(x=lat, y=TD)) +
  geom_point() +
  geom_smooth(method = "lm")

summary(lm(TD ~ lat + lat2, data = sumstats.mean.lat.df))
TD.lat.quad <- ggplot(data = sumstats.mean.lat.df, aes(x=lat, y=TD)) +
  stat_smooth(method = "lm", formula = y ~ poly(x, 2), size = 1, colour = "gray20") +
  geom_point(size = 2) +
  annotate(geom="text", x=35, y=1, label="R2 = 0.71, P = 0.0016",
              color="black") +
  ylab("Tajima's D") +
  xlab("latitude") +
  theme_cowplot()
TD.lat.quad
```

Nucleotide diversity:
```{r, eval= F}
summary(lm(pi ~ lat, data = sumstats.mean.lat.df))
ggplot(data = sumstats.mean.lat.df, aes(x=lat, y=pi)) +
  geom_point() +
  geom_smooth(method = "lm")

summary(lm(pi ~ lat + lat2, data = sumstats.mean.lat.df))
pi.lat.quad <- ggplot(data = sumstats.mean.lat.df, aes(x=lat, y=pi)) +
  stat_smooth(method = "lm", formula = y ~ poly(x, 2), size = 1, colour = "gray20") +
  geom_point(size = 2) +
  annotate(geom="text", x=42, y=0.015, label="R2 = 0.86, P = 5.45e-05",
              color="black") +
  ylab("π") +
  xlab("latitude") +
  theme_cowplot()
pi.lat.quad
```

```{r, eval= F}
plot_grid(theta.lat.quad, TD.lat.quad, pi.lat.quad, align = "v", nrow = 1)
```

Quadratic models fit better than linear models (explain higher proportion of variance) in all cases. Linear models not plotted 


### Patterns of diversity along latitude for non-inversion regions


1. Identify all windows outside of any inversion for all populations

```{r, eval= F}
sv <- read.csv("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
svs.major = svs.major[,c(2,10,11)]
names(svs.major) = c("seqnames","start","end")
svs.major.ranges =  svs.major %>% as_granges()
sv.maria <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4)]
names(sv.maria2) = c("seqnames","start","end")
sv.maria.chr.ranges =  sv.maria2 %>% as_granges()
allSVs = rbind(svs.major, sv.maria2)
allSVs$start2 = (round(allSVs$start * 1000000))
allSVs$end2 = (round(allSVs$end * 1000000))
allSVs = allSVs[,c(1,4,5)]
names(allSVs) = c("seqnames","start","end")
allSVs.ranges =  allSVs %>% as_granges()

pi.jiga.filt$start = pi.jiga.filt$midwin - 4999
pi.jiga.filt$end = pi.jiga.filt$midwin + 5000
pi.jiga.filt = pi.jiga.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.jiga.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.jiga.filt.ranges =  pi.jiga.filt %>% as_granges()
pi.jiga.filt.noinv.ranges =  pi.jiga.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
jiga.noinv.pi = as_tibble(pi.jiga.filt.noinv.ranges %>% summarise(pi = mean(pi)))
jiga.noinv.TD = as_tibble(pi.jiga.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))
pi.jiga.filt.inv.ranges =  pi.jiga.filt.ranges %>% filter_by_overlaps(allSVs.ranges)

pi.lrsc.filt$start = pi.lrsc.filt$midwin - 4999
pi.lrsc.filt$end = pi.lrsc.filt$midwin + 5000
pi.lrsc.filt = pi.lrsc.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.lrsc.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.lrsc.filt.ranges =  pi.lrsc.filt %>% as_granges()
pi.lrsc.filt.noinv.ranges =  pi.lrsc.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
lrsc.noinv.pi = as_tibble(pi.lrsc.filt.noinv.ranges %>% summarise(pi = mean(pi)))
lrsc.noinv.TD = as_tibble(pi.lrsc.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.mcnc.filt$start = pi.mcnc.filt$midwin - 4999
pi.mcnc.filt$end = pi.mcnc.filt$midwin + 5000
pi.mcnc.filt = pi.mcnc.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.mcnc.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.mcnc.filt.ranges =  pi.mcnc.filt %>% as_granges()
pi.mcnc.filt.noinv.ranges =  pi.mcnc.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
mcnc.noinv.pi = as_tibble(pi.mcnc.filt.noinv.ranges %>% summarise(pi = mean(pi)))
mcnc.noinv.TD = as_tibble(pi.mcnc.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.oinc.filt$start = pi.oinc.filt$midwin - 4999
pi.oinc.filt$end = pi.oinc.filt$midwin + 5000
pi.oinc.filt = pi.oinc.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.oinc.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.oinc.filt.ranges =  pi.oinc.filt %>% as_granges()
pi.oinc.filt.noinv.ranges =  pi.oinc.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
oinc.noinv.pi = as_tibble(pi.oinc.filt.noinv.ranges %>% summarise(pi = mean(pi)))
oinc.noinv.TD = as_tibble(pi.oinc.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))
pi.oinc.filt.inv.ranges =  pi.oinc.filt.ranges %>% filter_by_overlaps(allSVs.ranges)

pi.ctva.filt$start = pi.ctva.filt$midwin - 4999
pi.ctva.filt$end = pi.ctva.filt$midwin + 5000
pi.ctva.filt = pi.ctva.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.ctva.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.ctva.filt.ranges =  pi.ctva.filt %>% as_granges()
pi.ctva.filt.noinv.ranges =  pi.ctva.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
ctva.noinv.pi = as_tibble(pi.ctva.filt.noinv.ranges %>% summarise(pi = mean(pi)))
ctva.noinv.TD = as_tibble(pi.ctva.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.pany.filt$start = pi.pany.filt$midwin - 4999
pi.pany.filt$end = pi.pany.filt$midwin + 5000
pi.pany.filt = pi.pany.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.pany.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.pany.filt.ranges =  pi.pany.filt %>% as_granges()
pi.pany.filt.noinv.ranges =  pi.pany.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
pany.noinv.pi = as_tibble(pi.pany.filt.noinv.ranges %>% summarise(pi = mean(pi)))
pany.noinv.TD = as_tibble(pi.pany.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.mcct.filt$start = pi.mcct.filt$midwin - 4999
pi.mcct.filt$end = pi.mcct.filt$midwin + 5000
pi.mcct.filt = pi.mcct.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.mcct.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.mcct.filt.ranges =  pi.mcct.filt %>% as_granges()
pi.mcct.filt.noinv.ranges =  pi.mcct.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
mcct.noinv.pi = as_tibble(pi.mcct.filt.noinv.ranges %>% summarise(pi = mean(pi)))
mcct.noinv.TD = as_tibble(pi.mcct.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.dxma.filt$start = pi.dxma.filt$midwin - 4999
pi.dxma.filt$end = pi.dxma.filt$midwin + 5000
pi.dxma.filt = pi.dxma.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.dxma.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.dxma.filt.ranges =  pi.dxma.filt %>% as_granges()
pi.dxma.filt.noinv.ranges =  pi.dxma.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
dxma.noinv.pi = as_tibble(pi.dxma.filt.noinv.ranges %>% summarise(pi = mean(pi)))
dxma.noinv.TD = as_tibble(pi.dxma.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.mdme.filt$start = pi.mdme.filt$midwin - 4999
pi.mdme.filt$end = pi.mdme.filt$midwin + 5000
pi.mdme.filt = pi.mdme.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.mdme.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.mdme.filt.ranges =  pi.mdme.filt %>% as_granges()
pi.mdme.filt.noinv.ranges =  pi.mdme.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
mdme.noinv.pi = as_tibble(pi.mdme.filt.noinv.ranges %>% summarise(pi = mean(pi)))
mdme.noinv.TD = as_tibble(pi.mdme.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.mbns.filt$start = pi.mbns.filt$midwin - 4999
pi.mbns.filt$end = pi.mbns.filt$midwin + 5000
pi.mbns.filt = pi.mbns.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.mbns.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.mbns.filt.ranges =  pi.mbns.filt %>% as_granges()
pi.mbns.filt.noinv.ranges =  pi.mbns.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
mbns.noinv.pi = as_tibble(pi.mbns.filt.noinv.ranges %>% summarise(pi = mean(pi)))
mbns.noinv.TD = as_tibble(pi.mbns.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.hxns.filt$start = pi.hxns.filt$midwin - 4999
pi.hxns.filt$end = pi.hxns.filt$midwin + 5000
pi.hxns.filt = pi.hxns.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.hxns.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.hxns.filt.ranges =  pi.hxns.filt %>% as_granges()
pi.hxns.filt.noinv.ranges =  pi.hxns.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
hxns.noinv.pi = as_tibble(pi.hxns.filt.noinv.ranges %>% summarise(pi = mean(pi)))
hxns.noinv.TD = as_tibble(pi.hxns.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

pi.maqu.filt$start = pi.maqu.filt$midwin - 4999
pi.maqu.filt$end = pi.maqu.filt$midwin + 5000
pi.maqu.filt = pi.maqu.filt[,c(2,16,17,3,4,5,9,14,15)]
names(pi.maqu.filt) = c("seqnames","start","end","mid","tW","tP","Tajima","nsites","pi")
pi.maqu.filt.ranges =  pi.maqu.filt %>% as_granges()
pi.maqu.filt.noinv.ranges =  pi.maqu.filt.ranges %>% filter_by_non_overlaps(allSVs.ranges)
maqu.noinv.pi = as_tibble(pi.maqu.filt.noinv.ranges %>% summarise(pi = mean(pi)))
maqu.noinv.TD = as_tibble(pi.maqu.filt.noinv.ranges %>% summarise(TD = mean(Tajima)))

noinv.pi <- as.data.frame(rbind(jiga.noinv.pi,
                             lrsc.noinv.pi,
                             mcnc.noinv.pi,
                             oinc.noinv.pi,
                             ctva.noinv.pi,
                             pany.noinv.pi,
                             mcct.noinv.pi,
                             dxma.noinv.pi,
                             mdme.noinv.pi,
                             mbns.noinv.pi,
                             hxns.noinv.pi,
                             maqu.noinv.pi))

noinv.TD <- as.data.frame(rbind(jiga.noinv.TD,
                             lrsc.noinv.TD,
                             mcnc.noinv.TD,
                             oinc.noinv.TD,
                             ctva.noinv.TD,
                             pany.noinv.TD,
                             mcct.noinv.TD,
                             dxma.noinv.TD,
                             mdme.noinv.TD,
                             mbns.noinv.TD,
                             hxns.noinv.TD,
                             maqu.noinv.TD))
```


```{r, eval= F}
sumstats.noinv.mean <- as.data.frame(cbind(noinv.TD, noinv.pi))
names(sumstats.noinv.mean) <- c("TD","pi")

lat <- c(31.0, 33.9,34.7,35.8,37.9,40.8,41.3,42.0,44.4,45.3,44.6,47.4)
batch <- c("A", "B","B","B","B","A","C","B","B","A","B","A")
sumstats.noinv.mean.lat.df <- as_tibble(cbind(sumstats.noinv.mean, lat, batch))
```

Analyse correlations of summary statistics with latitude:

Tajima's D:
```{r, eval= F}
summary(lm(TD ~ lat, data = sumstats.noinv.mean.lat.df))
ggplot(data = sumstats.noinv.mean.lat.df, aes(x=lat, y=TD)) +
  geom_point() +
  geom_smooth(method = "lm")

summary(lm(TD ~ lat + lat2, data = sumstats.noinv.mean.lat.df))
TD.lat.quad <- ggplot(data = sumstats.noinv.mean.lat.df, aes(x=lat, y=TD)) +
  stat_smooth(method = "lm", formula = y ~ poly(x, 2), size = 1, colour = "gray20") +
  geom_point(size = 2) +
  annotate(geom="text", x=35, y=1, label="R2 = 0.71, P = 0.0016",
              color="black") +
  ylab("Tajima's D") +
  xlab("latitude") +
  theme_cowplot()
TD.lat.quad
```

Nucleotide diversity:
```{r, eval= F}
summary(lm(pi ~ lat, data = sumstats.noinv.mean.lat.df))
ggplot(data = sumstats.noinv.mean.lat.df, aes(x=lat, y=pi)) +
  geom_point() +
  geom_smooth(method = "lm")

summary(lm(pi ~ lat + lat2, data = sumstats.noinv.mean.lat.df))
pi.lat.quad <- ggplot(data = sumstats.noinv.mean.lat.df, aes(x=lat, y=pi)) +
  stat_smooth(method = "lm", formula = y ~ poly(x, 2), size = 1, colour = "gray20") +
  geom_point(size = 2) +
  annotate(geom="text", x=42, y=0.015, label="R2 = 0.86, P = 5.45e-05",
              color="black") +
  ylab("π") +
  xlab("latitude") +
  theme_cowplot()
pi.lat.quad
```

```{r, eval= F}
plot_grid(TD.lat.quad, pi.lat.quad, align = "v", nrow = 1)
```

Quadratic models fit better than linear models (explain higher proportion of variance) in all cases. Linear models not plotted 


### Compare TD an pi within and outside inversions
```{r, eval= F}
#JIGA
pi.jiga.filt.inv.df = as_tibble(pi.jiga.filt.inv.ranges)
pi.jiga.filt.noinv.df = as_tibble(pi.jiga.filt.noinv.ranges)

ggplot() +
  geom_histogram(data = pi.jiga.filt.noinv.df, aes(x = Tajima), fill = "blue", alpha = 0.5) +
  geom_histogram(data = pi.jiga.filt.inv.df, aes(x = Tajima), fill = "red", alpha = 0.5)

ggplot() +
  geom_histogram(data = pi.jiga.filt.noinv.df, aes(x = pi), fill = "blue", alpha = 0.5) +
  geom_histogram(data = pi.jiga.filt.inv.df, aes(x = pi), fill = "red", alpha = 0.5)


#OINC
pi.oinc.filt.inv.df = as_tibble(pi.oinc.filt.inv.ranges)
pi.oinc.filt.noinv.df = as_tibble(pi.oinc.filt.noinv.ranges)

ggplot() +
  geom_histogram(data = pi.oinc.filt.noinv.df, aes(x = Tajima), fill = "blue", alpha = 0.5) +
  geom_histogram(data = pi.oinc.filt.inv.df, aes(x = Tajima), fill = "red", alpha = 0.5)

ggplot() +
  geom_histogram(data = pi.oinc.filt.noinv.df, aes(x = pi), fill = "blue", alpha = 0.5) +
  geom_histogram(data = pi.oinc.filt.inv.df, aes(x = pi), fill = "red", alpha = 0.5)
```








#### Test the effect of batch on estimates (e.g. π) in linear and quadratic model:
```{r, eval= F}
summary(lm(pi ~ batch + lat, data = sumstats.mean.lat.df))
sumstats.mean.lat.df$lat2 <- (sumstats.mean.lat.df$lat)^2
summary(lm(pi ~ lat + lat2 + batch, data = sumstats.mean.lat.df))
```

Batch (read length) has no measurable effect on nucleotide diversity (p of batch > 0.5).









### Plot Tajima's D along the genome for each population:

JIGA:
```{r, eval= F}
tajD.jiga <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/JIGA_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(tajD.jiga)[1] <- "region"
names(tajD.jiga)[2] <- "chromo"
names(tajD.jiga)[3] <- "midwin"
names(tajD.jiga)[14] <- "nsites"
names(tajD.jiga)[9] <- "tajimaD"

hist(tajD.jiga$nsites, breaks = 100)
tajD.jiga.filt <- na.omit(filter(tajD.jiga, nsites >= 50))
tajD.jiga.filt <- dplyr::filter(tajD.jiga.filt, grepl('Mme_', chromo)) 
#Plot results with highlighted outliers and relativ positions using ggplot:
jiga.data <- tajD.jiga.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(tajD.jiga.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = jiga.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

jiga.tajD <- ggplot() +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("Tajima's D") +
  xlab("Chromosome 1 - 24") +
  ylim(-4, 4) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))

qtl <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
sv.maria <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4,2)]
names(sv.maria)[1] = "chromo"
sv <- read.csv("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
names(svs.major)[2] = "chromo"

jiga.td = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = -3, ymax=2), alpha=0.5) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = -3.5, ymax=-3), fill = "firebrick", alpha=0.5) +
  geom_point(data = jiga.data, aes(x=(midwin/1000000), y=tajimaD), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = quantile(jiga.data$tajimaD, prob = 1 - 99/100, na.rm = TRUE), colour = "firebrick") +
  ylab("Tajima's D") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
jiga.td
```

```{r, eval= F}
#oinc:
tajD.oinc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/OINC_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(tajD.oinc)[1] <- "region"
names(tajD.oinc)[2] <- "chromo"
names(tajD.oinc)[3] <- "midwin"
names(tajD.oinc)[14] <- "nsites"
names(tajD.oinc)[9] <- "tajimaD"

hist(tajD.oinc$nsites, breaks = 100)
tajD.oinc.filt <- na.omit(filter(tajD.oinc, nsites >= 50))
tajD.oinc.filt <- dplyr::filter(tajD.oinc.filt, grepl('Mme_', chromo)) 

#Plot results with highlighted outliers and relativ positions using ggplot:
oinc.data <- tajD.oinc.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(tajD.oinc.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = oinc.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

oinc.tajD <- ggplot(oinc.data, aes(x=BPcum, y=tajimaD)) +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("Tajima's D") +
  xlab("Chromosome 1 - 24") +
  ylim(-4, 4) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))

qtl <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
sv.maria <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4,2)]
names(sv.maria)[1] = "chromo"
sv <- read.csv("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
names(svs.major)[2] = "chromo"

oinc.td = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = -3, ymax=1), alpha=0.5) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = -3.5, ymax=-3), fill = "firebrick", alpha=0.5) +
  geom_point(data = oinc.data, aes(x=(midwin/1000000), y=tajimaD), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = quantile(oinc.data$tajimaD, prob = 1 - 99/100, na.rm = TRUE), colour = "firebrick") +
  ylab("Tajima's D") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
oinc.td
```

Plot Tajima's D for PANY along the genome:
```{r, eval= F}
#PANY:
tajD.PANY <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/PANY_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(tajD.PANY)[1] <- "region"
names(tajD.PANY)[2] <- "chromo"
names(tajD.PANY)[3] <- "midwin"
names(tajD.PANY)[14] <- "nsites"
names(tajD.PANY)[9] <- "tajimaD"

hist(tajD.PANY$nsites, breaks = 100)
tajD.PANY.filt <- na.omit(filter(tajD.PANY, nsites >= 50))
tajD.PANY.filt <- dplyr::filter(tajD.PANY.filt, grepl('Mme_', chromo)) 

#Plot results with highlighted outliers and relativ positions using ggplot:
PANY.data <- tajD.PANY.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(tajD.PANY.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = PANY.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

qtl <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
sv.maria <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4,2)]
names(sv.maria)[1] = "chromo"
sv <- read.csv("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
names(svs.major)[2] = "chromo"

pany.td = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = -3, ymax=1), alpha=0.5) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = -3.5, ymax=-3), fill = "firebrick", alpha=0.5) +
  geom_point(data = PANY.data, aes(x=(midwin/1000000), y=tajimaD), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = quantile(PANY.data$tajimaD, prob = 1 - 99/100, na.rm = TRUE), colour = "firebrick") +
  ylab("Tajima's D") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
pany.td
```

```{r, eval= F}
#mdme:
tajD.mdme <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MDME_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(tajD.mdme)[1] <- "region"
names(tajD.mdme)[2] <- "chromo"
names(tajD.mdme)[3] <- "midwin"
names(tajD.mdme)[14] <- "nsites"
names(tajD.mdme)[9] <- "tajimaD"

hist(tajD.mdme$nsites, breaks = 100)
tajD.mdme.filt <- na.omit(filter(tajD.mdme, nsites >= 50))
tajD.mdme.filt <- dplyr::filter(tajD.mdme.filt, grepl('Mme_', chromo)) 

#Plot results with highlighted outliers and relativ positions using ggplot:
mdme.data <- tajD.mdme.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(tajD.mdme.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = mdme.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

qtl <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
sv.maria <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4,2)]
names(sv.maria)[1] = "chromo"
sv <- read.csv("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
names(svs.major)[2] = "chromo"

mdme.td = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = -3, ymax=1), alpha=0.5) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = -3.5, ymax=-3), fill = "firebrick", alpha=0.5) +
  geom_point(data = mdme.data, aes(x=(midwin/1000000), y=tajimaD), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = quantile(mdme.data$tajimaD, prob = 1 - 99/100, na.rm = TRUE), colour = "firebrick") +
  ylab("Tajima's D") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
mdme.td
```

Plot Tajima's D for MBNS along the genome:
```{r, eval= F}
#MBNS:
tajD.MBNS <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MBNS_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

sum(tajD.MBNS$nSites)

names(tajD.MBNS)[1] <- "region"
names(tajD.MBNS)[2] <- "chromo"
names(tajD.MBNS)[3] <- "midwin"
names(tajD.MBNS)[14] <- "nsites"
names(tajD.MBNS)[9] <- "tajimaD"

hist(tajD.MBNS$nsites, breaks = 100)
tajD.MBNS.filt <- na.omit(filter(tajD.MBNS, nsites >= 50))
tajD.MBNS.filt <- dplyr::filter(tajD.MBNS.filt, grepl('Mme_', chromo)) 

#Plot results with highlighted outliers and relativ positions using ggplot:
MBNS.data <- tajD.MBNS.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(tajD.MBNS.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = MBNS.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

qtl <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
sv.maria <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4,2)]
names(sv.maria)[1] = "chromo"
sv <- read.csv("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
names(svs.major)[2] = "chromo"

mbns.td = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = -3, ymax=1), alpha=0.5) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = -3.5, ymax=-3), fill = "firebrick", alpha=0.5) +
  geom_point(data = MBNS.data, aes(x=(midwin/1000000), y=tajimaD), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = quantile(MBNS.data$tajimaD, prob = 1 - 99/100, na.rm = TRUE), colour = "firebrick") +
  ylab("Tajima's D") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
mbns.td
```

Plot Tajima's D for MAQU along the genome:
```{r, eval= F}
#MAQU:
tajD.MAQU <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MAQU_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(tajD.MAQU)[1] <- "region"
names(tajD.MAQU)[2] <- "chromo"
names(tajD.MAQU)[3] <- "midwin"
names(tajD.MAQU)[14] <- "nsites"
names(tajD.MAQU)[9] <- "tajimaD"

hist(tajD.MAQU$nsites, breaks = 100)
tajD.MAQU.filt <- na.omit(filter(tajD.MAQU, nsites >= 50))
tajD.MAQU.filt <- dplyr::filter(tajD.MAQU.filt, grepl('Mme_', chromo)) 

#Plot results with highlighted outliers and relativ positions using ggplot:
maqu.data <- tajD.MAQU.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(tajD.MAQU.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = maqu.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

qtl <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
sv.maria <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4,2)]
names(sv.maria)[1] = "chromo"
sv <- read.csv("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
names(svs.major)[2] = "chromo"

maqu.td = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = -3, ymax=1), alpha=0.5) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = -3.5, ymax=-3), fill = "firebrick", alpha=0.5) +
  geom_point(data = maqu.data, aes(x=(midwin/1000000), y=tajimaD), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = quantile(maqu.data$tajimaD, prob = 1 - 99/100, na.rm = TRUE), colour = "firebrick") +
  ylab("Tajima's D") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
maqu.td
```


```{r, eval=F}
td.grid = maqu.td / mbns.td / mdme.td / pany.td / oinc.td / jiga.td
```


Plot π for JIGA along the genome:
```{r, eval= F}
#JIGA:
pi.jiga <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/JIGAallsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(pi.jiga)[1] <- "region"
names(pi.jiga)[2] <- "chromo"
names(pi.jiga)[3] <- "midwin"
names(pi.jiga)[14] <- "nsites"

hist(pi.jiga$nsites, breaks = 100)
pi.jiga.filt <- na.omit(filter(pi.jiga, nsites >= 50))
pi.jiga.filt <- dplyr::filter(pi.jiga.filt, grepl('Mme_', chromo)) 

hist((pi.jiga$tP/pi.jiga$nsites), breaks = 100)

pi.jiga.filt$pi <- pi.jiga.filt$tP/pi.jiga.filt$nsites
mean(pi.jiga.filt$pi)


#Plot results with highlighted outliers and relativ positions using ggplot:
jiga.data <- pi.jiga.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(pi.jiga.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = jiga.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

jiga.pi <- ggplot(jiga.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))
jiga.pi

jiga.pi.smooth <- ggplot(jiga.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_smooth(aes(color=as.factor(chromo)), method = "loess", se = F, span = 0.2) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))
```

Plot π for pany along the genome:
```{r, eval= F}
#pany:
pi.pany <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/PANYallsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(pi.pany)[1] <- "region"
names(pi.pany)[2] <- "chromo"
names(pi.pany)[3] <- "midwin"
names(pi.pany)[14] <- "nsites"

hist(pi.pany$nsites, breaks = 100)
pi.pany.filt <- na.omit(filter(pi.pany, nsites >= 50))
pi.pany.filt <- dplyr::filter(pi.pany.filt, grepl('Mme_', chromo)) 

hist((pi.pany$tP/pi.pany$nsites), breaks = 100)

pi.pany.filt$pi <- pi.pany.filt$tP/pi.pany.filt$nsites
mean(pi.pany.filt$pi)



#Plot results with highlighted outliers and relativ positions using ggplot:
pany.data <- pi.pany.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(pi.pany.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = pany.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

pany.pi <- ggplot(pany.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))
pany.pi

pany.pi.smooth <- ggplot(pany.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_smooth(aes(color=as.factor(chromo)), method = "loess", se = F, span = 0.2) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))
```

Plot π for MBNS along the genome:
```{r, eval= F}
#MBNS:
pi.MBNS <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MBNSallsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(pi.MBNS)[1] <- "region"
names(pi.MBNS)[2] <- "chromo"
names(pi.MBNS)[3] <- "midwin"
names(pi.MBNS)[14] <- "nsites"

hist(pi.MBNS$nsites, breaks = 100)
pi.MBNS.filt <- na.omit(filter(pi.MBNS, nsites >= 50))
pi.MBNS.filt <- dplyr::filter(pi.MBNS.filt, grepl('Mme_', chromo)) 

hist((pi.MBNS$tP/pi.MBNS$nsites), breaks = 100)

pi.MBNS.filt$pi <- pi.MBNS.filt$tP/pi.MBNS.filt$nsites
mean(pi.MBNS.filt$pi)



#Plot results with highlighted outliers and relativ positions using ggplot:
MBNS.data <- pi.MBNS.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(pi.MBNS.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = MBNS.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

MBNS.pi <- ggplot(MBNS.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))

MBNS.pi.smooth <- ggplot(MBNS.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_smooth(aes(color=as.factor(chromo)), method = "loess", se = F, span = 0.2) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))
```

Plot π for maqu along the genome:
```{r, eval= F}
#maqu:
pi.maqu <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/maquallsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

names(pi.maqu)[1] <- "region"
names(pi.maqu)[2] <- "chromo"
names(pi.maqu)[3] <- "midwin"
names(pi.maqu)[14] <- "nsites"

hist(pi.maqu$nsites, breaks = 100)
pi.maqu.filt <- na.omit(filter(pi.maqu, nsites >= 50))
pi.maqu.filt <- dplyr::filter(pi.maqu.filt, grepl('Mme_', chromo)) 

hist((pi.maqu$tP/pi.maqu$nsites), breaks = 100)
pi.maqu.filt$pi <- pi.maqu.filt$tP/pi.maqu.filt$nsites
mean(pi.maqu.filt$pi)

#Plot results with highlighted outliers and relativ positions using ggplot:
maqu.data <- pi.maqu.filt %>% 
  
  # Compute chromosome size
  dplyr::group_by(chromo) %>% 
  dplyr::summarise(chr_len=max(midwin)) %>% 
  
  # Calculate cumulative position of each chromosome
  dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  dplyr::left_join(pi.maqu.filt, ., by=c("chromo"="chromo")) %>%
  
  # Add a cumulative position of each SNP
  dplyr::arrange(chromo, midwin) %>%
  dplyr::mutate(BPcum=midwin+tot)
axisdf = maqu.data %>% dplyr::group_by(chromo) %>% dplyr::summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

maqu.pi <- ggplot(maqu.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))

maqu.pi.smooth <- ggplot(maqu.data, aes(x=BPcum, y=(tP/nsites))) +
  geom_smooth(aes(color=as.factor(chromo)), method = "loess", se = F, span = 0.2) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  ylab("π") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 0.05) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r, eval=F}
plot_grid(maqu.pi.smooth, MBNS.pi.smooth, pany.pi.smooth, jiga.pi.smooth, align = "h", labels = "auto", ncol = 1)
ggsave2(filename = "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/PI_genomescan_JIGA_PANY_MBNS_MAQU_gridplot.pdf", width = 15, height = 10)

```

Get mean π for all other pops:
```{r, eval= F}
pi.lrsc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/LRSC_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.lrsc)[1] <- "region"
names(pi.lrsc)[2] <- "chromo"
names(pi.lrsc)[3] <- "midwin"
names(pi.lrsc)[14] <- "nsites"
pi.lrsc.filt <- na.omit(filter(pi.lrsc, nsites >= 50))
pi.lrsc.filt <- dplyr::filter(pi.lrsc.filt, grepl('Mme_', chromo)) 
pi.lrsc.filt$pi <- pi.lrsc.filt$tP/pi.lrsc.filt$nsites
mean(pi.lrsc.filt$pi)

pi.mcnc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/mcnc_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.mcnc)[1] <- "region"
names(pi.mcnc)[2] <- "chromo"
names(pi.mcnc)[3] <- "midwin"
names(pi.mcnc)[14] <- "nsites"
pi.mcnc.filt <- na.omit(filter(pi.mcnc, nsites >= 50))
pi.mcnc.filt <- dplyr::filter(pi.mcnc.filt, grepl('Mme_', chromo)) 
pi.mcnc.filt$pi <- pi.mcnc.filt$tP/pi.mcnc.filt$nsites
mean(pi.mcnc.filt$pi)

pi.oinc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/oinc_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.oinc)[1] <- "region"
names(pi.oinc)[2] <- "chromo"
names(pi.oinc)[3] <- "midwin"
names(pi.oinc)[14] <- "nsites"
pi.oinc.filt <- na.omit(filter(pi.oinc, nsites >= 50))
pi.oinc.filt <- dplyr::filter(pi.oinc.filt, grepl('Mme_', chromo)) 
pi.oinc.filt$pi <- pi.oinc.filt$tP/pi.oinc.filt$nsites
mean(pi.oinc.filt$pi)

pi.ctva <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/ctva_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.ctva)[1] <- "region"
names(pi.ctva)[2] <- "chromo"
names(pi.ctva)[3] <- "midwin"
names(pi.ctva)[14] <- "nsites"
pi.ctva.filt <- na.omit(filter(pi.ctva, nsites >= 50))
pi.ctva.filt <- dplyr::filter(pi.ctva.filt, grepl('Mme_', chromo)) 
pi.ctva.filt$pi <- pi.ctva.filt$tP/pi.ctva.filt$nsites
mean(pi.ctva.filt$pi)

pi.mcct <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MCCT_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.mcct)[1] <- "region"
names(pi.mcct)[2] <- "chromo"
names(pi.mcct)[3] <- "midwin"
names(pi.mcct)[14] <- "nsites"
pi.mcct.filt <- na.omit(filter(pi.mcct, nsites >= 50))
pi.mcct.filt <- dplyr::filter(pi.mcct.filt, grepl('Mme_', chromo)) 
pi.mcct.filt$pi <- pi.mcct.filt$tP/pi.mcct.filt$nsites
mean(pi.mcct.filt$pi)


pi.dxma <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/dxma_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.dxma)[1] <- "region"
names(pi.dxma)[2] <- "chromo"
names(pi.dxma)[3] <- "midwin"
names(pi.dxma)[14] <- "nsites"
pi.dxma.filt <- na.omit(filter(pi.dxma, nsites >= 50))
pi.dxma.filt <- dplyr::filter(pi.dxma.filt, grepl('Mme_', chromo)) 
pi.dxma.filt$pi <- pi.dxma.filt$tP/pi.dxma.filt$nsites
mean(pi.dxma.filt$pi)

pi.mdme <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/mdme_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.mdme)[1] <- "region"
names(pi.mdme)[2] <- "chromo"
names(pi.mdme)[3] <- "midwin"
names(pi.mdme)[14] <- "nsites"
pi.mdme.filt <- na.omit(filter(pi.mdme, nsites >= 50))
pi.mdme.filt <- dplyr::filter(pi.mdme.filt, grepl('Mme_', chromo)) 
pi.mdme.filt$pi <- pi.mdme.filt$tP/pi.mdme.filt$nsites
mean(pi.mdme.filt$pi)

pi.hxns <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/hxns_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.hxns)[1] <- "region"
names(pi.hxns)[2] <- "chromo"
names(pi.hxns)[3] <- "midwin"
names(pi.hxns)[14] <- "nsites"
pi.hxns.filt <- na.omit(filter(pi.hxns, nsites >= 50))
pi.hxns.filt <- dplyr::filter(pi.hxns.filt, grepl('Mme_', chromo)) 
pi.hxns.filt$pi <- pi.hxns.filt$tP/pi.hxns.filt$nsites
mean(pi.hxns.filt$pi)
```



Create MAF dataframe for all Atlantic silverside populations for MAF-based analyses:
```{r, eval=F}
setwd("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/")

MAF.df <- cbind(JIGA.maf, LRSC.maf, MCNC.maf, OINC.maf, CTVA.maf,PANY.maf,DXMA.maf,MDME.maf,MBNS.maf,HXNS.maf,MAQU.maf)
str(MAF.df)

#Create a combined dataframe with all populations:
MAF.df <- MAF.df[,c(1,2,3,4,5,6,13,20,27,34,41,48,55,62,69,76)]
str(MAF.df)
names(MAF.df)[6] <- "JIGA.maf"
names(MAF.df)[7] <- "LRSC.maf"
names(MAF.df)[8] <- "MCNC.maf"
names(MAF.df)[9] <- "OINC.maf"
names(MAF.df)[10] <- "CTVA.maf"
names(MAF.df)[11] <- "PANY.maf"
names(MAF.df)[12] <- "DXMA.maf"
names(MAF.df)[13] <- "MDME.maf"
names(MAF.df)[14] <- "MBNS.maf"
names(MAF.df)[15] <- "HXNS.maf"
names(MAF.df)[16] <- "MAQU.maf"
MAF.df <- as_tibble(MAF.df)

write_delim(MAF.df, "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_inputfiles/mme_filtsnps_global_maf_df.txt", delim = "\t", col_names = T, quote_escape = F)
```


## Estimate Tajima's D and π in 10kb non-overlapping windows based on the GATK-GL SAF and folded SFS:
```{bash, eval=F}
#JIGA
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt JIGA /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/JIGA_allsites_gatk_folded.saf.idx /workdir/arne/results/summary_statistics/ _gatk_allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/jiga_gatk_thetastats_folded_nohup.log & 

#PANY
nohup sh ./angsd_thetastats_win.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt PANY /workdir/arne/ReferenceSeq/Mmenidia_refgenome_anchored.all_renamed_v2.fasta /workdir/arne/results/summary_statistics/PANY_allsites_gatk_folded.saf.idx /workdir/arne/results/summary_statistics/ _gatk_allsites_thetastats 10000 10000 > /workdir/arne/output_logfiles/PANY_gatk_thetastats_folded_nohup.log & 
```


Load theta stats for Samtools and GATK model and compare:
```{r, eval= F}
pi.jiga <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/JIGAallsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.jiga)[1] <- "region"
names(pi.jiga)[2] <- "chromo"
names(pi.jiga)[3] <- "midwin"
names(pi.jiga)[14] <- "nsites"
pi.jiga.filt <- na.omit(filter(pi.jiga, nsites >= 50))
pi.jiga.filt <- dplyr::filter(pi.jiga.filt, grepl('Mme_', chromo)) 
pi.jiga.filt$pi <- pi.jiga.filt$tP/pi.jiga.filt$nsites
jiga.theta <- mean((pi.jiga.filt$tW/pi.jiga.filt$nsites))
jiga.pi <- mean(pi.jiga.filt$pi)
jiga.TD <- mean(pi.jiga.filt$Tajima)

pi.jiga.gatk <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/JIGA_gatk_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.jiga.gatk)[1] <- "region"
names(pi.jiga.gatk)[2] <- "chromo"
names(pi.jiga.gatk)[3] <- "midwin"
names(pi.jiga.gatk)[14] <- "nsites"
pi.jiga.gatk.filt <- na.omit(filter(pi.jiga.gatk, nsites >= 50))
pi.jiga.gatk.filt <- dplyr::filter(pi.jiga.gatk.filt, grepl('Mme_', chromo)) 
pi.jiga.gatk.filt$pi <- pi.jiga.gatk.filt$tP/pi.jiga.gatk.filt$nsites
jiga.gatk.theta <- mean((pi.jiga.gatk.filt$tW/pi.jiga.gatk.filt$nsites))
jiga.gatk.pi <- mean(pi.jiga.gatk.filt$pi)
jiga.gatk.TD <- mean(pi.jiga.gatk.filt$Tajima)


pi.pany <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/PANYallsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.pany)[1] <- "region"
names(pi.pany)[2] <- "chromo"
names(pi.pany)[3] <- "midwin"
names(pi.pany)[14] <- "nsites"
pi.pany.filt <- na.omit(filter(pi.pany, nsites >= 50))
pi.pany.filt <- dplyr::filter(pi.pany.filt, grepl('Mme_', chromo)) 
pi.pany.filt$pi <- pi.pany.filt$tP/pi.pany.filt$nsites
pany.theta <- mean((pi.pany.filt$tW/pi.pany.filt$nsites))
pany.pi <- mean(pi.pany.filt$pi)
pany.TD <- mean(pi.pany.filt$Tajima)

pi.pany.gatk <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/PANY_gatk_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
names(pi.pany.gatk)[1] <- "region"
names(pi.pany.gatk)[2] <- "chromo"
names(pi.pany.gatk)[3] <- "midwin"
names(pi.pany.gatk)[14] <- "nsites"
pi.pany.gatk.filt <- na.omit(filter(pi.pany.gatk, nsites >= 50))
pi.pany.gatk.filt <- dplyr::filter(pi.pany.gatk.filt, grepl('Mme_', chromo)) 
pi.pany.gatk.filt$pi <- pi.pany.gatk.filt$tP/pi.pany.gatk.filt$nsites
pany.gatk.theta <- mean((pi.pany.gatk.filt$tW/pi.pany.gatk.filt$nsites))
pany.gatk.pi <- mean(pi.pany.gatk.filt$pi)
pany.gatk.TD <- mean(pi.pany.gatk.filt$Tajima)
```


Correlation between genome-wide distributions of pi, TD and theta in JIGA:
```{r, eval= F}
JIGA.merged = as_tibble(merge(pi.jiga.filt, pi.jiga.gatk.filt, by=c("chromo","midwin")))

summary(lm(pi.x ~ pi.y, data = JIGA.merged))
ggplot(data = JIGA.merged, aes(x= pi.x, y=pi.y)) +
  geom_point()

summary(lm((tW.x/nsites.x) ~ (tW.y/nsites.y), data = JIGA.merged))
ggplot(data = JIGA.merged, aes(x= (tW.x/nsites.x), y=(tW.y/nsites.y))) +
  geom_point()

summary(lm(Tajima.x ~ Tajima.y, data = JIGA.merged))
ggplot(data = JIGA.merged, aes(x= Tajima.x, y=Tajima.y)) +
  geom_point()
```

Correlation between genome-wide distributions of pi, TD and theta in PANY:
```{r, eval= F}
PANY.merged = as_tibble(merge(pi.pany.filt, pi.pany.gatk.filt, by=c("chromo","midwin")))

summary(lm(pi.x ~ pi.y, data = PANY.merged))
ggplot(data = PANY.merged, aes(x= pi.x, y=pi.y)) +
  geom_point()

summary(lm((tW.x/nsites.x) ~ (tW.y/nsites.y), data = PANY.merged))
ggplot(data = PANY.merged, aes(x= (tW.x/nsites.x), y=(tW.y/nsites.y))) +
  geom_point()

summary(lm(Tajima.x ~ Tajima.y, data = PANY.merged))
ggplot(data = PANY.merged, aes(x= Tajima.x, y=Tajima.y)) +
  geom_point()
```





******************************************************************************************************************

Estimate unfolded saf for all sites using ancestral outgroup genome (for estimating Fay's H)
```{bash, eval=F}
nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt JIGA /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/summary_statistics/ _allsites_unfolded 24 16 200 /workdir/arne/sampleinfo/mme_chromosomes_regionfile.txt > /workdir/arne/output_logfiles/JIGA_unfolded_sfs_allsites_nohup.log &

#nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_LRSC.txt LRSC #/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa #/workdir/arne/results/summary_statistics/ _allsites_unfolded 25 16 200 > #/workdir/arne/output_logfiles/LRSC_unfolded_sfs_allsites_nohup.log &

#nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCNC.txt MCNC #/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa #/workdir/arne/results/summary_statistics/ _allsites_unfolded 25 16 200 > #/workdir/arne/output_logfiles/MCNC_unfolded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_OINC.txt OINC /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/summary_statistics/ _allsites_unfolded 24 16 200 /workdir/arne/sampleinfo/mme_chromosomes_regionfile.txt > /workdir/arne/output_logfiles/OINC_unfolded_sfs_allsites_nohup.log &

#nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_CTVA.txt CTVA #/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa #/workdir/arne/results/summary_statistics/ _allsites_unfolded 23 15 200 > #/workdir/arne/output_logfiles/CTVA_unfolded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt PANY /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/summary_statistics/ _allsites_unfolded 24 16 200 /workdir/arne/sampleinfo/mme_chromosomes_regionfile.txt > /workdir/arne/output_logfiles/pany_unfolded_sfs_allsites_nohup.log &

#nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCCT.txt MCCT #/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa #/workdir/arne/results/summary_statistics/ _allsites_unfolded 25 16 200 > #/workdir/arne/output_logfiles/MCCT_unfolded_sfs_allsites_nohup.log &
#
#nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_DXMA.txt DXMA #/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa #/workdir/arne/results/summary_statistics/ _allsites_unfolded 25 16 200 > #/workdir/arne/output_logfiles/DXMA_unfolded_sfs_allsites_nohup.log &

#nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MDME.txt MDME #/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa #/workdir/arne/results/summary_statistics/ _allsites_unfolded 25 16 200 > #/workdir/arne/output_logfiles/MDME_unfolded_sfs_allsites_nohup.log &

nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MBNS.txt MBNS /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/summary_statistics/ _allsites_unfolded 25 16 200 /workdir/arne/sampleinfo/mme_chromosomes_regionfile.txt > /workdir/arne/output_logfiles/MBNS_unfolded_sfs_allsites_nohup.log &

#nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_HXNS.txt HXNS #/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa #/workdir/arne/results/summary_statistics/ _allsites_unfolded 23 15 200 > #/workdir/arne/output_logfiles/HXNS_unfolded_sfs_allsites_nohup.log &
#
nohup sh ./angsd_sfs_allsites_unfolded.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MAQU.txt MAQU /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/summary_statistics/ _allsites_unfolded 21 14 200 /workdir/arne/sampleinfo/mme_chromosomes_regionfile.txt > /workdir/arne/output_logfiles/MAQU_unfolded_sfs_allsites_nohup.log &
```

Estimate theta stats in 10kb non-overlapping windows based on the unfolded SAF/SFS of all sites:
```{bash, eval=F}
nohup sh ./angsd_thetastats_unfolded_win_wsfs.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt JIGA /workdir/arne/results/summary_statistics/JIGA_allsites_unfolded.sfs /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/summary_statistics/ _unfolded_thetastats 10000 10000 24 16 200 /workdir/arne/sampleinfo/mme_chromosomes_regionfile.txt > /workdir/arne/output_logfiles/JIGA_thetastats_unfolded_nohup.log & 

nohup sh ./angsd_thetastats_unfolded_win_wsfs.sh \
/workdir/arne/sampleinfo/BamFiles_clean_list_MAQU.txt \
MAQU \
/workdir/arne/results/summary_statistics/MAQU_allsites_unfolded.sfs \
/workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa \
/workdir/arne/results/summary_statistics/ \
_unfolded_thetastats \
10000 10000 24 16 200 \
/workdir/arne/sampleinfo/mme_chromosomes_regionfile.txt \
> /workdir/arne/output_logfiles/MAQU_thetastats_unfolded_nohup.log &
```



















******************************************************************************************************************

# Estimating population genetic differentiation and divergence

## Absolute allele frequency difference (AFD):
```{r, eval=F}
#Read MAF dataframe:
MAF.df <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/allele_frequency_inputfiles/mme_filtsnps_global_maf_df.txt", delim = "\t", col_names = T)

#Calculate AFD for each population pair:
AFD <- function(x,y) ((abs(x-y) + abs((1-x)-(1-y)) )/2)

MAF.df$AFD.MAQU.JIGA <- AFD(MAF.df$MAQU.maf,MAF.df$JIGA.maf)
MAF.df$AFD.MBNS.JIGA <- AFD(MAF.df$MBNS.maf,MAF.df$JIGA.maf)
MAF.df$AFD.HXNS.JIGA <- AFD(MAF.df$HXNS.maf,MAF.df$JIGA.maf)
MAF.df$AFD.MDME.JIGA <- AFD(MAF.df$MDME.maf,MAF.df$JIGA.maf)
MAF.df$AFD.DXMA.JIGA <- AFD(MAF.df$DXMA.maf,MAF.df$JIGA.maf)
MAF.df$AFD.PANY.JIGA <- AFD(MAF.df$PANY.maf,MAF.df$JIGA.maf)
MAF.df$AFD.CTVA.JIGA <- AFD(MAF.df$CTVA.maf,MAF.df$JIGA.maf)
MAF.df$AFD.OINC.JIGA <- AFD(MAF.df$OINC.maf,MAF.df$JIGA.maf)
MAF.df$AFD.MCNC.JIGA <- AFD(MAF.df$MCNC.maf,MAF.df$JIGA.maf)
MAF.df$AFD.LRSC.JIGA <- AFD(MAF.df$LRSC.maf,MAF.df$JIGA.maf)
MAF.df$AFD.MAQU.MBNS <- AFD(MAF.df$MAQU.maf,MAF.df$MBNS.maf)

#Plot along genome in 10kb windows
MAF.df.chr <- dplyr::filter(MAF.df, grepl('Mme_', chromo)) 

#Summarize in 10kb windows:
pos_AFD_win <- winScan(x = MAF.df.chr, 
					 groups = "chromo", 
					 position = "position", 
					 values = c("AFD.MAQU.MBNS","AFD.MAQU.JIGA",
					            "AFD.MBNS.JIGA","AFD.HXNS.JIGA",
					       "AFD.MDME.JIGA","AFD.DXMA.JIGA","AFD.PANY.JIGA",
					       "AFD.CTVA.JIGA","AFD.OINC.JIGA","AFD.MCNC.JIGA",
					            "AFD.LRSC.JIGA"), 
					 win_size = 10000,
					 win_step = 10000,
					 funs = c("mean"))
pos_AFD_win_filt <- filter(pos_AFD_win, AFD.MAQU.JIGA_n >= 15)
```


## Estimate pairwise average genetic differentiation (Fst):
```{bash, eval=F}
nohup sh get_fst.sh /workdir/arne/results/snp_datasets/ /workdir/arne/sampleinfo/SampleTable_mergedBams_mme.txt 5 _fst_filtsnps > /workdir/arne/output_logfiles/mme_pairwise_fst_nohup.log &
```


### Create neutral SNP list

Select SNPs below 90th percentile of Fst distribution in each population comparison (run this script in RStudio Server):
```{r, eval=F}
#Function to select 'non-differentiated' SNPs
non_diff.snps <- function(df) {
  df.filt <- filter(df, df$X5 < quantile(df$X5, prob = 1 - 10/100, na.rm = TRUE))
  df.filt <- filter(df.filt, df.filt$X5 > 0)
  return(df.filt[,c(1,2)])
}

files <- list.files(path="/workdir/arne/results/summary_statistics/snp_fst/",pattern="*.fst", full.names=TRUE, recursive=FALSE)
out <- (lapply(files, function(x) {
    df <- read_delim(x, delim = "\t", col_names = F, col_types = list(X1 = col_character())) # load file
    # apply function
    non_diff.snps(df)
}))
neutral.df <- out %>% reduce(left_join, by = c("X1", "X2"))
write_delim(out, "/workdir/arne/results/summary_statistics/snp_fst/neutralFst_df_allcomb.txt", delim="\t")
```

Select list of intergenic SNPs from snpEff annotation
```{bash, eval=F}
nohup java -Xmx8g -jar /programs/snpEff-4.3/SnpSift.jar filter "ANN[*].EFFECT has 'intergenic_region'" /workdir/arne/results/snp_datasets/mme_filtSNPs.ann.vcf > /workdir/arne/results/snp_datasets/mme_filtSNPs.ann.intergenic.vcf &
```

Create list of non-differentiated intragenic SNPs (within and outside inversions):
```{r, eval= F}
snps.intergenic <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/mme_filtSNPs.ann.intergenic.snpdf.txt", delim = "\t", col_names = T)

snps.neutral <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/neutralFst_df_allcomb.txt", delim = "\t", col_names = T, col_types = list(CHROM = col_character()))

snp.neutral.intergenic=inner_join(snps.neutral, snps.intergenic, by = c("CHROM","POS"))
write_delim(snp.neutral.intergenic, "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/mme_filtSNPs_neutral_intergenic_snpdf.txt", delim = "\t", col_names = T)
```

Create list of non-differentiated intragenic SNPs outside inversions:
```{r, eval= F}
snps.neutral.filt1 <- filter(snps.neutral, CHROM != "Mme_chr04",CHROM != "Mme_chr08", CHROM != "Mme_chr18", CHROM != "Mme_chr24", CHROM != "Mme_chr07", CHROM != "Mme_chr19", CHROM != "Mme_chr11")
snps.neutral.filt.chr04 <- filter(snps.neutral, CHROM == "Mme_chr04", POS > 14136773 | POS < 12104495)
snps.neutral.filt.chr07 <- filter(snps.neutral, CHROM == "Mme_chr07", POS > 14273829 | POS < 12447777)
snps.neutral.filt.chr08 <- filter(snps.neutral, CHROM == "Mme_chr08", POS > 15747066 | POS < 2727314)
snps.neutral.filt.chr11 <- filter(snps.neutral, CHROM == "Mme_chr11", POS > 10729607 | POS < 3622131)
snps.neutral.filt.chr18 <- filter(snps.neutral, CHROM == "Mme_chr18", POS > 8880140 | POS < 723668)
snps.neutral.filt.chr19 <- filter(snps.neutral, CHROM == "Mme_chr18", POS > 3486888 | POS < 2220950)
snps.neutral.filt.chr24 <- filter(snps.neutral, CHROM == "Mme_chr24", POS > 903817 | POS < 92880 | POS > 13065337 | POS < 3166287)
snps.neutral.noinv <- bind_rows(snps.neutral.filt1, snps.neutral.filt.chr04, snps.neutral.filt.chr07, snps.neutral.filt.chr08, snps.neutral.filt.chr11, snps.neutral.filt.chr18, snps.neutral.filt.chr19, snps.neutral.filt.chr24)

snp.neutral.intergenic.noinv=inner_join(snps.neutral.noinv, snps.intergenic, by = c("CHROM","POS"))

write_delim(snp.neutral.intergenic.noinv, "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/mme_filtSNPs_neutral_intergenic_noinv_snpdf.txt", delim = "\t", col_names = T)
```



### Estimate pairwise genome-wide Fst for neutral SNP data:

With inversions
```{r, eval=F}
#Function to select 'non-differentiated' SNPs
neutFst <- function(fst.df, snp.df) {
  fst.df.filt=inner_join(fst.df, snp.df, by=c("X1","X2"))
  sum.A=sum(fst.df.filt$X3)
  sum.B=sum(fst.df.filt$X4)
  fst=sum.A/sum.B
  return(fst)
}

snp.df <- read_delim("/workdir/arne/results/summary_statistics/snp_fst/mme_filtSNPs_neutral_intergenic_snpdf.txt", delim = "\t", col_names = F, col_types = list(X1 = col_character()))

files <- list.files(path="/workdir/arne/results/summary_statistics/snp_fst/",pattern="*.fst", full.names=TRUE, recursive=FALSE)
out <- (lapply(files, function(x) {
    fst.df <- read_delim(x, delim = "\t", col_names = F, col_types = list(X1 = col_character())) # load file
    # apply function
    neutFst(fst.df, snp.df)
}))
neutfst.df=as_tibble(plyr::ldply(out, data.frame))
files=as_tibble(files)
neutfst.df.names=as_tibble(cbind(files,neutfst.df))
names(neutfst.df.names)=c("comp","fst")
write_delim(neutfst.df.names, "/workdir/arne/results/summary_statistics/snp_fst/mme_mean_neutralFst_allcomb_df.txt", delim="\t")
```



Without inversions
```{r, eval=F}
#Function to select 'non-differentiated' SNPs
neutFst <- function(fst.df, snp.df) {
  fst.df.filt=inner_join(fst.df, snp.df, by=c("X1","X2"))
  sum.A=sum(fst.df.filt$X3)
  sum.B=sum(fst.df.filt$X4)
  fst=sum.A/sum.B
  return(fst)
}

snp.df <- read_delim("/workdir/arne/results/summary_statistics/snp_fst/mme_filtSNPs_neutral_intergenic_noinv_snpdf.txt", delim = "\t", col_names = F, col_types = list(X1 = col_character()))

files <- list.files(path="/workdir/arne/results/summary_statistics/snp_fst/",pattern="*.fst", full.names=TRUE, recursive=FALSE)
out <- (lapply(files, function(x) {
    fst.df <- read_delim(x, delim = "\t", col_names = F, col_types = list(X1 = col_character())) # load file
    # apply function
    neutFst(fst.df, snp.df)
}))
neutfst.df=as_tibble(plyr::ldply(out, data.frame))
files=as_tibble(files)
neutfst.df.names=as_tibble(cbind(files,neutfst.df))
names(neutfst.df.names)=c("comp","fst")
write_delim(neutfst.df.names, "/workdir/arne/results/summary_statistics/snp_fst/mme_mean_neutralFst_noinv_allcomb_df.txt", delim="\t")
```

Run Rscripts:
```{bash, eval=F}
nohup Rscript estimate_neutral_pairwiseFst.R &
nohup Rscript estimate_neutral_pairwiseFst_noinv.R &
```


Subset filtered SNP-list to only include neutral SNPs (with and without inversions:)
```{r, eval=F}
mme.filt.snps=read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/snp_calling/mme_global_filtSNP_list.txt", delim = "\t", col_names = F, col_types = list(X1 = col_character()))

neutral.snps=read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/snp_calling/mme_filtSNPs_neutral_intergenic_snpdf.txt", delim = "\t", col_names = F, col_types = list(X1 = col_character()))

neutral.snps.noinv=read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/snp_calling/mme_filtSNPs_neutral_intergenic_noinv_snpdf.txt", delim = "\t", col_names = F, col_types = list(X1 = col_character()))


mme.filt.snps.neutral=left_join(neutral.snps, mme.filt.snps, by=c("X1","X2"))
write_delim(mme.filt.snps.neutral, "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/snp_calling/mme_global_filtSNP_list_neutral.txt", delim="\t", col_names = F)

mme.filt.snps.neutral.noinv=left_join(neutral.snps.noinv, mme.filt.snps, by=c("X1","X2"))
write_delim(mme.filt.snps.neutral.noinv, "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/snp_calling/mme_global_filtSNP_list_neutral_noinv.txt", delim="\t", col_names = F)
```

Index neutral SNPlists
```{bash, eval=F}
nohup /programs/angsd0.920/angsd/angsd sites index /workdir/arne/results/snp_datasets/mme_global_filtSNP_list_neutral.txt > /workdir/arne/output_logfiles/angsd_index_neutSNPlist_nohup.log &

nohup /programs/angsd0.920/angsd/angsd sites index /workdir/arne/results/snp_datasets/mme_global_filtSNP_list_neutral_noinv.txt > /workdir/arne/output_logfiles/angsd_index_neutSNPlist_noinv_nohup.log &
```


### Genetic differentiation (Fst) genomescan in 10kb windows:

```{bash, eval=F}
nohup sh get_winfst_fromfstidx.sh /workdir/arne/results/summary_statistics /workdir/arne/sampleinfo/SampleTable_mergedBams_mme.txt 5 _fst_filtsnps 10000 10000 > /workdir/arne/output_logfiles/mme_pairwise_fst_10kb_win_nohup.log &
```

Plot window-based Fst across the genome:
```{r, eval=F}
qtl <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")

sv.maria <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4)]
names(sv.maria2) = c("seqnames","start","end")
sv.maria.chr.ranges =  sv.maria2 %>% as_granges()

#with SV regions highlighted
sv <- read.csv("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
svs.major2 = svs.major[,c(2,10,11)]
names(svs.major2) = c("seqnames","start","end")
svs.major.ranges =  svs.major2 %>% as_granges()

all.sv = rbind(sv.maria2, svs.major2)
all.sv.ranges =  all.sv %>% as_granges()
```

```{r, eval=F}
#Load genome-scan results:
jiga_lrsc_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_lrsc_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_lrsc_fst_scan)[1] <- "region"
names(jiga_lrsc_fst_scan)[2] <- "chromo"
names(jiga_lrsc_fst_scan)[3] <- "midwin"
names(jiga_lrsc_fst_scan)[4] <- "nsites"
names(jiga_lrsc_fst_scan)[5] <- "fst"

jiga_lrsc_fst_scan <- dplyr::filter(jiga_lrsc_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_lrsc_fst_scan$nsites, breaks = 100)
jiga_lrsc_fst_scan <- filter(jiga_lrsc_fst_scan, nsites >= 30)

#Estimate cutoff for inverted and collinear genome
jiga_lrsc_fst_scan$start = (jiga_lrsc_fst_scan$midwin - 4999)/1000000
jiga_lrsc_fst_scan$end = (jiga_lrsc_fst_scan$midwin + 5000)/1000000
jiga_lrsc_fst_scan2 = jiga_lrsc_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_lrsc_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_lrsc_fst_scan.ranges =  jiga_lrsc_fst_scan2 %>% as_granges()

jiga_lrsc_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_lrsc_fst_scan.ranges, all.sv.ranges))
jiga_lrsc_fst_scan.ranges.coll = as_tibble(jiga_lrsc_fst_scan.ranges[!jiga_lrsc_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_lrsc = quantile(jiga_lrsc_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_lrsc = quantile(jiga_lrsc_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_lrsc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_lrsc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_lrsc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_lrsc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_lrsc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_lrsc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_lrsc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_lrsc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_lrsc_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_lrsc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_lrsc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#plot with cmplot
jiga_lrsc_fst_scan <- unite(jiga_lrsc_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_lrsc_fst_scan.plot <- as.data.frame(jiga_lrsc_fst_scan[,c(2,3,4,6)])
names(jiga_lrsc_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_lrsc_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_lrsc_fst")

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_lrsc <- jiga_lrsc_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_lrsc_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_lrsc %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=0.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 0.11, ymax=0.15), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.05, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_lrsc, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = all.fst.99.jiga_lrsc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_lrsc, colour="orange", linetype="dashed") +
  ylab("Fst") +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_mcnc_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_mcnc_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_mcnc_fst_scan)[1] <- "region"
names(jiga_mcnc_fst_scan)[2] <- "chromo"
names(jiga_mcnc_fst_scan)[3] <- "midwin"
names(jiga_mcnc_fst_scan)[4] <- "nsites"
names(jiga_mcnc_fst_scan)[5] <- "fst"

jiga_mcnc_fst_scan <- dplyr::filter(jiga_mcnc_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_mcnc_fst_scan$nsites, breaks = 100)
jiga_mcnc_fst_scan <- filter(jiga_mcnc_fst_scan, nsites >= 30)

#Estimate cutoff for inverted and collinear genome
jiga_mcnc_fst_scan$start = (jiga_mcnc_fst_scan$midwin - 4999)/1000000
jiga_mcnc_fst_scan$end = (jiga_mcnc_fst_scan$midwin + 5000)/1000000
jiga_mcnc_fst_scan2 = jiga_mcnc_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_mcnc_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_mcnc_fst_scan.ranges =  jiga_mcnc_fst_scan2 %>% as_granges()

jiga_mcnc_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_mcnc_fst_scan.ranges, all.sv.ranges))
jiga_mcnc_fst_scan.ranges.coll = as_tibble(jiga_mcnc_fst_scan.ranges[!jiga_mcnc_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_mcnc = quantile(jiga_mcnc_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_mcnc = quantile(jiga_mcnc_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mcnc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mcnc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_mcnc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mcnc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_mcnc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mcnc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mcnc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mcnc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mcnc_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mcnc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mcnc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#plot with cmplot
jiga_mcnc_fst_scan <- unite(jiga_mcnc_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_mcnc_fst_scan.plot <- as.data.frame(jiga_mcnc_fst_scan[,c(2,3,4,6)])
names(jiga_mcnc_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_mcnc_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_mcnc_fst")

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_mcnc <- jiga_mcnc_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_mcnc_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_mcnc %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=0.5), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.55), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.05, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_mcnc, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_mcnc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mcnc, colour="orange", linetype="dashed") +
  ylab("Fst") +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_oinc_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_oinc_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

#plot with cmplot
jiga_oinc_fst_scan <- unite(jiga_oinc_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_oinc_fst_scan.plot <- as.data.frame(jiga_oinc_fst_scan[,c(2,3,4,6)])
names(jiga_oinc_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_oinc_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_oinc_fst")


names(jiga_oinc_fst_scan)[1] <- "region"
names(jiga_oinc_fst_scan)[2] <- "chromo"
names(jiga_oinc_fst_scan)[3] <- "midwin"
names(jiga_oinc_fst_scan)[4] <- "nsites"
names(jiga_oinc_fst_scan)[5] <- "fst"

jiga_oinc_fst_scan <- dplyr::filter(jiga_oinc_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_oinc_fst_scan$nsites, breaks = 100)
jiga_oinc_fst_scan <- filter(jiga_oinc_fst_scan, nsites >= 30)

#Estimate cutoff for inverted and collinear genome
jiga_oinc_fst_scan$start = (jiga_oinc_fst_scan$midwin - 4999)/1000000
jiga_oinc_fst_scan$end = (jiga_oinc_fst_scan$midwin + 5000)/1000000
jiga_oinc_fst_scan2 = jiga_oinc_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_oinc_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_oinc_fst_scan.ranges =  jiga_oinc_fst_scan2 %>% as_granges()

jiga_oinc_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_oinc_fst_scan.ranges, all.sv.ranges))
jiga_oinc_fst_scan.ranges.coll = as_tibble(jiga_oinc_fst_scan.ranges[!jiga_oinc_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_oinc = quantile(jiga_oinc_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_oinc = quantile(jiga_oinc_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_oinc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_oinc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_oinc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_oinc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_oinc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_oinc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_oinc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_oinc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_oinc_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_oinc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_oinc, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_oinc <- jiga_oinc_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_oinc_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_oinc %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_oinc, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_oinc, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_oinc, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_ctva_fst_scan <- read_delim("~/Library/CloudStorage//Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/CTVA_JIGA_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)


names(jiga_ctva_fst_scan)[1] <- "region"
names(jiga_ctva_fst_scan)[2] <- "chromo"
names(jiga_ctva_fst_scan)[3] <- "midwin"
names(jiga_ctva_fst_scan)[4] <- "nsites"
names(jiga_ctva_fst_scan)[5] <- "fst"

jiga_ctva_fst_scan <- dplyr::filter(jiga_ctva_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_ctva_fst_scan$nsites, breaks = 100)
jiga_ctva_fst_scan <- filter(jiga_ctva_fst_scan, nsites >= 30)

#plot with cmplot
jiga_ctva_fst_scan <- unite(jiga_ctva_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_ctva_fst_scan.plot <- as.data.frame(jiga_ctva_fst_scan[,c(2,3,4,6)])
names(jiga_ctva_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_ctva_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_ctva_fst")


#Estimate cutoff for inverted and collinear genome
jiga_ctva_fst_scan$start = (jiga_ctva_fst_scan$midwin - 4999)/1000000
jiga_ctva_fst_scan$end = (jiga_ctva_fst_scan$midwin + 5000)/1000000
jiga_ctva_fst_scan2 = jiga_ctva_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_ctva_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_ctva_fst_scan.ranges =  jiga_ctva_fst_scan2 %>% as_granges()

jiga_ctva_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_ctva_fst_scan.ranges, all.sv.ranges))
jiga_ctva_fst_scan.ranges.coll = as_tibble(jiga_ctva_fst_scan.ranges[!jiga_ctva_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_ctva = quantile(jiga_ctva_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_ctva = quantile(jiga_ctva_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_ctva, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_ctva, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_ctva, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_ctva, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_ctva, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_ctva, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_ctva, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_ctva, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_ctva_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_ctva, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_ctva, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_ctva <- jiga_ctva_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_ctva_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_ctva %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_ctva, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_ctva, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_ctva, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_pany_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_pany_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_pany_fst_scan)[1] <- "region"
names(jiga_pany_fst_scan)[2] <- "chromo"
names(jiga_pany_fst_scan)[3] <- "midwin"
names(jiga_pany_fst_scan)[4] <- "nsites"
names(jiga_pany_fst_scan)[5] <- "fst"

jiga_pany_fst_scan <- dplyr::filter(jiga_pany_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_pany_fst_scan$nsites, breaks = 100)
jiga_pany_fst_scan <- filter(jiga_pany_fst_scan, nsites >= 30)


#plot with cmplot
jiga_pany_fst_scan <- unite(jiga_pany_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_pany_fst_scan.plot <- as.data.frame(jiga_pany_fst_scan[,c(2,3,4,6)])
names(jiga_pany_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_pany_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_pany_fst")


#Estimate cutoff for inverted and collinear genome
jiga_pany_fst_scan$start = (jiga_pany_fst_scan$midwin - 4999)/1000000
jiga_pany_fst_scan$end = (jiga_pany_fst_scan$midwin + 5000)/1000000
jiga_pany_fst_scan2 = jiga_pany_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_pany_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_pany_fst_scan.ranges =  jiga_pany_fst_scan2 %>% as_granges()

jiga_pany_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_pany_fst_scan.ranges, all.sv.ranges))
jiga_pany_fst_scan.ranges.coll = as_tibble(jiga_pany_fst_scan.ranges[!jiga_pany_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_pany = quantile(jiga_pany_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_pany = quantile(jiga_pany_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_pany, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_pany, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_pany, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_pany, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_pany, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_pany, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
    geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1), fill = "steelblue", colour="grey20", alpha = 0.3) +
  geom_smooth(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, se = F, colour = "goldenrod") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

ggplot() +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
    geom_rect(aes(xmin=(791056/1000000), xmax=(1607828/1000000), ymin = 0, ymax=1), fill = "steelblue", colour="grey20", alpha = 0.3) +
     geom_rect(aes(xmin=(1607828/1000000), xmax=(4345655/1000000), ymin = 0, ymax=1), fill = "steelblue", colour="grey20", alpha = 0.3) +
    geom_rect(aes(xmin=(4345655/1000000), xmax=(8508524/1000000), ymin = 0, ymax=1), fill = "steelblue", colour="grey20", alpha = 0.3) +
  geom_smooth(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, se = F, colour = "goldenrod") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_pany, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_pany, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_pany <- jiga_pany_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_pany_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_pany %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_pany, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_pany, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_pany, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_mcct_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_mcct_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)


names(jiga_mcct_fst_scan)[1] <- "region"
names(jiga_mcct_fst_scan)[2] <- "chromo"
names(jiga_mcct_fst_scan)[3] <- "midwin"
names(jiga_mcct_fst_scan)[4] <- "nsites"
names(jiga_mcct_fst_scan)[5] <- "fst"

jiga_mcct_fst_scan <- dplyr::filter(jiga_mcct_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_mcct_fst_scan$nsites, breaks = 100)
jiga_mcct_fst_scan <- filter(jiga_mcct_fst_scan, nsites >= 30)

#plot with cmplot
jiga_mcct_fst_scan <- unite(jiga_mcct_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_mcct_fst_scan.plot <- as.data.frame(jiga_mcct_fst_scan[,c(2,3,4,6)])
names(jiga_mcct_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_mcct_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_mcct_fst")
```


```{r, eval=F}
#Load genome-scan results:
jiga_dxma_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/DXMA_JIGA_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_dxma_fst_scan)[1] <- "region"
names(jiga_dxma_fst_scan)[2] <- "chromo"
names(jiga_dxma_fst_scan)[3] <- "midwin"
names(jiga_dxma_fst_scan)[4] <- "nsites"
names(jiga_dxma_fst_scan)[5] <- "fst"

jiga_dxma_fst_scan <- dplyr::filter(jiga_dxma_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_dxma_fst_scan$nsites, breaks = 100)
jiga_dxma_fst_scan <- filter(jiga_dxma_fst_scan, nsites >= 30)


#plot with cmplot
jiga_dxma_fst_scan <- unite(jiga_dxma_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_dxma_fst_scan.plot <- as.data.frame(jiga_dxma_fst_scan[,c(2,3,4,6)])
names(jiga_dxma_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_dxma_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_dxma_fst")

#Estimate cutoff for inverted and collinear genome
jiga_dxma_fst_scan$start = (jiga_dxma_fst_scan$midwin - 4999)/1000000
jiga_dxma_fst_scan$end = (jiga_dxma_fst_scan$midwin + 5000)/1000000
jiga_dxma_fst_scan2 = jiga_dxma_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_dxma_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_dxma_fst_scan.ranges =  jiga_dxma_fst_scan2 %>% as_granges()

jiga_dxma_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_dxma_fst_scan.ranges, all.sv.ranges))
jiga_dxma_fst_scan.ranges.coll = as_tibble(jiga_dxma_fst_scan.ranges[!jiga_dxma_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_dxma = quantile(jiga_dxma_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_dxma = quantile(jiga_dxma_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_dxma, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_dxma, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_dxma, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_dxma, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_dxma, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_dxma, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_dxma, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_dxma, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_dxma_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_dxma, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_dxma, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_dxma <- jiga_dxma_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_dxma_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_dxma %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_dxma, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_dxma, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_dxma, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_mdme_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_mdme_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_mdme_fst_scan)[1] <- "region"
names(jiga_mdme_fst_scan)[2] <- "chromo"
names(jiga_mdme_fst_scan)[3] <- "midwin"
names(jiga_mdme_fst_scan)[4] <- "nsites"
names(jiga_mdme_fst_scan)[5] <- "fst"

jiga_mdme_fst_scan <- dplyr::filter(jiga_mdme_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_mdme_fst_scan$nsites, breaks = 100)
jiga_mdme_fst_scan <- filter(jiga_mdme_fst_scan, nsites >= 30)

#plot with cmplot
jiga_mdme_fst_scan <- unite(jiga_mdme_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_mdme_fst_scan.plot <- as.data.frame(jiga_mdme_fst_scan[,c(2,3,4,6)])
names(jiga_mdme_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_mdme_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_mdme_fst")


#Estimate cutoff for inverted and collinear genome
jiga_mdme_fst_scan$start = (jiga_mdme_fst_scan$midwin - 4999)/1000000
jiga_mdme_fst_scan$end = (jiga_mdme_fst_scan$midwin + 5000)/1000000
jiga_mdme_fst_scan2 = jiga_mdme_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_mdme_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_mdme_fst_scan.ranges =  jiga_mdme_fst_scan2 %>% as_granges()

jiga_mdme_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_mdme_fst_scan.ranges, all.sv.ranges))
jiga_mdme_fst_scan.ranges.coll = as_tibble(jiga_mdme_fst_scan.ranges[!jiga_mdme_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_mdme = quantile(jiga_mdme_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_mdme = quantile(jiga_mdme_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mdme, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mdme, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_mdme, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mdme, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_mdme, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mdme, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mdme, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mdme, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mdme_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mdme, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mdme, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_mdme <- jiga_mdme_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_mdme_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_mdme %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_mdme, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_mdme, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mdme, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_mbns_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MBNS_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_mbns_fst_scan)[1] <- "region"
names(jiga_mbns_fst_scan)[2] <- "chromo"
names(jiga_mbns_fst_scan)[3] <- "midwin"
names(jiga_mbns_fst_scan)[4] <- "nsites"
names(jiga_mbns_fst_scan)[5] <- "fst"


jiga_mbns_fst_scan <- dplyr::filter(jiga_mbns_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_mbns_fst_scan$nsites, breaks = 100)
jiga_mbns_fst_scan <- filter(jiga_mbns_fst_scan, nsites >= 30)

#plot with cmplot
jiga_mbns_fst_scan <- unite(jiga_mbns_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_mbns_fst_scan.plot <- as.data.frame(jiga_mbns_fst_scan[,c(2,3,4,6)])
names(jiga_mbns_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_mbns_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_mbns_fst")

#Estimate cutoff for inverted and collinear genome
jiga_mbns_fst_scan$start = (jiga_mbns_fst_scan$midwin - 4999)/1000000
jiga_mbns_fst_scan$end = (jiga_mbns_fst_scan$midwin + 5000)/1000000
jiga_mbns_fst_scan2 = jiga_mbns_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_mbns_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_mbns_fst_scan.ranges =  jiga_mbns_fst_scan2 %>% as_granges()

jiga_mbns_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_mbns_fst_scan.ranges, all.sv.ranges))
jiga_mbns_fst_scan.ranges.coll = as_tibble(jiga_mbns_fst_scan.ranges[!jiga_mbns_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_mbns = quantile(jiga_mbns_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_mbns = quantile(jiga_mbns_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mbns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mbns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_mbns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mbns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_mbns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mbns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mbns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mbns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_mbns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mbns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_mbns <- jiga_mbns_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_mbns_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_mbns %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_mbns, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_mbns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_mbns, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_hxns_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/HXNS_JIGA_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_hxns_fst_scan)[1] <- "region"
names(jiga_hxns_fst_scan)[2] <- "chromo"
names(jiga_hxns_fst_scan)[3] <- "midwin"
names(jiga_hxns_fst_scan)[4] <- "nsites"
names(jiga_hxns_fst_scan)[5] <- "fst"

jiga_hxns_fst_scan <- dplyr::filter(jiga_hxns_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_hxns_fst_scan$nsites, breaks = 100)
jiga_hxns_fst_scan <- filter(jiga_hxns_fst_scan, nsites >= 30)

#plot with cmplot
jiga_hxns_fst_scan <- unite(jiga_hxns_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_hxns_fst_scan.plot <- as.data.frame(jiga_hxns_fst_scan[,c(2,3,4,6)])
names(jiga_hxns_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_hxns_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_hxns_fst")


#Estimate cutoff for inverted and collinear genome
jiga_hxns_fst_scan$start = (jiga_hxns_fst_scan$midwin - 4999)/1000000
jiga_hxns_fst_scan$end = (jiga_hxns_fst_scan$midwin + 5000)/1000000
jiga_hxns_fst_scan2 = jiga_hxns_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_hxns_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_hxns_fst_scan.ranges =  jiga_hxns_fst_scan2 %>% as_granges()

jiga_hxns_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_hxns_fst_scan.ranges, all.sv.ranges))
jiga_hxns_fst_scan.ranges.coll = as_tibble(jiga_hxns_fst_scan.ranges[!jiga_hxns_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_hxns = quantile(jiga_hxns_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_hxns = quantile(jiga_hxns_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_hxns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_hxns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_hxns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_hxns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_hxns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_hxns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_hxns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_hxns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_hxns_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_hxns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_hxns, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_hxns <- jiga_hxns_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_hxns_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_hxns %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_hxns, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_hxns, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_hxns, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
#Load genome-scan results:
jiga_maqu_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MAQU_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_maqu_fst_scan)[1] <- "region"
names(jiga_maqu_fst_scan)[2] <- "chromo"
names(jiga_maqu_fst_scan)[3] <- "midwin"
names(jiga_maqu_fst_scan)[4] <- "nsites"
names(jiga_maqu_fst_scan)[5] <- "fst"

jiga_maqu_fst_scan <- dplyr::filter(jiga_maqu_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_maqu_fst_scan$nsites, breaks = 100)
jiga_maqu_fst_scan <- filter(jiga_maqu_fst_scan, nsites >= 30)

#plot with cmplot
jiga_maqu_fst_scan <- unite(jiga_maqu_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_maqu_fst_scan.plot <- as.data.frame(jiga_maqu_fst_scan[,c(2,3,4,6)])
names(jiga_maqu_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(jiga_maqu_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "jiga_maqu_fst")


#Estimate cutoff for inverted and collinear genome
jiga_maqu_fst_scan$start = (jiga_maqu_fst_scan$midwin - 4999)/1000000
jiga_maqu_fst_scan$end = (jiga_maqu_fst_scan$midwin + 5000)/1000000
jiga_maqu_fst_scan2 = jiga_maqu_fst_scan[,c(2,6,7,3,4,5)]
names(jiga_maqu_fst_scan2) = c("seqnames","start","end","midwin","nsites","fst")
jiga_maqu_fst_scan.ranges =  jiga_maqu_fst_scan2 %>% as_granges()

jiga_maqu_fst_scan.ranges.inv = as_tibble(join_overlap_inner(jiga_maqu_fst_scan.ranges, all.sv.ranges))
jiga_maqu_fst_scan.ranges.coll = as_tibble(jiga_maqu_fst_scan.ranges[!jiga_maqu_fst_scan.ranges %over% all.sv.ranges,])

inv.fst.99.jiga_maqu = quantile(jiga_maqu_fst_scan$fst, prob = 1 - 0.1/100, na.rm = TRUE)
coll.fst.99.jiga_maqu = quantile(jiga_maqu_fst_scan.ranges.coll$fst, prob = 1 - 0.1/100, na.rm = TRUE)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr11 = ggplot() +
  geom_point(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr11"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 0.5, ymax=0.6), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.6, ymax=0.7), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_maqu, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_maqu, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr13 = ggplot() +
  geom_point(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr13"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_maqu, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_maqu, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr17 = ggplot() +
  geom_point(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr17"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 0.25, ymax=0.3), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=-0.01, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = inv.fst.99.jiga_maqu, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_maqu, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr18 = ggplot() +
  geom_point(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr18"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_maqu, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_maqu, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

fst.chr24 = ggplot() +
  geom_point(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst)) + 
  geom_smooth(data = filter(jiga_maqu_fst_scan, chromo == "Mme_chr24"), aes(x=midwin/1000000 , y=fst), method = "loess", span = 0.1, colour = "goldenrod") +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 1.05, ymax=1.15), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -0.15, ymax=-0.05, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 1.15, ymax=1.25), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = inv.fst.99.jiga_maqu, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_maqu, colour="orange", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("Fst") +
  theme_cowplot()

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data.jiga_maqu <- jiga_maqu_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_maqu_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data.jiga_maqu %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=1.1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 1.2, ymax=1.3), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -0.5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = Fst.data.jiga_maqu, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = inv.fst.99.jiga_maqu, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = coll.fst.99.jiga_maqu, colour="orange", linetype="dashed") +
  ylab("Fst") +
  ylim(-0.6, 1.4) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```

```{r, eval=F}
plot_grid(Fst.jiga_maqu.10kb, Fst.jiga_hxns.10kb, Fst.jiga_mbns.10kb, Fst.jiga_mdme.10kb,Fst.jiga_dxma.10kb,Fst.jiga_pany.10kb,Fst.jiga_ctva.10kb,Fst.jiga_oinc.10kb,Fst.jiga_mcnc.10kb,Fst.jiga_lrsc.10kb, align = "h", ncol = 1)
ggsave2(filename = "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/Fst_10kb_wind_genomescan_gridplot.png", width = 12, height = 10)
```

```{r, eval=F}
#Load genome-scan results:
mbns_maqu_fst_scan <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/MAQU_MBNS_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(mbns_maqu_fst_scan)[1] <- "region"
names(mbns_maqu_fst_scan)[2] <- "chromo"
names(mbns_maqu_fst_scan)[3] <- "midwin"
names(mbns_maqu_fst_scan)[4] <- "nsites"
names(mbns_maqu_fst_scan)[5] <- "fst"

mbns_maqu_fst_scan <- dplyr::filter(mbns_maqu_fst_scan, grepl('Mme_', chromo)) 
hist(mbns_maqu_fst_scan$nsites, breaks = 100)
mbns_maqu_fst_scan <- filter(mbns_maqu_fst_scan, nsites >= 30)
hist(mbns_maqu_fst_scan$nsites, breaks = 100)

#plot with cmplot
mbns_maqu_fst_scan <- unite(mbns_maqu_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
mbns_maqu_fst_scan.plot <- as.data.frame(mbns_maqu_fst_scan[,c(2,3,4,6)])
names(mbns_maqu_fst_scan.plot) <- c("SNP","chr","pos","Fst")
CMplot(mbns_maqu_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="Fst", file.output = T, file = "tiff", cex = 0.25, height = 2, width = 12, ylim = c(0,1), file.name = "mbns_maqu_fst")


#plot with cmplot
mbns_maqu_fst_scan <- unite(mbns_maqu_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
mbns_maqu_fst_scan.plot <- as.data.frame(mbns_maqu_fst_scan[,c(2,3,4,6)])
names(mbns_maqu_fst_scan.plot) <- c("SNP","chr","pos","Fst")
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr01"] <- 1
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr06"] <- 6
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr02"] <- 2
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr03"] <- 3
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr04"] <- 4
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr05"] <- 5
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr07"] <- 7
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr08"] <- 8
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr09"] <- 9
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr10"] <- 10
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr11"] <- 11
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr12"] <- 12
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr13"] <- 13
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr14"] <- 14
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr15"] <- 15
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr16"] <- 16
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr17"] <- 17
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr18"] <- 18
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr19"] <- 19
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr20"] <- 20
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr21"] <- 21
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr22"] <- 22
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr23"] <- 23
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr24"] <- 24

CMplot(mbns_maqu_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), cex = 0.5, LOG10=FALSE, ylab="Fst",ylim=c(0,1), file.output = T, width = 20, height = 5, band = 0)

#Plot chromosomes with inversions including breakpoints inferred based on mate-pair mapping
fst.chr4 = ggplot(data = filter(mbns_maqu_fst_scan, chromo == "Mme_chr04"), aes(x=midwin , y=fst)) +
  geom_point() + geom_smooth(method = "loess", span = 0.1, colour = "firebrick") +
  geom_vline(xintercept = 1917361, linetype = "dashed") + geom_vline(xintercept = 3411995, linetype = "dashed") +
  geom_vline(xintercept = 14116707, linetype = "dashed") + geom_vline(xintercept = 15566158, linetype = "dashed")

fst.chr7 = ggplot(data = filter(mbns_maqu_fst_scan, chromo == "Mme_chr07"), aes(x=midwin , y=fst)) +
  geom_point() + geom_smooth(method = "loess", span = 0.1, colour = "firebrick") +
  geom_vline(xintercept = 4756720, linetype = "dashed") + geom_vline(xintercept = 8488880, linetype = "dashed") +
  geom_vline(xintercept = 12461298, linetype = "dashed") + geom_vline(xintercept = 14309199, linetype = "dashed")

fst.chr8 = ggplot(data = filter(mbns_maqu_fst_scan, chromo == "Mme_chr08"), aes(x=midwin , y=fst)) +
  geom_point() + geom_smooth(method = "loess", span = 0.1, colour = "firebrick") +
  geom_vline(xintercept = 2705257, linetype = "dashed") + geom_vline(xintercept = 15290882, linetype = "dashed")

fst.chr11 = ggplot(data = filter(mbns_maqu_fst_scan, chromo == "Mme_chr11"), aes(x=midwin , y=fst)) +
  geom_point() + geom_smooth(method = "loess", span = 0.1, colour = "firebrick") +
  geom_vline(xintercept = 5918, linetype = "dashed") + geom_vline(xintercept = 1196215, linetype = "dashed") +
  geom_vline(xintercept = 1681812, linetype = "dashed") + geom_vline(xintercept = 2737423, linetype = "dashed") +
  geom_vline(xintercept = 8180880, linetype = "dashed") + geom_vline(xintercept = 9364918, linetype = "dashed")
	
fst.chr19 = ggplot(data = filter(mbns_maqu_fst_scan, chromo == "Mme_chr19"), aes(x=midwin , y=fst)) +
  geom_point() + geom_smooth(method = "loess", span = 0.1, colour = "firebrick") +
  geom_vline(xintercept = 12938465, linetype = "dashed") + geom_vline(xintercept = 13429978, linetype = "dashed") +
  geom_vline(xintercept = 1288951, linetype = "dashed") + geom_vline(xintercept = 1390981, linetype = "dashed")

fst.chr18 = ggplot(data = filter(mbns_maqu_fst_scan, chromo == "Mme_chr18"), aes(x=midwin , y=fst)) +
  geom_point() + geom_smooth(method = "loess", span = 0.1, colour = "firebrick") +
  geom_vline(xintercept = 2980184, linetype = "dashed") + geom_vline(xintercept = 10397061, linetype = "dashed")

fst.chr24 = ggplot(data = filter(mbns_maqu_fst_scan, chromo == "Mme_chr24"), aes(x=midwin , y=fst)) +
  geom_point() + geom_smooth(method = "loess", span = 0.1, colour = "firebrick") +
  geom_vline(xintercept = 3558528, linetype = "dashed") + geom_vline(xintercept = 12971900, linetype = "dashed")
 
#plot with cmplot
mbns_maqu_fst_scan <- unite(mbns_maqu_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
mbns_maqu_fst_scan.plot <- as.data.frame(mbns_maqu_fst_scan[,c(2,3,4,6)])
names(mbns_maqu_fst_scan.plot) <- c("SNP","chr","pos","Fst")
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr01"] <- 1
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr06"] <- 6
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr02"] <- 2
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr03"] <- 3
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr04"] <- 4
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr05"] <- 5
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr07"] <- 7
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr08"] <- 8
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr09"] <- 9
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr10"] <- 10
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr11"] <- 11
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr12"] <- 12
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr13"] <- 13
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr14"] <- 14
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr15"] <- 15
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr16"] <- 16
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr17"] <- 17
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr18"] <- 18
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr19"] <- 19
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr20"] <- 20
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr21"] <- 21
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr22"] <- 22
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr23"] <- 23
mbns_maqu_fst_scan.plot[mbns_maqu_fst_scan.plot == "Mme_chr24"] <- 24

CMplot(mbns_maqu_fst_scan.plot, type="p",plot.type="m",col=c("grey30","grey60"), cex = 0.5, LOG10=FALSE, ylab="Fst",ylim=c(0,1), file.output = F, width = 200, height = 50, band = 0)
CMplot(mbns_maqu_fst_scan.plot, type="h",plot.type="m",col=c("grey30","grey60"), cex = 1, LOG10=FALSE, ylab="Fst",ylim=c(0,1), file.output = F, width = 200, height = 50, band = 0)

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data <- mbns_maqu_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(mbns_maqu_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#Fst_manhattan 10kbwin
Fst.mbns_maqu.10kb <- ggplot(Fst.data, aes(x=BPcum, y= fst)) +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     
  ylab("Fst") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 1) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank())
Fst.mbns_maqu.10kb

ggsave2(Fst.mbns_maqu.10kb, filename = "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/Fst_10kb_wind_genomescan_mbns_maqu.png", width = 10, height = 2)
```



## Plot Fst for individual chromosomes with QTL and inversions
```{r, eval=F}
names(svs.major)[1] = "chromo"
names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"

fst.chr13 = ggplot() +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.2, ymax=-0.1), fill = "firebrick") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=0, fill = Trait), colour = "gray20") +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr13"), aes(x=(midwin/1000000) , y=fst), colour = "gold", alpha=0.5) + 
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr13"), aes(x=(midwin/1000000) , y=fst), colour = "red", alpha=0.5) + 
  geom_smooth(method = "loess", span = 0.1, colour = "firebrick")
fst.chr13
  	
fst.chr17 = ggplot() +
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.2, ymax=-0.1), fill = "firebrick") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -0.1, ymax=0, fill = Trait), colour = "gray20") +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr17"), aes(x=(midwin/1000000) , y=fst), colour = "gold", alpha=0.5) + 
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr17"), aes(x=(midwin/1000000) , y=fst), colour = "red", alpha=0.5) + 
  geom_smooth(method = "loess", span = 0.1, colour = "firebrick")
fst.chr17

```





## Plot individual Fst genome scan comparisons
```{r, eval=F}
#Load genome-scan results:
mbns_maqu_fst_scan <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/MAQU_MBNS_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(mbns_maqu_fst_scan)[1] <- "region"
names(mbns_maqu_fst_scan)[2] <- "chromo"
names(mbns_maqu_fst_scan)[3] <- "midwin"
names(mbns_maqu_fst_scan)[4] <- "nsites"
names(mbns_maqu_fst_scan)[5] <- "fst"

mbns_maqu_fst_scan <- dplyr::filter(mbns_maqu_fst_scan, grepl('Mme_', chromo)) 
hist(mbns_maqu_fst_scan$nsites, breaks = 100)
mbns_maqu_fst_scan <- filter(mbns_maqu_fst_scan, nsites >= 30)

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data <- mbns_maqu_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(mbns_maqu_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#Fst_manhattan 10kbwin
Fst.mbns_maqu.10kb <- ggplot(Fst.data, aes(x=BPcum, y= fst)) +
  geom_point(aes(color=as.factor(chromo)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("gray30", "slategray"), 100)) +
  scale_x_continuous( label = axisdf$chromo, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) +     
  ylab("Fst") +
  xlab("Chromosome 1 - 24") +
  ylim(0, 1) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank())
Fst.mbns_maqu.10kb
ggsave2(Fst.mbns_maqu.10kb, filename = "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/Fst_10kb_wind_genomescan_mbns_maqu.png", width = 10, height = 2)
```


### Create lists of highly differentiated SNPs beteween PANY and JIGA (that also are differentiated between JIGA and OINC, MBNS and MAQU)

Select top SNPs per chromosome/peak in PANY vs JIGA, and check against JIGA vs others

NY vs GA
```{r, eval=F}
snpfst.nyVSga = read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_PANY_fst_filtsnps.fst", col_names = F)
snpfst.nyVSga <-  as_tibble(dplyr::filter(snpfst.nyVSga, grepl('Mme_', X1)))

ggplot(data = filter(snpfst.nyVSga, X1 == "Mme_chr24"), aes(x=X2, y=X5)) + 
  geom_point() + 
  geom_hline(yintercept = quantile(filter(snpfst.nyVSga, X1 == "Mme_chr24")$X5, prob = 1 - 0.1/100, na.rm = TRUE), colour="firebrick")

chr1.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr01", X2 < 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr1.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr01", X2 > 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr2.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr02", X2 < 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr2.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr02", X2 > 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr3.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr03", X2 < 20000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr3.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr03", X2 > 20000000, X2 < 24000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr3.snpfst.top3 = snpfst.nyVSga %>% filter(X1 == "Mme_chr03", X2 > 24000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr4.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr04", X2 < 12000000)  %>% .[with(.,order(-X5)),]  %>% .[1:6,]
chr4.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr04", X2 > 12400000)  %>% .[with(.,order(-X5)),]  %>% .[1:10,]
chr5.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr05", X2 < 20000000)  %>% .[with(.,order(-X5)),]  %>% .[1:6,]
chr5.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr05", X2 > 20000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr6.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr06", X2 < 10000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr7.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr07", X2 < 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr7.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr07", X2 > 10000000)  %>% .[with(.,order(-X5)),]  %>% .[1:10,]
chr8.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr08")  %>% .[with(.,order(-X5)),]  %>% .[1:10,]
chr10.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr10")  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr11.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr11", X2 < 9000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr11.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr11", X2 > 9000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr12.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr12", X2 < 9000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr13.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr13", X2 < 20000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr14.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr14")  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr15.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr15")  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr16.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr16", X2 > 15000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr17.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr17", X2 > 10000000,X2 < 15000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr18.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr18", X2 < 4000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr18.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr18", X2 > 7500000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr19.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr19", X2 < 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:10,]
chr19.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr19", X2 > 10000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr20.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr20", X2 < 10000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr21.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr21")  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr22.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr22")  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr23.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr23")  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr24.snpfst.top1 = snpfst.nyVSga %>% filter(X1 == "Mme_chr24", X2 < 2000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr24.snpfst.top2 = snpfst.nyVSga %>% filter(X1 == "Mme_chr24", X2 > 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:10,]

snpfst.nyVSga.topFst = rbind(chr1.snpfst.top1,
chr1.snpfst.top2,
chr2.snpfst.top1,
chr2.snpfst.top2,
chr3.snpfst.top1,
chr3.snpfst.top2,
chr3.snpfst.top3,
chr4.snpfst.top1,
chr4.snpfst.top2,
chr5.snpfst.top1,
chr5.snpfst.top2,
chr6.snpfst.top1,
chr7.snpfst.top1,
chr7.snpfst.top2,
chr8.snpfst.top1,
chr10.snpfst.top1,
chr11.snpfst.top1,
chr11.snpfst.top2,
chr12.snpfst.top1,
chr13.snpfst.top1,
chr14.snpfst.top1,
chr15.snpfst.top1,
chr16.snpfst.top1,
chr17.snpfst.top1,
chr18.snpfst.top1,
chr18.snpfst.top2,
chr19.snpfst.top1,
chr19.snpfst.top2,
chr20.snpfst.top1,
chr21.snpfst.top1,
chr22.snpfst.top1,
chr23.snpfst.top1,
chr24.snpfst.top1,
chr24.snpfst.top2)
write_csv(snpfst.nyVSga.topFst, "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/top_snpfst_nyvsga_bychr_combined.csv")

```

MB vs GA
```{r, eval=F}
snpfst.mbVSga = read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MBNS_fst_filtsnps.fst", col_names = F)
snpfst.mbVSga <-  as_tibble(dplyr::filter(snpfst.mbVSga, grepl('Mme_', X1)))

ggplot(data = filter(snpfst.mbVSga, X1 == "Mme_chr11"), aes(x=X2, y=X5)) + 
  geom_point() + 
  geom_hline(yintercept = quantile(filter(snpfst.nyVSga, X1 == "Mme_chr11")$X5, prob = 1 - 0.1/100, na.rm = TRUE), colour="firebrick")

chr5.snpfst.mb.top1 = snpfst.mbVSga %>% filter(X1 == "Mme_chr05", X2 < 1500000)  %>% .[with(.,order(-X5)),]  %>% .[1:6,]
chr5.snpfst.mb.top2 = snpfst.mbVSga %>% filter(X1 == "Mme_chr05", X2 > 15000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr11.snpfst.top1 = snpfst.mbVSga %>% filter(X1 == "Mme_chr11", X2 < 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]
chr11.snpfst.top2 = snpfst.mbVSga %>% filter(X1 == "Mme_chr11", X2 > 5000000)  %>% .[with(.,order(-X5)),]  %>% .[1:5,]

snpfst.mbVSga.topFst = rbind(chr5.snpfst.mb.top1,
chr5.snpfst.mb.top2,
chr11.snpfst.top1,
chr11.snpfst.top2)
write_csv(snpfst.mbVSga.topFst, "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/top_snpfst_mbvsga_bychr_combined.csv")
```

Test correlation of top Fst with mb vs ga, maqu vs ga, oinc vs ga
```{r, eval=F}
snpfst.nyVSga = read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_PANY_fst_filtsnps.fst", col_names = F)
snpfst.nyVSga <-  as_tibble(dplyr::filter(snpfst.nyVSga, grepl('Mme_', X1)))
NyGA.snpfst.nyVSga.corr = inner_join(snpfst.nyVSga, snpfst.mbVSga.topFst, by=c("X1","X2"))
NyGA.snpfst.nyVSga.corr = NyGA.snpfst.nyVSga.corr[,c(1,2,3,4,5)]
names(NyGA.snpfst.nyVSga.corr) = c("X1","X2","X3","X4","X5")
snpfst.nyVSga.topFst = rbind(snpfst.nyVSga.topFst, NyGA.snpfst.nyVSga.corr)


snpfst.mbVSga = read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MBNS_fst_filtsnps.fst", col_names = F)
snpfst.mbVSga <-  as_tibble(dplyr::filter(snpfst.mbVSga, grepl('Mme_', X1)))
NyGA.snpfst.mbVSga.corr = inner_join(snpfst.nyVSga.topFst, snpfst.mbVSga, by=c("X1","X2"))
NyGA.snpfst.mbVSga.corr$comp = "MBNS"

snpfst.oincVSga = read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_OINC_fst_filtsnps.fst", col_names = F)
snpfst.oincVSga <-  as_tibble(dplyr::filter(snpfst.oincVSga, grepl('Mme_', X1)))
NyGA.snpfst.oincVSga.corr = inner_join(snpfst.nyVSga.topFst, snpfst.oincVSga, by=c("X1","X2"))
NyGA.snpfst.oincVSga.corr$comp = "OINC"

snpfst.maquVSga = read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MAQU_fst_filtsnps.fst", col_names = F)
snpfst.maquVSga <-  as_tibble(dplyr::filter(snpfst.maquVSga, grepl('Mme_', X1)))
NyGA.snpfst.maquVSga.corr = inner_join(snpfst.nyVSga.topFst, snpfst.maquVSga, by=c("X1","X2"))
NyGA.snpfst.maquVSga.corr$comp = "maqu"

snpfst.mcncVSga = read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MCNC_fst_filtsnps.fst", col_names = F)
snpfst.mcncVSga <-  as_tibble(dplyr::filter(snpfst.mcncVSga, grepl('Mme_', X1)))
NyGA.snpfst.mcncVSga.corr = inner_join(snpfst.nyVSga.topFst, snpfst.mcncVSga, by=c("X1","X2"))
NyGA.snpfst.mcncVSga.corr$comp = "mcnc"


FSTcomp = rbind(NyGA.snpfst.mcncVSga.corr, NyGA.snpfst.oincVSga.corr, NyGA.snpfst.mbVSga.corr, NyGA.snpfst.maquVSga.corr)

ggplot(FSTcomp, aes(x=X5.x, y=X5.y, colour=comp)) + geom_point() + xlim(0,1) + ylim(0,1) + facet_wrap(~X1) +
  xlab("Fst: NY vs GA") + ylab("Fst: COMP vs GA")

``` 



Fst smoothed for multiple pops and candidate chromosomes
```{r, eval=F}
jiga_mcnc_fst_scan <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MCNC_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)
jiga_oinc_fst_scan <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_OINC_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)
jiga_pany_fst_scan <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_PANY_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)
jiga_mdme_fst_scan <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MDME_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)
jiga_mbns_fst_scan <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MBNS_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)
jiga_maqu_fst_scan <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MAQU_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

#chr11
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = -0.05, ymax=1), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(jiga_maqu_fst_scan, X2 == "Mme_chr11"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#3D5A80") +
  geom_smooth(data = filter(jiga_mbns_fst_scan, X2 == "Mme_chr11"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#05668D") +
  geom_smooth(data = filter(jiga_mdme_fst_scan, X2 == "Mme_chr11"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#0F85A0") +
  geom_smooth(data = filter(jiga_pany_fst_scan, X2 == "Mme_chr11"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#FFC100") +
  geom_smooth(data = filter(jiga_oinc_fst_scan, X2 == "Mme_chr11"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#EE7600") +
  geom_smooth(data = filter(jiga_mcnc_fst_scan, X2 == "Mme_chr11"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#CE4257") +
  ylab("Fst") +
  xlab("position [Mb]") +
  theme_cowplot()
Fst_smoothed_5pops_chr11

#chr13
ggplot() +
  geom_smooth(data = filter(jiga_maqu_fst_scan, X2 == "Mme_chr13"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#3D5A80") +
  geom_smooth(data = filter(jiga_mbns_fst_scan, X2 == "Mme_chr13"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#05668D") +
  geom_smooth(data = filter(jiga_mdme_fst_scan, X2 == "Mme_chr13"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#0F85A0") +
  geom_smooth(data = filter(jiga_pany_fst_scan, X2 == "Mme_chr13"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#FFC100") +
  geom_smooth(data = filter(jiga_oinc_fst_scan, X2 == "Mme_chr13"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#EE7600") +
  geom_smooth(data = filter(jiga_mcnc_fst_scan, X2 == "Mme_chr13"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#CE4257") +
  ylab("Fst") +
  xlab("position [Mb]") +
  theme_cowplot()
Fst_smoothed_5pops_chr13

#chr17
ggplot() +
  geom_smooth(data = filter(jiga_maqu_fst_scan, X2 == "Mme_chr17"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#3D5A80") +
  geom_smooth(data = filter(jiga_mbns_fst_scan, X2 == "Mme_chr17"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#05668D") +
  geom_smooth(data = filter(jiga_mdme_fst_scan, X2 == "Mme_chr17"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#0F85A0") +
  geom_smooth(data = filter(jiga_pany_fst_scan, X2 == "Mme_chr17"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#FFC100") +
  geom_smooth(data = filter(jiga_oinc_fst_scan, X2 == "Mme_chr17"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#EE7600") +
  geom_smooth(data = filter(jiga_mcnc_fst_scan, X2 == "Mme_chr17"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#CE4257") +
  ylab("Fst") +
  xlab("position [Mb]") +
  theme_cowplot()
Fst_smoothed_5pops_chr17

#chr18
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = -0.05, ymax=1), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(jiga_maqu_fst_scan, X2 == "Mme_chr18"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#3D5A80") +
  geom_smooth(data = filter(jiga_mbns_fst_scan, X2 == "Mme_chr18"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#05668D") +
  geom_smooth(data = filter(jiga_mdme_fst_scan, X2 == "Mme_chr18"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#0F85A0") +
  geom_smooth(data = filter(jiga_pany_fst_scan, X2 == "Mme_chr18"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#FFC100") +
  geom_smooth(data = filter(jiga_oinc_fst_scan, X2 == "Mme_chr18"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#EE7600") +
  geom_smooth(data = filter(jiga_mcnc_fst_scan, X2 == "Mme_chr18"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#CE4257") +
  ylab("Fst") +
  xlab("position [Mb]") +
  theme_cowplot()
Fst_smoothed_5pops_chr18

#chr24
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = -0.05, ymax=1), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(jiga_maqu_fst_scan, X2 == "Mme_chr24"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#3D5A80") +
  geom_smooth(data = filter(jiga_mbns_fst_scan, X2 == "Mme_chr24"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#05668D") +
  geom_smooth(data = filter(jiga_mdme_fst_scan, X2 == "Mme_chr24"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#0F85A0") +
  geom_smooth(data = filter(jiga_pany_fst_scan, X2 == "Mme_chr24"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#FFC100") +
  geom_smooth(data = filter(jiga_oinc_fst_scan, X2 == "Mme_chr24"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#EE7600") +
  geom_smooth(data = filter(jiga_mcnc_fst_scan, X2 == "Mme_chr24"), aes(x=(X3/1000000), y=X5), method = "loess", se = F, span=0.005, colour = "#CE4257") +
  ylab("Fst") +
  xlab("position [Mb]") +
  theme_cowplot()
Fst_smoothed_5pops_chr24
```

Tajima's D smoothed for multiple pops and candidate chromosomes
```{r, eval=F}
tajD.jiga <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/JIGA_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.oinc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/OINC_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.pany <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/PANY_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.mdme <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MDME_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.mbns <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MBNS_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.maqu <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MAQU_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

#chr11
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = -3, ymax=2), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("TD") +
  xlab("position [Mb]") +
  theme_cowplot()
TD_smoothed_chr11

#chr13
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("TD") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr17
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("TD") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr18
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("TD") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr24
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=Tajima), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("TD") +
  xlab("position [Mb]") +
  theme_cowplot()
```

PI smoothed for multiple pops and candidate chromosomes
```{r, eval=F}
tajD.jiga <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/JIGA_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.oinc <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/OINC_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.pany <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/PANY_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.mdme <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MDME_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.mbns <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MBNS_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)
tajD.maqu <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/theta_stats/MAQU_allsites_thetastats_theta.thetasWindow.gz.pestPG", delim = "\t", col_names = T)

#chr11
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.0, ymax=0.025), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr11"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("π") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr11

#chr13
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("π") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr17
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("π") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr17


#chr18
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.0, ymax=0.025), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("π") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr18


#chr24
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.0, ymax=0.025), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("π") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr24
```




## Estimate the correlation between summary statistics (Fst, π, TD, dxy) and recombination rate in 100 kb windows:

+ Calculate PCA for each summary statistic
+ Estimate correlation between PC1 for each sumstat (e.g. Fst-PC1) with the average recombination rate for that winwdow

Adjust scripts:
```{r, eval=F}
F2.rec.map <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/RecombinationRate/F2_final_recombinationrate/F2_map_recombinationrate_estimate_allchr_setzero.txt", delim = " ")
pos_F2rr_win <- winScan(x = F2.rec.map, 
					 groups = "map", 
					 position = "phys", 
					 values = c("spline"), 
					 win_size = 50000,
					 win_step = 50000,
					 funs = c("mean"))
pos_F2rr_win <- as_tibble(pos_F2rr_win)

Fst.pany.jiga <- read_delim("/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/inversion_ms_maria/mme_jiga_pany_50kbwin_50kbstep_fst_slidingwindow.txt", delim = "\t", col_names = F)
names(Fst.pany.jiga)[2] <- "map"
names(Fst.pany.jiga)[3] <- "win_mid"
names(Fst.pany.jiga)[4] <- "snps"
names(Fst.pany.jiga)[5] <- "Fst"

Fst.pany.jiga.50kb.rr <- merge(pos_F2rr_win, Fst.pany.jiga, by = c("map","win_mid"), all.x = T)
Fst.pany.jiga.50kb.rr <- as_tibble(na.omit(Fst.pany.jiga.50kb.rr))

Fst.pany.jiga.50kb.rr.noinv <- filter(Fst.pany.jiga.50kb.rr, map != "Mme_chr08", map != "Mme_chr11", map != "Mme_chr18", map != "Mme_chr24")

Fst.pany.jiga.50kb.rr.inv8 <- filter(Fst.pany.jiga.50kb.rr, map == "Mme_chr08")
Fst.pany.jiga.50kb.rr.inv11 <- filter(Fst.pany.jiga.50kb.rr, map == "Mme_chr11")
Fst.pany.jiga.50kb.rr.inv18 <- filter(Fst.pany.jiga.50kb.rr, map == "Mme_chr18")
Fst.pany.jiga.50kb.rr.inv24 <- filter(Fst.pany.jiga.50kb.rr, map == "Mme_chr24")
Fst.pany.jiga.50kb.rr.inv <- rbind(Fst.pany.jiga.50kb.rr.inv8,Fst.pany.jiga.50kb.rr.inv11,Fst.pany.jiga.50kb.rr.inv18,Fst.pany.jiga.50kb.rr.inv24)

ggplot(data = Fst.pany.jiga.50kb.rr, aes(x = spline_mean, y = Fst)) +
  geom_point(colour = "gray70") +
  geom_point(data = Fst.pany.jiga.50kb.rr.inv, aes(x = spline_mean, y = Fst), colour = "orange") +
  xlab("Recombination rate [cM/Mb]") +
  geom_smooth(method = "loess", colour = "black") +
  theme_cowplot()
ggsave2(filename = "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/genetic_differentiation/Fst_vs_recrate_allchr_panyjiga_f2_invhighlighted.pdf", width = 5, height = 5)

ggplot(data = Fst.pany.jiga.50kb.rr, aes(x = spline_mean, y = Fst)) +
  geom_point(colour = "gray70") +
  xlab("Recombination rate [cM/Mb]") +
  geom_smooth(method = "loess", colour = "black") +
  theme_cowplot()
ggsave2(filename = "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/genetic_differentiation/Fst_vs_recrate_allchr_panyjiga_f2.pdf", width = 5, height = 5)

ggplot(data = Fst.pany.jiga.50kb.rr.noinv, aes(x = spline_mean, y = Fst)) +
  geom_point(colour = "gray70") +
  xlab("Recombination rate [cM/Mb]") +
  geom_smooth(method = "loess", colour = "black") +
  theme_cowplot()
ggsave2(filename = "/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/genetic_differentiation/Fst_vs_recrate_nomajorinvchr_panyjiga_f2.pdf", width = 5, height = 5)

cor.test(x = Fst.pany.jiga.50kb.rr$spline_mean, y = Fst.pany.jiga.50kb.rr$Fst, method = "spearman",exact = F)

cor.test(x = Fst.pany.jiga.50kb.rr.noinv$spline_mean, y = Fst.pany.jiga.50kb.rr.noinv$Fst, method = "spearman",exact = F)
```



