---
title: "Silverside Selection Analyses"
author: "Arne Jacobs"
output: github_document
highlight: pygments
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
names(knitr::knit_engines$get())
```

# Plotting locations and phenotypic trait variation for M. menidia sample sites:

Load R-packages:
```{r, eval=F, include=F}
#Plotting:
library(tidyverse)
library(ggfortify)
library(RColorBrewer)
library(wesanderson)
library(cowplot)
library(ggridges)
library(reshape2)
library(CMplot)
library(ggrastr)
library(patchwork)
library(ggbreak)

#Analysis
library(windowscanr)
library(fdrtool)
library(SuperExactTest)
library(csaw)
library(plyranges)
library(bedtoolsr)
```



# Detecting clinal polymorphism using a binomial glm of AF vs latitude

1. Load MAF data for each population and create dataframe with MAF for all filtered SNPs:
```{r, eval=F, include=F}
JIGA.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_JIGA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
LRSC.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_LRSC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
CTVA.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_CTVA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
OINC.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_OINC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCNC.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCNC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
PANY.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_PANY_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCCT.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCCT_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
DXMA.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_DXMA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MDME.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MDME_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
HXNS.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_HXNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MBNS.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MBNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MAQU.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MAQU_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))

MAF.df <- cbind(JIGA.maf[,c(1,2,6)], 
                LRSC.maf[,6], 
                MCNC.maf[,6], 
                OINC.maf[,6], 
                CTVA.maf[,6],
                PANY.maf[,6],
                MCCT.maf[,6],
                DXMA.maf[,6],
                MDME.maf[,6],
                MBNS.maf[,6],
                HXNS.maf[,6],
                MAQU.maf[,6])

names(MAF.df) <- c("chromo","position","jiga.MAF"
                   ,"lrsc.MAF"
                   ,"mcnc.MAF"
                   ,"oinc.MAF"
                   ,"ctva.MAF"
                   ,"pany.MAF"
                   ,"mcct.MAF"
                   ,"dxma.MAF"
                   ,"mdme.MAF"
                   ,"mbns.MAF"
                   ,"hxns.MAF"
                   ,"maqu.MAF")
MAF.df <- as_tibble(MAF.df)
MAF.df <- unite(MAF.df, "SNP", c("chromo","position"),sep = ":", remove = T)
write_delim(MAF.df,"~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MAFs_filtSNPs_combined.txt",delim = "\t")


MAF.df.test <- filter(MAF.df, chromo == "Mme_chr24")
MAF.df.test.snps <- MAF.df.test[,c(1,2)]
MAF.df.test.freq <- MAF.df.test[,c(3:14)]
MAF.df.t <- t(MAF.df.test)
str(MAF.df.t)
MAF.df.t <- as_tibble(MAF.df.t)
MAF.df.t <- MAF.df.t[3:14,]
MAF.df.t2 <- type_convert(MAF.df.t)
#write_csv(MAF.df.t2, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/MAF_filtSNPs_combined_transposed", col_names = T)
```


2. Add column with latitude per population:
```{r, eval=F, include=F}
pop.id <- read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/geographic_environmental_data/silverside_popsamples_coords.csv") 
pop.id$pop <- ordered(pop.id$pop, levels = c("JIGA", "LRSC", "MCNC", "OINC", "CTVA", "PANY", "MCCT", "DXMA", "MDME", "MBNS", "HXNS", "MAQU"))
lat <- c(31.0, 33.9,34.7,35.8,37.9,40.8,41.3,42.0,44.4,45.3,44.6,47.4)
```

3. Run GLM for each SNP vs lat
```{r, eval=F, include=F}
MAF.df.t3 <- cbind(lat, MAF.df.t2)
ncol(MAF.df.t3)
MAF.df.t <- as_tibble(MAF.df.t3[,c(1:ncol(MAF.df.t3))])
normalized = (x-min(x))/(max(x)-min(x))

#Test GLM for one SNP
MAF.df.t4 <- as_tibble(MAF.df.t3[,c(1:2)])
MAF.df.t4$normlat <- (MAF.df.t4$lat-min(MAF.df.t4$lat))/(max(MAF.df.t4$lat)-min(MAF.df.t4$lat))
myprobit <- glm(V1 ~ lat, family = gaussian, weights = NULL, data = MAF.df.t4)
summary(myprobit)

#GLM outputting coeff and confint
#proc_glm <- function(snps) {
#   univariate <- glm(snps ~ MAF.df.t$lat , family = binomial)
#   return(exp(cbind(Pr = coef(summary(univariate))[,3], OR = coef(univariate), #confint(univariate))))
#}


#GLM outputting p values
proc_glm <- function(snps) {
   univariate <- glm(snps ~ MAF.df.t$lat , family = gaussian, weights = NULL)
   return(coef(summary(univariate))[,4])
}

# run GLM outputting full summary for each SNP  
#glm_list <- lapply(MAF.df.t[2:10], proc_glm)
# run GLM outputting only p-value for lat
glm_list <- as_tibble(t(as_tibble(lapply(MAF.df.t[2:10], proc_glm))[2,]))

hist(-log10(glm_list$V1), breaks = 100)
glm_list.padj <- as_tibble(p.adjust(glm_list$V1, method = "BH"))
hist(glm_list.padj$value, breaks = 100)
hist(-log10(glm_list.padj$value), breaks = 100)
min(glm_list.padj$value)

glm_list.fdr <- fdrtool(glm_list$V1, statistic="pvalue",
plot=F, color.figure=F, verbose=TRUE,
cutoff.method="fndr")
fdrtool(glm_list$V1, statistic="pvalue",
plot=T, color.figure=T, verbose=TRUE,
cutoff.method="fndr")
hist(glm_list.fdr$qval, breaks = 100)
min(glm_list.fdr$qval)
```


4. Run the full optimized pipeline for all SNPs
```{r, eval=F, include=F}
nohup Rscript GLM_maf_latitude.R > glm_maf_lat_Rscript_test_nohup.log &
```


5.Plot GLM results across chromosomes with major qtl:

```{r, eval=F, include=F}
qtl <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
glm.lat.df = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/glm_clinal_snps/GLM_maf_vs_lat_bysnp_res.csv") 
names(glm.lat.df)[1] = "Chr"
sv.maria <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv <- read.csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
```

chr11
```{r, eval=F, include=F}
glm.lat.df.chr11 = filter(glm.lat.df, Chr=="Mme_chr11")
qtl.chr11 = filter(qtl, Chr=="Mme_chr11")
sv.maria.chr11 = filter(sv.maria, chr=="Mme_chr11")
svs.major.chr11 = filter(svs.major, mme_chr=="Mme_chr11")

chr.11.glm.qtl = ggplot() +
  geom_rect(data = sv.maria.chr11, aes(xmin=inv_start, xmax=inv_end, ymin = -0.2, ymax=0), fill = "steelblue3", alpha=0.5) +
  geom_rect(data = svs.major.chr11, aes(xmin=startMB , xmax=endMB, ymin = -0.4, ymax=-0.2), fill = "firebrick", alpha=0.5) +
  geom_rect(data = qtl.chr11, aes(xmin=start, xmax=end, ymin = 10, ymax=10.2, fill = Trait), alpha=0.5) +
  geom_point(data = glm.lat.df.chr11, aes(x=(position/1000000), y=(-log10(pval))), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = 5.2, colour = "firebrick", linetype="dashed") +
  ylab("-log10(p-value)") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())
```

chr13
```{r, eval=F, include=F}
glm.lat.df.chr13 = filter(glm.lat.df, Chr=="Mme_chr13")
qtl.chr13 = filter(qtl, Chr=="Mme_chr13")
sv.maria.chr13 = filter(sv.maria, chr=="Mme_chr13")
svs.major.chr13 = filter(svs.major, mme_chr=="Mme_chr13")

chr.13.glm.qtl = ggplot() +
  geom_rect(data = svs.major.chr13, aes(xmin=startMB , xmax=endMB, ymin = -0.4, ymax=-0.2), fill = "firebrick", alpha=0.5) +
  geom_rect(data = qtl.chr13, aes(xmin=start, xmax=end, ymin = 10, ymax=10.2, fill = Trait), alpha=0.5) +
  geom_point(data = glm.lat.df.chr13, aes(x=(position/1000000), y=(-log10(pval))), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = 5.2, colour = "firebrick", linetype="dashed") +
  ylab("-log10(p-value)") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())
```

chr17
```{r, eval=F, include=F}
glm.lat.df.chr17 = filter(glm.lat.df, Chr=="Mme_chr17")
qtl.chr17 = filter(qtl, Chr=="Mme_chr17")
sv.maria.chr17 = filter(sv.maria, chr=="Mme_chr17")
svs.major.chr17 = filter(svs.major, mme_chr=="Mme_chr17")

chr.17.glm.qtl = ggplot() +
  geom_rect(data = svs.major.chr17, aes(xmin=startMB , xmax=endMB, ymin = -0.4, ymax=-0.2), fill = "firebrick", alpha=0.5) +
  geom_rect(data = qtl.chr17, aes(xmin=start, xmax=end, ymin = 10, ymax=10.2, fill = Trait), alpha=0.5) +
  geom_point(data = glm.lat.df.chr17, aes(x=(position/1000000), y=(-log10(pval))), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = 5.2, colour = "firebrick", linetype="dashed") +
  ylab("-log10(p-value)") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())
```

chr18
```{r, eval=F, include=F}
glm.lat.df.chr18 = filter(glm.lat.df, Chr=="Mme_chr18")
qtl.chr18 = filter(qtl, Chr=="Mme_chr18")
sv.maria.chr18 = filter(sv.maria, chr=="Mme_chr18")
svs.major.chr18 = filter(svs.major, mme_chr=="Mme_chr18")

chr.18.glm.qtl = ggplot() +
  geom_rect(data = sv.maria.chr18, aes(xmin=inv_start, xmax=inv_end, ymin = -0.2, ymax=0), fill = "steelblue3", alpha=0.5) +
  geom_rect(data = svs.major.chr18, aes(xmin=startMB , xmax=endMB, ymin = -0.4, ymax=-0.2), fill = "firebrick", alpha=0.5) +
  geom_rect(data = qtl.chr18, aes(xmin=start, xmax=end, ymin = 10, ymax=10.2, fill = Trait), alpha=0.5) +
  geom_point(data = glm.lat.df.chr18, aes(x=(position/1000000), y=(-log10(pval))), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = 5.2, colour = "firebrick", linetype="dashed") +
  ylab("-log10(p-value)") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())
```

chr24
```{r, eval=F, include=F}
glm.lat.df.chr24 = filter(glm.lat.df, Chr=="Mme_chr24")
qtl.chr24 = filter(qtl, Chr=="Mme_chr24")
sv.maria.chr24 = filter(sv.maria, chr=="Mme_chr24")
svs.major.chr24 = filter(svs.major, mme_chr=="Mme_chr24")

chr.24.glm.qtl = ggplot() +
  geom_rect(data = sv.maria.chr24, aes(xmin=inv_start, xmax=inv_end, ymin = -0.2, ymax=0), fill = "steelblue3", alpha=0.5) +
  geom_rect(data = svs.major.chr24, aes(xmin=startMB , xmax=endMB, ymin = -0.4, ymax=-0.2), fill = "firebrick", alpha=0.5) +
  geom_rect(data = qtl.chr24, aes(xmin=start, xmax=end, ymin = 10, ymax=10.2, fill = Trait), alpha=0.5) +
  geom_point(data = glm.lat.df.chr24, aes(x=(position/1000000), y=(-log10(pval))), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = 5.2, colour = "firebrick", linetype="dashed") +
  ylab("-log10(p-value)") +
    theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())
```

Combine figures
```{r, eval=F, include=F}
glm.qtl.plots = (chr.11.glm.qtl / chr.13.glm.qtl / chr.17.glm.qtl) + (chr.18.glm.qtl / chr.24.glm.qtl / plot_spacer())

glm.qtl.plots2 = chr.11.glm.qtl + chr.13.glm.qtl + chr.17.glm.qtl + chr.18.glm.qtl + chr.24.glm.qtl +  plot_layout(ncol = 2)
```




***************************************************************************************************************




# Detect signals of selective sweeps using sweepfinder2:

+ Create a derived allele frequency input file for each chromosome and population:

  position x n folded
  460000 9 100 0
  460010 100 100 0
  460210 30 78 1
  463000 0 94 0

with, position = position on chromosome
x = allele count
n = sample size (n genotyped individuals)
folded = indicator as to whether the site has been polarized. If the site is polarized, then the entry in the folded column should be 0


Estimate derived AF for all variant sites using ancestral outgroup genome for each population
```{bash, eval=F}
#!/bin/bash
export OMP_NUM_THREADS=4

BAMLIST=$1 # Path to textfile listing bamfiles to include in global SNP calling with absolute paths
REFERENCE=$2 # Path to reference genome
OUTDIR=$3 # Path to output files
OUTBASE=$4 # Basename for output files
SITES=$5 # Global SNPlist

/home/nt246_0001/Programs/angsd_0.912/angsd/angsd -b $BAMLIST -anc $REFERENCE -out $OUTDIR$OUTBASE -sites $SITES -dosaf 1 -GL 1 -doGlf 3 -doMaf 1 -doMajorMinor 1 -P 16

nohup sh angsd_estMAF_bypop.sh 
```

```{bash, eval=F}
nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_JIGA.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ JIGA_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/jiga_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_LRSC.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ LRSC_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/lrsc_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCNC.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ MCNC_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/mcnc_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_OINC.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ OINC_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/oinc_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_CTVA.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ CTVA_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/ctva_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_PANY.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ PANY_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/pany_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MCCT.txt  /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ MCCT_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/mcct_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_DXMA.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ DXMA_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/dxma_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MDME.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ MDME_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/mdme_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MBNS.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ MBNS_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/mbns_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_HXNS.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ HXNS_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/hxns_daf_snps_nohup.log &

nohup sh angsd_estDAF_bypop.sh /workdir/arne/sampleinfo/BamFiles_clean_list_MAQU.txt /workdir/arne/results/outgroup_genome/menidia_outgrp_ancfasta_maskedsnps.fa /workdir/arne/results/snp_datasets/ MAQU_daf /workdir/arne/results/snp_datasets/mme_global_filtSNP_list.txt > /workdir/arne/output_logfiles/maqu_daf_snps_nohup.log &
```


Plot derived allele frequency distributions:
```{r, eval=F}
JIGA.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/JIGA_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
JIGA.daf$pop = "JIGA"
LRSC.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/LRSC_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
LRSC.daf$pop = "LRSC"
MCNC.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/MCNC_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCNC.daf$pop = "MCNC"
OINC.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/OINC_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
OINC.daf$pop = "OINC"
CTVA.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/CTVA_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
CTVA.daf$pop = "CTVA"
PANY.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/PANY_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
PANY.daf$pop = "PANY"
MCCT.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/MCCT_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCCT.daf$pop = "MCCT"
DXMA.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/DXMA_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
DXMA.daf$pop = "DXMA"
MDME.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/MDME_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MDME.daf$pop = "MDME"
MBNS.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/MBNS_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MBNS.daf$pop = "MBNS"
HXNS.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/HXNS_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
HXNS.daf$pop = "HXNS"
MAQU.daf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/MAQU_daf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MAQU.daf$pop = "MAQU"


DAF.comb = as_tibble(rbind(JIGA.daf, LRSC.daf, MCNC.daf, OINC.daf, CTVA.daf, PANY.daf,
                           MCCT.daf, DXMA.daf, MDME.daf, MBNS.daf, HXNS.daf, MAQU.daf))


ggplot(DAF.comb, aes(x = unknownEM, y = pop)) + geom_density_ridges2(scale = 0.9) + theme_ridges(grid = FALSE, center_axis_labels = TRUE)
```


Reformat to match requirements for sweepfinder and create individual file for each chromosome:

```{r, eval=F}
chr = c("Mme_chr01","Mme_chr02","Mme_chr03","Mme_chr03","Mme_chr04","Mme_chr05","Mme_chr06","Mme_chr07","Mme_chr08","Mme_chr09","Mme_chr10","Mme_chr11","Mme_chr12","Mme_chr13","Mme_chr14","Mme_chr15","Mme_chr16","Mme_chr17","Mme_chr18","Mme_chr19","Mme_chr20","Mme_chr21","Mme_chr22","Mme_chr23","Mme_chr24")
list_of_chr_sf2 <- list()
for (i in chr) { 
  df = filter(HXNS.daf, chromo == i)
  df$x=round((df$unknownEM)*(2*(df$nInd)), digits = 0)
  df$folded=0
  df$n=(df$nInd*2)
  df=filter(df, x > 0)
  sf2.inp = df[,c(2,9,11,10)]
  list_of_chr_sf2[[i]] <- sf2.inp
}

write_delim(list_of_chr_sf2$Mme_chr01, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr01_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr02, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr02_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr03, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr03_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr04, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr04_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr05, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr05_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr06, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr06_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr07, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr07_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr08, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr08_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr09, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr09_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr10, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr10_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr11, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr11_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr12, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr12_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr13, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr13_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr14, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr14_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr15, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr15_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr16, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr16_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr17, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr17_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr18, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr18_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr19, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr19_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr20, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr20_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr21, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr21_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr22, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr22_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr23, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr23_snps_sf2_daf.txt",delim = "\t")
write_delim(list_of_chr_sf2$Mme_chr24, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr24_snps_sf2_daf.txt",delim = "\t")

combinedfreq.file = rbind(list_of_chr_sf2$Mme_chr01,list_of_chr_sf2$Mme_chr02,list_of_chr_sf2$Mme_chr03,list_of_chr_sf2$Mme_chr04,list_of_chr_sf2$Mme_chr05,list_of_chr_sf2$Mme_chr06,list_of_chr_sf2$Mme_chr07,list_of_chr_sf2$Mme_chr08,list_of_chr_sf2$Mme_chr09,list_of_chr_sf2$Mme_chr10,list_of_chr_sf2$Mme_chr11,list_of_chr_sf2$Mme_chr12,list_of_chr_sf2$Mme_chr13,list_of_chr_sf2$Mme_chr14,list_of_chr_sf2$Mme_chr15,list_of_chr_sf2$Mme_chr16,list_of_chr_sf2$Mme_chr17,list_of_chr_sf2$Mme_chr18,list_of_chr_sf2$Mme_chr19,list_of_chr_sf2$Mme_chr20,list_of_chr_sf2$Mme_chr21,list_of_chr_sf2$Mme_chr22,list_of_chr_sf2$Mme_chr23,list_of_chr_sf2$Mme_chr24)
write_delim(combinedfreq.file, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_combined_daf_freqfile.txt",delim = "\t")
```



Create a recombination rate file for each site by extrapolating from female JIGA linkage map (for JIGA, LRSC), Hybrid map (for MCNC, OINC, CTVA) and PANY map (PANY, MCCT, DXMA, MDME, MBNS, HXNS, MAQU)

Find and Replace population name, change input map if needed
```{r, eval=F}
map = read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/RecombinationRate/PANY_F1_recombinationrate/PANY_femF1_recombrate_forplotting.txt", delim = "\t", col_names = T)
map = map[,c(2,3,4,5,7)]
names(map) = c("chromo","mkr","position","cM","recomb")
df.map = full_join(PANY.daf, map, by=c("chromo","position"))
df.map = arrange(df.map, chromo, position)


chr = c("Mme_chr01","Mme_chr02","Mme_chr03","Mme_chr03","Mme_chr04","Mme_chr05","Mme_chr06","Mme_chr07","Mme_chr08","Mme_chr09","Mme_chr10","Mme_chr11","Mme_chr12","Mme_chr13","Mme_chr14","Mme_chr15","Mme_chr16","Mme_chr17","Mme_chr18","Mme_chr19","Mme_chr20","Mme_chr21","Mme_chr22","Mme_chr23","Mme_chr24")
list_of_chr <- list()
for (i in chr) { 
  df = filter(df.map, chromo == i)
  n = length(df$position)
  interpolated_map = approx(x=df$cM, n=n)
  df = as_tibble(cbind(df, interpolated_map$y))
  df.filt = filter(df, pop == "PANY")
  df.filt = df[,c(1,2,12)]
  names(df.filt)[3] = "cM"
  df.filt = mutate(df.filt, rate= (cM-lag(cM)))
  df.filt[is.na(df.filt)] <- 0
  df.filt = df.filt[,c(2,4)]
  list_of_chr[[i]] <- df.filt
}

write_delim(list_of_chr$Mme_chr01, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr01_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr02, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr02_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr03, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr03_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr04, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr04_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr05, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr05_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr06, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr06_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr07, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr07_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr08, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr08_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr09, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr09_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr10, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr10_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr11, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr11_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr12, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr12_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr13, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr13_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr14, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr14_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr15, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr15_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr16, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr16_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr17, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr17_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr18, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr18_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr19, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr19_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr20, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr20_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr21, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr21_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr22, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr22_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr23, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr23_snps_interpolated_cM_distance.txt",delim = "\t")
write_delim(list_of_chr$Mme_chr24, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr24_snps_interpolated_cM_distance.txt",delim = "\t")

df = as_tibble(list_of_chr$Mme_chr18)
ggplot(data = df, aes(x=position, y=rate)) +
  geom_point()
```


Run Sweepfinder2:

1. Estimate genome-wide spectrum:
```{bash, eval=F}
nohup /workdir/arne/programs/SF2/SweepFinder2 -f /workdir/arne/results/selection/sweepfinder2/jiga_sf2_input/jiga_combined_daf_freqfile.txt /workdir/arne/results/selection/sweepfinder2/jiga_sf2_input/jiga_genomewide_daf_spectfile &

nohup /workdir/arne/programs/SF2/SweepFinder2 -f /workdir/arne/results/selection/sweepfinder2/oinc_sf2_input/oinc_combined_daf_freqfile.txt /workdir/arne/results/selection/sweepfinder2/oinc_sf2_input/oinc_genomewide_daf_spectfile &

nohup /workdir/arne/programs/SF2/SweepFinder2 -f /workdir/arne/results/selection/sweepfinder2/pany_sf2_input/pany_combined_daf_freqfile.txt /workdir/arne/results/selection/sweepfinder2/pany_sf2_input/pany_genomewide_daf_spectfile &

nohup /workdir/arne/programs/SF2/SweepFinder2 -f /workdir/arne/results/selection/sweepfinder2/mdme_sf2_input/mdme_combined_daf_freqfile.txt /workdir/arne/results/selection/sweepfinder2/mdme_sf2_input/mdme_genomewide_daf_spectfile &

nohup /workdir/arne/programs/SF2/SweepFinder2 -f /workdir/arne/results/selection/sweepfinder2/mbns_sf2_input/mbns_combined_daf_freqfile.txt /workdir/arne/results/selection/sweepfinder2/mbns_sf2_input/mbns_genomewide_daf_spectfile &

nohup /workdir/arne/programs/SF2/SweepFinder2 -f /workdir/arne/results/selection/sweepfinder2/hxns_sf2_input/hxns_combined_daf_freqfile.txt /workdir/arne/results/selection/sweepfinder2/hxns_sf2_input/hxns_genomewide_daf_spectfile &

nohup /workdir/arne/programs/SF2/SweepFinder2 -f /workdir/arne/results/selection/sweepfinder2/maqu_sf2_input/maqu_combined_daf_freqfile.txt /workdir/arne/results/selection/sweepfinder2/maqu_sf2_input/maqu_genomewide_daf_spectfile &
```


2a. Run Sweepfinder on all chromosomes in 1kb grid with recombination rate:
```{bash, eval=F}
#!/bin/bash

GRID=$1 #Estimate CLR for every nth SNP
DIR=$2 #Directory for input and output files
POP=$3 #population code used in input and output files
CHRLIST=$4 #List with chromosome numbers: one number per row (01,02,03...) 

for CHR in `cat $CHRLIST`; do

/workdir/arne/programs/SF2/SweepFinder2 -lrg $GRID $DIR$POP'_chr'$CHR'_snps_sf2_daf.txt' $DIR$POP'_genomewide_daf_spectfile' $DIR$POP'_chr'$CHR'_snps_interpolated_cM_distance.txt' $DIR$POP'_chr'$CHR'_sf2_output.txt'

done
```


Adjust population names and run script in scripts folder (divided into four sets to parallelise computing):
```{bash, eval=F}
nohup sh run_sf2.sh 1000 /workdir/arne/results/selection/sweepfinder2/pany_sf2_input/ pany /workdir/arne/results/selection/sweepfinder2/chrlist_chr01_to_chr06.txt > /workdir/arne/output_logfiles/sf2_chr01_to_chr06_pany_nohup.log &

nohup sh run_sf2.sh 1000 /workdir/arne/results/selection/sweepfinder2/pany_sf2_input/ pany /workdir/arne/results/selection/sweepfinder2/chrlist_chr07_to_chr12.txt > /workdir/arne/output_logfiles/sf2_chr07_to_chr12_pany_nohup.log &

nohup sh run_sf2.sh 1000 /workdir/arne/results/selection/sweepfinder2/pany_sf2_input/ pany /workdir/arne/results/selection/sweepfinder2/chrlist_chr13_to_chr18.txt > /workdir/arne/output_logfiles/sf2_chr13_to_chr18_pany_nohup.log &

nohup sh run_sf2.sh 1000 /workdir/arne/results/selection/sweepfinder2/pany_sf2_input/ pany /workdir/arne/results/selection/sweepfinder2/chrlist_chr19_to_chr24.txt > /workdir/arne/output_logfiles/sf2_chr19_to_chr24_pany_nohup.log &
```

Run maqu chromosome 11 with F2 map to check for effect of recombination:
```{bash, eval=F}
nohup sh run_sf2_F2map.sh 1000 /workdir/arne/results/selection/sweepfinder2/maqu_sf2_input/ maqu /workdir/arne/results/selection/sweepfinder2/chrlist_chr07_to_chr12.txt > /workdir/arne/output_logfiles/sf2_F2_chr07_to_chr12_maqu_nohup.log &
```

2b. Run Sweepfinder on all chromosomes in 1kb grid without recombination rate:
```{bash, eval=F}
#!/bin/bash

GRID=$1 #Estimate CLR for every nth SNP
DIR=$2 #Directory for input and output files
POP=$3 #population code used in input and output files
CHRLIST=$4 #List with chromosome numbers: one number per row (01,02,03...) 

for CHR in `cat $CHRLIST`; do

/workdir/arne/programs/SF2/SweepFinder2 -lrg $GRID $DIR$POP'_chr'$CHR'_snps_sf2_daf.txt' $DIR$POP'_genomewide_daf_spectfile' $DIR$POP'_chr'$CHR'_snps_interpolated_cM_distance.txt' $DIR$POP'_chr'$CHR'_sf2_output.txt'

done
```


Adjust population names and run script in scripts folder (divided into four sets to parallelise computing):
```{bash, eval=F}
nohup sh run_sf2_norec.sh 1000 /workdir/arne/results/selection/sweepfinder2/maqu_sf2_input/ maqu /workdir/arne/results/selection/sweepfinder2/chrlist_chr01_to_chr06.txt > /workdir/arne/output_logfiles/sf2_chr01_to_chr06_maqu_nohup.log &

nohup sh run_sf2_norec.sh 1000 /workdir/arne/results/selection/sweepfinder2/maqu_sf2_input/ maqu /workdir/arne/results/selection/sweepfinder2/chrlist_chr07_to_chr12.txt > /workdir/arne/output_logfiles/sf2_chr07_to_chr12_maqu_nohup.log &

nohup sh run_sf2_norec.sh 1000 /workdir/arne/results/selection/sweepfinder2/maqu_sf2_input/ maqu /workdir/arne/results/selection/sweepfinder2/chrlist_chr13_to_chr18.txt > /workdir/arne/output_logfiles/sf2_chr13_to_chr18_maqu_nohup.log &

nohup sh run_sf2_norec.sh 1000 /workdir/arne/results/selection/sweepfinder2/maqu_sf2_input/ maqu /workdir/arne/results/selection/sweepfinder2/chrlist_chr19_to_chr24.txt > /workdir/arne/output_logfiles/sf2_chr19_to_chr24_maqu_nohup.log &
```


## Plot CLR across the genome:
Load inversion breakpoints:
```{r, eval=F}
JIGA.F1 <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/RecombinationRate/JIGA_F1_recombinationrate/JIGA_femF1_renamedchr.txt", delim = "\t")
PANY.F1 <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/RecombinationRate/PANY_F1_recombinationrate/PANY_femF1_recombrate_renamedchr_rev.txt", delim = "\t")
PAxJI.F2 <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/RecombinationRate/F2_final_recombinationrate/PAxJI_femF2_renamedchr_rev.txt", delim = "\t")

sv <- read.csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major.clr = filter(sv, inv.type == "major" | inv.type == "minor")
svs.major.clr = separate(svs.major.clr, col = "mme_chr", into = c("mme","CHR"), sep = "_chr", remove = F)
write_csv(svs.major.clr, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr_CLR.csv")
svs.major.clr = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr_CLR.csv")
svs.major.clr = svs.major.clr %>% mutate(CHR = as.numeric(CHR))


sv.maria.clr <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria.clr = separate(sv.maria.clr, col = "chr", into = c("mme","CHR"), sep = "_chr", remove = F)
write_csv(sv.maria.clr, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/mainv_CLR.csv")
sv.maria.clr = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/mainv_CLR.csv")
sv.maria.clr = sv.maria.clr %>% mutate(CHR = as.numeric(CHR))

qtl <- read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
qtl.clr = qtl
qtl.clr = separate(qtl.clr, col = "Chr", into = c("mme","CHR"), sep = "_chr", remove = F)
write_csv(qtl.clr, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/qtl_CLR.csv")
qtl.clr = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/qtl_CLR.csv")
qtl.clr = qtl.clr %>% mutate(CHR = as.numeric(CHR))
```





### Plot CLR across the genome in MAQU:

With recombination rate grid
```{r, eval=F}
maqu.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr01_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr02_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr03_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr04_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr05_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr06_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr07_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr08_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr09_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr10_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr11_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr12_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr13_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr14_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr15_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr16_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr17_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr18_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr19_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr20_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr21_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr22_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr23_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr24_sf2_output.txt",delim = "\t")

maqu.sf2_1kb.chr01$CHR=1
maqu.sf2_1kb.chr02$CHR=2
maqu.sf2_1kb.chr03$CHR=3
maqu.sf2_1kb.chr04$CHR=4
maqu.sf2_1kb.chr05$CHR=5
maqu.sf2_1kb.chr06$CHR=6
maqu.sf2_1kb.chr07$CHR=7
maqu.sf2_1kb.chr08$CHR=8
maqu.sf2_1kb.chr09$CHR=9
maqu.sf2_1kb.chr10$CHR=10
maqu.sf2_1kb.chr11$CHR=11
maqu.sf2_1kb.chr12$CHR=12
maqu.sf2_1kb.chr13$CHR=13
maqu.sf2_1kb.chr14$CHR=14
maqu.sf2_1kb.chr15$CHR=15
maqu.sf2_1kb.chr16$CHR=16
maqu.sf2_1kb.chr17$CHR=17
maqu.sf2_1kb.chr18$CHR=18
maqu.sf2_1kb.chr19$CHR=19
maqu.sf2_1kb.chr20$CHR=20
maqu.sf2_1kb.chr21$CHR=21
maqu.sf2_1kb.chr22$CHR=22
maqu.sf2_1kb.chr23$CHR=23
maqu.sf2_1kb.chr24$CHR=24

maqu.sf2.res=rbind(maqu.sf2_1kb.chr01,
maqu.sf2_1kb.chr02,
maqu.sf2_1kb.chr03,
maqu.sf2_1kb.chr04,
maqu.sf2_1kb.chr05,
maqu.sf2_1kb.chr06,
maqu.sf2_1kb.chr07,
maqu.sf2_1kb.chr08,
maqu.sf2_1kb.chr09,
maqu.sf2_1kb.chr10,
maqu.sf2_1kb.chr11,
maqu.sf2_1kb.chr12,
maqu.sf2_1kb.chr13,
maqu.sf2_1kb.chr14,
maqu.sf2_1kb.chr15,
maqu.sf2_1kb.chr16,
maqu.sf2_1kb.chr17,
maqu.sf2_1kb.chr18,
maqu.sf2_1kb.chr19,
maqu.sf2_1kb.chr20,
maqu.sf2_1kb.chr21,
maqu.sf2_1kb.chr22,
maqu.sf2_1kb.chr23,
maqu.sf2_1kb.chr24)

#Custom plot
maqu.sf2.data <- maqu.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(maqu.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = maqu.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

qtl = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint.csv")
qtl.data <- qtl %>% 
  # Compute chromosome size
  group_by(Chr) %>% 
  summarise(chr_len=max(start)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(qtl, ., by=c("Chr"="Chr")) %>%
  # Add a cumulative position of each SNP
  arrange(Chr, start) %>%
  mutate(BPcum=start+tot) %>%
  mutate(BPcumr=end+tot)
qtl.axisdf = qtl.data %>% group_by(Chr) %>% summarize(center=(max(BPcum) + min(BPcum) ) / 2 )

#Estimate threshold excluding chromosomes containing major inversions:
maqu.sf2.data.noinv = filter(maqu.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(maqu.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
maqu.sf2.data.top=filter(maqu.sf2.data, LR > quantile(maqu.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
maqu.sf2.data.top2=filter(maqu.sf2.data, LR > quantile(maqu.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))

maqu.sf2.data.top.all = rbind(maqu.sf2.data.top, maqu.sf2.data.top2)
write_delim(maqu.sf2.data.top.all, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/top_selected_snps/maqu_sf2_selectedSNPs_twothresholds.txt", delim = "\t")

maqu.clr.plot = ggplot() +
  geom_rect(data = sv.maria.clr, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=8300), fill = "steelblue", alpha=0.75) +
  geom_rect(data = qtl.clr, aes(xmin=start, xmax=end, ymin = 8400, ymax=8450), fill = "gray20", colour = "gray20") +
  geom_point(data = maqu.sf2.data, aes(x=(location/1000000), y= LR), colour="black", alpha=1, size=2, shape = 20) +
  geom_hline(yintercept = 952.8804, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = 179.9405, colour="steelblue", linetype="dashed") +
  ylab("CLR") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  scale_y_break(c(3000, 8000)) +
  facet_grid(~CHR, scales = "free_x")

maqu.clr.plot2 = ggplot() +
  geom_rect(data = sv.maria.clr, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=8300), fill = "steelblue", alpha=0.75) +
  geom_rect(data = qtl.clr, aes(xmin=start, xmax=end, ymin = 8400, ymax=8450), fill = "gray20", colour = "gray20") +
  geom_point(data = maqu.sf2.data, aes(x=(location/1000000), y= LR), colour="black", alpha=1, size=2, shape = 20) +
  geom_hline(yintercept = 952.8804, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = 179.9405, colour="steelblue", linetype="dashed") +
  ylab("CLR") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~CHR, scales = "free_x")


#CM manhattan plot
maqu.sf2.data2 <- unite(maqu.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
maqu.sf2.data2.plot <- as.data.frame(maqu.sf2.data2[,c(1,5,2,3)])
names(maqu.sf2.data2.plot) <- c("SNP","chr","pos","LR")

CMplot(maqu.sf2.data2.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", threshold = c(179.9405, 952.8804), threshold.col = c("steelblue","firebrick"), cex = 0.25, height = 2, width = 12, amplify = T, signal.cex = 0.25, signal.col = c("firebrick","steelblue"), file.name = "maqu_clr")



#Plot CLR for major QTL
clr.maqu.chr11 = ggplot() +
  geom_point(data = maqu.sf2_1kb.chr11, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(qtl, Chr == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -700, ymax=-400, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, , ymin = 8500, ymax=9000), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(maqu.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="steelblue", linetype="dashed") +
  geom_hline(yintercept = quantile(maqu.sf2.data.top$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()
clr.maqu.chr11

clr.maqu.chr13 = ggplot() +
  geom_point(data = maqu.sf2_1kb.chr13, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 200, ymax=250), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -50, ymax=-20, fill = Trait), alpha = 0.5, colour = "gray20")+
  geom_hline(yintercept = quantile(maqu.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="steelblue", linetype="dashed") +
  geom_hline(yintercept = quantile(maqu.sf2.data.top$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick",linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()+
  theme(legend.position = "none")
clr.maqu.chr13

clr.maqu.chr17 = ggplot() +
  geom_point(data = maqu.sf2_1kb.chr17, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(qtl, Chr == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -20, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = quantile(maqu.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="steelblue", linetype="dashed") +
  #geom_hline(yintercept = quantile(maqu.sf2.data.top$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick",linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()+
  theme(legend.position = "none")
clr.maqu.chr17

clr.maqu.chr18 = ggplot() +
  geom_point(data = maqu.sf2_1kb.chr18, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(qtl, Chr == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -50, ymax=-20, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, , ymin = 500, ymax=510), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(maqu.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()+
  theme(legend.position = "none")
clr.maqu.chr18

clr.maqu.chr24 = ggplot() +
  geom_point(data = maqu.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(sv.maria.clr, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 2500, ymax=2600), fill = "steelblue3", colour="grey20") +
  xlab("Chromosome 24 [Mb]") +
  ylab("CLR") +
  theme_cowplot()
clr.maqu.chr24

clr.maqu = (clr.maqu.chr11 + clr.maqu.chr13 + clr.maqu.chr17 + clr.maqu.chr18 + clr.maqu.chr24)
```





### HXNS CLR
```{r, eval=F}
hxns.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr01_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr02_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr03_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr04_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr05_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr06_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr07_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr08_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr09_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr10_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr11_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr12_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr13_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr14_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr15_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr16_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr17_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr18_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr19_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr20_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr21_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr22_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr23_sf2_output.txt",delim = "\t")
hxns.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/hxns_sf2_input/hxns_chr24_sf2_output.txt",delim = "\t")

hxns.sf2_1kb.chr01$CHR=1
hxns.sf2_1kb.chr02$CHR=2
hxns.sf2_1kb.chr03$CHR=3
hxns.sf2_1kb.chr04$CHR=4
hxns.sf2_1kb.chr05$CHR=5
hxns.sf2_1kb.chr06$CHR=6
hxns.sf2_1kb.chr07$CHR=7
hxns.sf2_1kb.chr08$CHR=8
hxns.sf2_1kb.chr09$CHR=9
hxns.sf2_1kb.chr10$CHR=10
hxns.sf2_1kb.chr11$CHR=11
hxns.sf2_1kb.chr12$CHR=12
hxns.sf2_1kb.chr13$CHR=13
hxns.sf2_1kb.chr14$CHR=14
hxns.sf2_1kb.chr15$CHR=15
hxns.sf2_1kb.chr16$CHR=16
hxns.sf2_1kb.chr17$CHR=17
hxns.sf2_1kb.chr18$CHR=18
hxns.sf2_1kb.chr19$CHR=19
hxns.sf2_1kb.chr20$CHR=20
hxns.sf2_1kb.chr21$CHR=21
hxns.sf2_1kb.chr22$CHR=22
hxns.sf2_1kb.chr23$CHR=23
hxns.sf2_1kb.chr24$CHR=24

hxns.sf2.res=rbind(hxns.sf2_1kb.chr01,
hxns.sf2_1kb.chr02,
hxns.sf2_1kb.chr03,
hxns.sf2_1kb.chr04,
hxns.sf2_1kb.chr05,
hxns.sf2_1kb.chr06,
hxns.sf2_1kb.chr07,
hxns.sf2_1kb.chr08,
hxns.sf2_1kb.chr09,
hxns.sf2_1kb.chr10,
hxns.sf2_1kb.chr11,
hxns.sf2_1kb.chr12,
hxns.sf2_1kb.chr13,
hxns.sf2_1kb.chr14,
hxns.sf2_1kb.chr15,
hxns.sf2_1kb.chr16,
hxns.sf2_1kb.chr17,
hxns.sf2_1kb.chr18,
hxns.sf2_1kb.chr19,
hxns.sf2_1kb.chr20,
hxns.sf2_1kb.chr21,
hxns.sf2_1kb.chr22,
hxns.sf2_1kb.chr23,
hxns.sf2_1kb.chr24)

#Custom plot
hxns.sf2.data <- hxns.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(hxns.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = hxns.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#Estimate threshold excluding chromosomes containing major inversions:
quantile(hxns.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE)
hxns.sf2.data.noinv = filter(hxns.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(hxns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
hxns.sf2.data.top=filter(hxns.sf2.data, LR > quantile(hxns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
hxns.sf2.data.top2=filter(hxns.sf2.data, LR > quantile(hxns.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))

hxns.clr.plot = ggplot() +
  geom_rect(data = sv.maria.clr, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=760), fill = "steelblue", alpha=0.75) +
  geom_rect(data = qtl.clr, aes(xmin=start, xmax=end, ymin = 770, ymax=780), fill = "gray20", colour = "gray20") +
  geom_point(data = hxns.sf2.data, aes(x=(location/1000000), y= LR),colour = "black", alpha=1, size=2, shape = 20) +
  geom_hline(yintercept = 128.9647, colour="firebrick", linetype="dashed") +
  geom_hline(yintercept = 94.47902, colour="steelblue", linetype="dashed") +
  ylab("CLR") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  scale_y_break(c(600, 700)) +
  facet_grid(~CHR, scales = "free_x") 

#CM manhattan plot
hxns.sf2.data2 <- unite(hxns.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
hxns.sf2.data2.plot <- as.data.frame(hxns.sf2.data2[,c(1,5,2,3)])
names(hxns.sf2.data2.plot) <- c("SNP","chr","pos","LR")
CMplot(hxns.sf2.data2.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", threshold = c(128.9647, 94.47902), threshold.col = c("firebrick","steelblue"), cex = 0.25, height = 2, width = 12, amplify = T,signal.cex = 0.25, signal.col = c("firebrick","steelblue"), file.name = "hxns_clr")


#Plot CLR for major QTL
clr.hxns.chr11 = ggplot() +
  geom_point(data = hxns.sf2_1kb.chr11, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 9000, ymax=9500), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -500, ymax=-50, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, , ymin = 8500, ymax=9000), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(hxns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()
clr.hxns.chr11

clr.hxns.chr13 = ggplot() +
  geom_point(data = hxns.sf2_1kb.chr13, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 200, ymax=250), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -50, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = quantile(hxns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.hxns.chr17 = ggplot() +
  geom_point(data = hxns.sf2_1kb.chr17, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 300, ymax=350), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -50, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = quantile(hxns.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
    geom_hline(yintercept = quantile(hxns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="steelblue", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.hxns.chr18 = ggplot() +
  geom_point(data = hxns.sf2_1kb.chr18, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 520, ymax=530), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -50, ymax=-20, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, , ymin = 500, ymax=510), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(hxns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.hxns.chr24 = ggplot() +
  geom_point(data = hxns.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 2500, ymax=2600), fill = "steelblue3", colour="grey20") +
  xlab("Chromosome 24 [Mb]") +
  ylab("CLR") +
  theme_cowplot()

ggplot() +
  geom_point(data = hxns.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(sv.maria.clr, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 1000, ymax=1100), fill = "steelblue3", colour="grey20") +
  xlab("Chromosome 24 [Mb]") +
  ylab("CLR") +
  theme_cowplot()

```


```{r, eval=F}
mbns.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr01_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr02_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr03_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr04_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr05_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr06_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr07_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr08_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr09_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr10_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr11_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr12_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr13_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr14_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr15_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr16_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr17_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr18_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr19_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr20_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr21_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr22_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr23_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr24_sf2_output.txt",delim = "\t")

mbns.sf2_1kb.chr01$CHR=1
mbns.sf2_1kb.chr02$CHR=2
mbns.sf2_1kb.chr03$CHR=3
mbns.sf2_1kb.chr04$CHR=4
mbns.sf2_1kb.chr05$CHR=5
mbns.sf2_1kb.chr06$CHR=6
mbns.sf2_1kb.chr07$CHR=7
mbns.sf2_1kb.chr08$CHR=8
mbns.sf2_1kb.chr09$CHR=9
mbns.sf2_1kb.chr10$CHR=10
mbns.sf2_1kb.chr11$CHR=11
mbns.sf2_1kb.chr12$CHR=12
mbns.sf2_1kb.chr13$CHR=13
mbns.sf2_1kb.chr14$CHR=14
mbns.sf2_1kb.chr15$CHR=15
mbns.sf2_1kb.chr16$CHR=16
mbns.sf2_1kb.chr17$CHR=17
mbns.sf2_1kb.chr18$CHR=18
mbns.sf2_1kb.chr19$CHR=19
mbns.sf2_1kb.chr20$CHR=20
mbns.sf2_1kb.chr21$CHR=21
mbns.sf2_1kb.chr22$CHR=22
mbns.sf2_1kb.chr23$CHR=23
mbns.sf2_1kb.chr24$CHR=24

mbns.sf2.res=rbind(mbns.sf2_1kb.chr01,
mbns.sf2_1kb.chr02,
mbns.sf2_1kb.chr03,
mbns.sf2_1kb.chr04,
mbns.sf2_1kb.chr05,
mbns.sf2_1kb.chr06,
mbns.sf2_1kb.chr07,
mbns.sf2_1kb.chr08,
mbns.sf2_1kb.chr09,
mbns.sf2_1kb.chr10,
mbns.sf2_1kb.chr11,
mbns.sf2_1kb.chr12,
mbns.sf2_1kb.chr13,
mbns.sf2_1kb.chr14,
mbns.sf2_1kb.chr15,
mbns.sf2_1kb.chr16,
mbns.sf2_1kb.chr17,
mbns.sf2_1kb.chr18,
mbns.sf2_1kb.chr19,
mbns.sf2_1kb.chr20,
mbns.sf2_1kb.chr21,
mbns.sf2_1kb.chr22,
mbns.sf2_1kb.chr23,
mbns.sf2_1kb.chr24)

#Custom plot
mbns.sf2.data <- mbns.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(mbns.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = mbns.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#Estimate threshold excluding chromosomes containing major inversions:
quantile(mbns.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE)
mbns.sf2.data.noinv = filter(mbns.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(mbns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
mbns.sf2.data.top=filter(mbns.sf2.data, LR > quantile(mbns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
mbns.sf2.data.top2=filter(mbns.sf2.data, LR > quantile(mbns.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))


mbns.clr.plot = ggplot(mbns.sf2.data, aes(x=BPcum, y= LR)) +
  geom_point(aes(color=as.factor(CHR)), alpha=1, size=1, shape = 20) +
  geom_point(data = mbns.sf2.data.top2, aes(x=BPcum, y= LR), alpha=1, size=1, shape = 20, colour = "firebrick") +
  scale_color_manual(values = rep(c("grey30","grey60"), 100)) +
  scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) + 
  ylab("CLR") +
  xlab("Chromosome 1 - 24") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())

ggsave2(plot = mbns.clr.plot, filename = "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/output_plots/mbns_sf2_manhattan.png", height=2, width=12)



clr.chr = clr.08 | clr.18 | clr.24


#Plot interpolated recombination rate:
mbns.sf2_1kb=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr18_snps_interpolated_cM_distance.txt",delim = "\t")

ggplot(data=mbns.sf2_1kb, aes(x=position, y=rate)) +
  geom_point()

#CM manhattan plot
mbns.sf2.data2 <- unite(mbns.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
mbns.sf2.data2.plot <- as.data.frame(mbns.sf2.data2[,c(1,5,2,3)])
names(mbns.sf2.data2.plot) <- c("SNP","chr","pos","LR")
CMplot(mbns.sf2.data2.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", threshold = c(141.4989 , 90.9078 ), threshold.col = c("firebrick","steelblue"), cex = 0.25, height = 2, width = 12, amplify = T,signal.cex = 0.25, signal.col = c("firebrick","steelblue"), file.name = "mbns_clr")

#Plot CLR for major QTL
clr.mbns.chr11 = ggplot() +
  geom_point(data = mbns.sf2_1kb.chr11, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 9000, ymax=9500), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -500, ymax=-50, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, , ymin = 8500, ymax=9000), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(mbns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.mbns.chr13 = ggplot() +
  geom_point(data = mbns.sf2_1kb.chr13, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 200, ymax=250), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -50, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = quantile(mbns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.mbns.chr17 = ggplot() +
  geom_point(data = mbns.sf2_1kb.chr17, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, mme_chr == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 300, ymax=350), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -50, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = quantile(mbns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.mbns.chr18 = ggplot() +
  geom_point(data = mbns.sf2_1kb.chr18, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 520, ymax=530), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -50, ymax=-20, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, , ymin = 500, ymax=510), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(mbns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.mbns.chr24 = ggplot() +
  geom_point(data = mbns.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 2700, ymax=2800), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -500, ymax=-50, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 2500, ymax=2600), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(mbns.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

ggplot() +
  geom_point(data = mbns.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(sv.maria.clr, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 1000, ymax=1100), fill = "steelblue3", colour="grey20") +
  xlab("Chromosome 24 [Mb]") +
  ylab("CLR") +
  theme_cowplot()

```


### Plot CLR, genetic differentiation and inversion breakpoints for target chromosomes in JIGA:
```{r, eval=F}
#Marey plots
marey.plot.chr08 <- ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr08", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=250), fill = "gray20", alpha = 0.5) +
  geom_point(data=filter(JIGA.F1, map == "Mme_chr08"), aes(x=(phys), y=gen), size = 0.75, colour = "firebrick") +
  geom_point(data=filter(PANY.F1, map == "Mme_chr08"), aes(x=(phys), y=gen), size = 0.75, colour = "dodgerblue") +
  geom_point(data=filter(PAxJI.F2, map == "Mme_chr08"), aes(x=(phys), y=gen), size = 0.75, colour = "orange") +
  ylim(0,250) +
  ylab("genetic distance [cM]") +
  xlab("physical distance [Bp]") +
  theme(strip.background = element_blank(), 
        strip.placement = "outside") +
  facet_wrap(~ map, scales = "free_x", nrow = 5)

marey.plot.chr17 <- ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr17", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=250), fill = "gray20", alpha = 0.5) +
  geom_point(data=filter(JIGA.F1, map == "Mme_chr17"), aes(x=(phys), y=gen), size = 0.75, colour = "firebrick") +
  geom_point(data=filter(PANY.F1, map == "Mme_chr17"), aes(x=(phys), y=gen), size = 0.75, colour = "dodgerblue") +
  geom_point(data=filter(PAxJI.F2, map == "Mme_chr17"), aes(x=(phys), y=gen), size = 0.75, colour = "orange") +
  ylim(0,250) +
  ylab("genetic distance [cM]") +
  xlab("physical distance [Bp]") +
  theme(strip.background = element_blank(), 
        strip.placement = "outside") +
  facet_wrap(~ map, scales = "free_x", nrow = 5)

marey.plot.chr18 <- ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr18", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=250), fill = "gray20", alpha = 0.5) +
  geom_point(data=filter(JIGA.F1, map == "Mme_chr18"), aes(x=(phys), y=gen), size = 0.75, colour = "firebrick") +
  geom_point(data=filter(PANY.F1, map == "Mme_chr18"), aes(x=(phys), y=gen), size = 0.75, colour = "dodgerblue") +
  geom_point(data=filter(PAxJI.F2, map == "Mme_chr18"), aes(x=(phys), y=gen), size = 0.75, colour = "orange") +
  ylim(0,250) +
  ylab("genetic distance [cM]") +
  xlab("physical distance [Bp]") +
  theme(strip.background = element_blank(), 
        strip.placement = "outside") +
  facet_wrap(~ map, scales = "free_x", nrow = 5)

marey.plot.chr24 <- ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr24", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=250), fill = "gray20", alpha = 0.5) +
  geom_point(data=filter(JIGA.F1, map == "Mme_chr24"), aes(x=(phys), y=gen), size = 0.75, colour = "firebrick") +
  geom_point(data=filter(PANY.F1, map == "Mme_chr24"), aes(x=(phys), y=gen), size = 0.75, colour = "dodgerblue") +
  geom_point(data=filter(PAxJI.F2, map == "Mme_chr24"), aes(x=(phys), y=gen), size = 0.75, colour = "orange") +
  ylim(0,250) +
  ylab("genetic distance [cM]") +
  xlab("physical distance [Bp]") +
  theme(strip.background = element_blank(), 
        strip.placement = "outside") +
  facet_wrap(~ map, scales = "free_x", nrow = 5)


#Genetic differentiation (laod data)
jiga_mbns_fst_scan <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_MBNS_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)
names(jiga_mbns_fst_scan)[1] <- "region"
names(jiga_mbns_fst_scan)[2] <- "chromo"
names(jiga_mbns_fst_scan)[3] <- "midwin"
names(jiga_mbns_fst_scan)[4] <- "nsites"
names(jiga_mbns_fst_scan)[5] <- "fst"

Fst.08 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr08", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=1), fill = "gray20", alpha = 0.5) +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr08"), aes(x=(midwin), y=fst), colour = "black") + 
  ylab("Fst") + 
  labs(title="Fst chr.08", x ="position [Bp]", y = "Fst")

Fst.17 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr17", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=1), fill = "gray20", alpha = 0.5) +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr17"), aes(x=(midwin), y=fst), colour = "black") + 
  ylab("Fst") + 
  labs(title="Fst chr.17", x ="position [Bp]", y = "Fst")

Fst.18 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr18", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=1), fill = "gray20", alpha = 0.5) +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr18"), aes(x=(midwin), y=fst), colour = "black") + 
  ylab("Fst") + 
  labs(title="Fst chr.18", x ="position [Bp]", y = "Fst")

Fst.24 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr24", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=1), fill = "gray20", alpha = 0.5) +
  geom_point(data = filter(jiga_mbns_fst_scan, chromo == "Mme_chr24"), aes(x=(midwin), y=fst), colour = "black") + 
  ylab("Fst") + 
  labs(title="Fst chr.24", x ="position [Bp]", y = "Fst")



#Plot chr08 alone:
mbns.sf2.data2.plot = as.data.frame(mbns.sf2.data[,c(4,1,2,3)])
mbns.sf2.data2.plot.chr08 = filter(mbns.sf2.data2.plot, CHR == 08)
clr.08 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr08"), aes(xmin=(start), xmax=(end), ymin=0, ymax=110), fill = "gray20", alpha = 0.5) +
  geom_point(data = mbns.sf2.data2.plot.chr08, aes(x=(location), y=LR, colour = 1/alpha)) + 
  ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.08", x ="position [Mb]", y = "CLR")

#Plot chr17 alone:
mbns.sf2.data2.plot = as.data.frame(mbns.sf2.data[,c(4,1,2,3)])
mbns.sf2.data2.plot.chr17 = filter(mbns.sf2.data2.plot, CHR == 17)
clr.17 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr17"), aes(xmin=(start), xmax=(end), ymin=0, ymax=300), fill = "gray20", alpha = 0.5) +
  geom_point(data = mbns.sf2.data2.plot.chr17, aes(x=(location), y=LR, colour = 1/alpha)) + 
  ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.17", x ="position [Mb]", y = "CLR")

#Plot chr18 alone:
mbns.sf2.data2.plot = as.data.frame(mbns.sf2.data[,c(4,1,2,3)])
mbns.sf2.data2.plot.chr18 = filter(mbns.sf2.data2.plot, CHR == 18)
clr.18 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr18"), aes(xmin=(start), xmax=(end), ymin=0, ymax=600), fill = "gray20", alpha = 0.5) +
  geom_point(data = mbns.sf2.data2.plot.chr18, aes(x=(location), y=LR, colour = 1/alpha)) + 
  ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.18", x ="position [Mb]", y = "CLR")

#Plot chr24 alone:
mbns.sf2.data2.plot = as.data.frame(mbns.sf2.data[,c(4,1,2,3)])
mbns.sf2.data2.plot.chr24 = filter(mbns.sf2.data2.plot, CHR == 24)
clr.24 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr24", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=400), fill = "gray20", alpha = 0.5) +
  geom_point(data = mbns.sf2.data2.plot.chr24, aes(x=(location), y=LR, colour = 1/alpha)) + 
  ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.24", x ="position [Mb]", y = "CLR")


#Combine all plots
clr.08.marey.mbns = marey.plot.chr08 / Fst.08 / clr.08
clr.17.marey.mbns = marey.plot.chr17 / Fst.17 / clr.17
clr.18.marey.mbns = marey.plot.chr18 / Fst.18 / clr.18
clr.24.marey.mbns = marey.plot.chr24 / Fst.24 / clr.24

clr.chr.mbns = clr.08.marey.mbns | clr.18.marey.mbns | clr.24.marey.mbns
```







```{r, eval=F}
mdme.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr01_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr02_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr03_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr04_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr05_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr06_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr07_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr08_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr09_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr10_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr11_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr12_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr13_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr14_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr15_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr16_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr17_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr18_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr19_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr20_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr21_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr22_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr23_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr24_sf2_output.txt",delim = "\t")

mdme.sf2_1kb.chr01$CHR=1
mdme.sf2_1kb.chr02$CHR=2
mdme.sf2_1kb.chr03$CHR=3
mdme.sf2_1kb.chr04$CHR=4
mdme.sf2_1kb.chr05$CHR=5
mdme.sf2_1kb.chr06$CHR=6
mdme.sf2_1kb.chr07$CHR=7
mdme.sf2_1kb.chr08$CHR=8
mdme.sf2_1kb.chr09$CHR=9
mdme.sf2_1kb.chr10$CHR=10
mdme.sf2_1kb.chr11$CHR=11
mdme.sf2_1kb.chr12$CHR=12
mdme.sf2_1kb.chr13$CHR=13
mdme.sf2_1kb.chr14$CHR=14
mdme.sf2_1kb.chr15$CHR=15
mdme.sf2_1kb.chr16$CHR=16
mdme.sf2_1kb.chr17$CHR=17
mdme.sf2_1kb.chr18$CHR=18
mdme.sf2_1kb.chr19$CHR=19
mdme.sf2_1kb.chr20$CHR=20
mdme.sf2_1kb.chr21$CHR=21
mdme.sf2_1kb.chr22$CHR=22
mdme.sf2_1kb.chr23$CHR=23
mdme.sf2_1kb.chr24$CHR=24

mdme.sf2.res=rbind(mdme.sf2_1kb.chr01,
mdme.sf2_1kb.chr02,
mdme.sf2_1kb.chr03,
mdme.sf2_1kb.chr04,
mdme.sf2_1kb.chr05,
mdme.sf2_1kb.chr06,
mdme.sf2_1kb.chr07,
mdme.sf2_1kb.chr08,
mdme.sf2_1kb.chr09,
mdme.sf2_1kb.chr10,
mdme.sf2_1kb.chr11,
mdme.sf2_1kb.chr12,
mdme.sf2_1kb.chr13,
mdme.sf2_1kb.chr14,
mdme.sf2_1kb.chr15,
mdme.sf2_1kb.chr16,
mdme.sf2_1kb.chr17,
mdme.sf2_1kb.chr18,
mdme.sf2_1kb.chr19,
mdme.sf2_1kb.chr20,
mdme.sf2_1kb.chr21,
mdme.sf2_1kb.chr22,
mdme.sf2_1kb.chr23,
mdme.sf2_1kb.chr24)

#Custom plot
mdme.sf2.data <- mdme.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(mdme.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = mdme.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#Estimate threshold excluding chromosomes containing major inversions:
quantile(mdme.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE)
mdme.sf2.data.noinv = filter(mdme.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(mdme.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
mdme.sf2.data.top=filter(mdme.sf2.data, LR > quantile(mdme.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
mdme.sf2.data.top2=filter(mdme.sf2.data, LR > quantile(mdme.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))


mdme.clr.plot = ggplot(mdme.sf2.data, aes(x=BPcum, y= LR)) +
  geom_point(aes(color=as.factor(CHR)), alpha=1, size=1, shape = 20) +
  geom_point(data = mdme.sf2.data.top2, aes(x=BPcum, y= LR), alpha=1, size=1, shape = 20, colour = "firebrick") +
  scale_color_manual(values = rep(c("grey30","grey60"), 100)) +
  scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) + 
  ylab("CLR") +
  xlab("Chromosome 1 - 24") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())

ggsave2(plot = mdme.clr.plot, filename = "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/output_plots/mdme_sf2_manhattan.png", height=2, width=12)

#CM manhattan plot
mdme.sf2.data2 <- unite(mdme.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
mdme.sf2.data2.plot <- as.data.frame(mdme.sf2.data2[,c(1,5,2,3)])
names(mdme.sf2.data2.plot) <- c("SNP","chr","pos","LR")
CMplot(mdme.sf2.data2.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", threshold = c(987.3352  , 104.8876 ), threshold.col = c("firebrick","steelblue"), cex = 0.25, height = 2, width = 12, amplify = T,signal.cex = 0.25, signal.col = c("firebrick","steelblue"), file.name = "mdme_clr")


ggplot() +
  geom_point(data = mdme.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(sv.maria.clr, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 2300, ymax=2400), fill = "steelblue3", colour="grey20") +
  xlab("Chromosome 24 [Mb]") +
  ylab("CLR") +
  theme_cowplot()
```



```{r, eval=F}
pany.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr01_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr02_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr03_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr04_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr05_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr06_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr07_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr08_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr09_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr10_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr11_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr12_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr13_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr14_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr15_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr16_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr17_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr18_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr19_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr20_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr21_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr22_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr23_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr24_sf2_output.txt",delim = "\t")

pany.sf2_1kb.chr01$CHR=1
pany.sf2_1kb.chr02$CHR=2
pany.sf2_1kb.chr03$CHR=3
pany.sf2_1kb.chr04$CHR=4
pany.sf2_1kb.chr05$CHR=5
pany.sf2_1kb.chr06$CHR=6
pany.sf2_1kb.chr07$CHR=7
pany.sf2_1kb.chr08$CHR=8
pany.sf2_1kb.chr09$CHR=9
pany.sf2_1kb.chr10$CHR=10
pany.sf2_1kb.chr11$CHR=11
pany.sf2_1kb.chr12$CHR=12
pany.sf2_1kb.chr13$CHR=13
pany.sf2_1kb.chr14$CHR=14
pany.sf2_1kb.chr15$CHR=15
pany.sf2_1kb.chr16$CHR=16
pany.sf2_1kb.chr17$CHR=17
pany.sf2_1kb.chr18$CHR=18
pany.sf2_1kb.chr19$CHR=19
pany.sf2_1kb.chr20$CHR=20
pany.sf2_1kb.chr21$CHR=21
pany.sf2_1kb.chr22$CHR=22
pany.sf2_1kb.chr23$CHR=23
pany.sf2_1kb.chr24$CHR=24

pany.sf2.res=rbind(pany.sf2_1kb.chr01,
pany.sf2_1kb.chr02,
pany.sf2_1kb.chr03,
pany.sf2_1kb.chr04,
pany.sf2_1kb.chr05,
pany.sf2_1kb.chr06,
pany.sf2_1kb.chr07,
pany.sf2_1kb.chr08,
pany.sf2_1kb.chr09,
pany.sf2_1kb.chr10,
pany.sf2_1kb.chr11,
pany.sf2_1kb.chr12,
pany.sf2_1kb.chr13,
pany.sf2_1kb.chr14,
pany.sf2_1kb.chr15,
pany.sf2_1kb.chr16,
pany.sf2_1kb.chr17,
pany.sf2_1kb.chr18,
pany.sf2_1kb.chr19,
pany.sf2_1kb.chr20,
pany.sf2_1kb.chr21,
pany.sf2_1kb.chr22,
pany.sf2_1kb.chr23,
pany.sf2_1kb.chr24)

#Custom plot
pany.sf2.data <- pany.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(pany.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = pany.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#Estimate threshold excluding chromosomes containing major inversions:
quantile(pany.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE)
pany.sf2.data.noinv = filter(pany.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(pany.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
pany.sf2.data.top=filter(pany.sf2.data, LR > quantile(pany.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
pany.sf2.data.top2=filter(pany.sf2.data, LR > quantile(pany.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))

pany.clr.plot = ggplot(pany.sf2.data, aes(x=BPcum, y= LR)) +
  geom_point(aes(color=as.factor(CHR)), alpha=1, size=1, shape = 20) +
  geom_point(data = pany.sf2.data.top2, aes(x=BPcum, y= LR), alpha=1, size=1, shape = 20, colour = "firebrick") +
  scale_color_manual(values = rep(c("grey30","grey60"), 100)) +
  scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) + 
  ylab("CLR") +
  xlab("Chromosome 1 - 24") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())

ggsave2(plot = pany.clr.plot, filename = "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/output_plots/pany_sf2_manhattan.png", height=2, width=12)

#Plot interpolated recombination rate:
pany.sf2_1kb=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr18_snps_interpolated_cM_distance.txt",delim = "\t")

ggplot(data=pany.sf2_1kb, aes(x=position, y=rate)) +
  geom_point()

pany.sf2.data2 <- unite(pany.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
pany.sf2.data2.plot <- as.data.frame(pany.sf2.data2[,c(1,5,2,3)])
names(pany.sf2.data2.plot) <- c("SNP","chr","pos","LR")
CMplot(pany.sf2.data2.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", threshold = c(429.9185   , 79.82179), threshold.col = c("firebrick","steelblue"), cex = 0.25, height = 2, width = 12, amplify = T,signal.cex = 0.25, signal.col = c("firebrick","steelblue"), file.name = "pany_clr")


ggplot() +
  geom_point(data = pany.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(sv.maria.clr, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 2300, ymax=2400), fill = "steelblue3", colour="grey20") +
  xlab("Chromosome 24 [Mb]") +
  ylab("CLR") +
  theme_cowplot()
```

### Plot CLR, genetic differentiation and inversion breakpoints for target chromosomes in PANY:
```{r, eval=F}
#Marey plots
marey.plot.chr18 <- ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr18", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=250), fill = "gray20", alpha = 0.5) +
  geom_point(data=filter(JIGA.F1, map == "Mme_chr18"), aes(x=(phys), y=gen), size = 0.75, colour = "firebrick") +
  geom_point(data=filter(PANY.F1, map == "Mme_chr18"), aes(x=(phys), y=gen), size = 0.75, colour = "dodgerblue") +
  geom_point(data=filter(PAxJI.F2, map == "Mme_chr18"), aes(x=(phys), y=gen), size = 0.75, colour = "orange") +
  ylim(0,250) +
  ylab("genetic distance [cM]") +
  xlab("physical distance [Bp]") +
  theme(strip.background = element_blank(), 
        strip.placement = "outside") +
  facet_wrap(~ map, scales = "free_x", nrow = 5)

marey.plot.chr24 <- ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr24", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=250), fill = "gray20", alpha = 0.5) +
  geom_point(data=filter(JIGA.F1, map == "Mme_chr24"), aes(x=(phys), y=gen), size = 0.75, colour = "firebrick") +
  geom_point(data=filter(PANY.F1, map == "Mme_chr24"), aes(x=(phys), y=gen), size = 0.75, colour = "dodgerblue") +
  geom_point(data=filter(PAxJI.F2, map == "Mme_chr24"), aes(x=(phys), y=gen), size = 0.75, colour = "orange") +
  ylim(0,250) +
  ylab("genetic distance [cM]") +
  xlab("physical distance [Bp]") +
  theme(strip.background = element_blank(), 
        strip.placement = "outside") +
  facet_wrap(~ map, scales = "free_x", nrow = 5)


#Genetic differentiation (laod data)
jiga_pany_fst_scan <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_PANY_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)
names(jiga_pany_fst_scan)[1] <- "region"
names(jiga_pany_fst_scan)[2] <- "chromo"
names(jiga_pany_fst_scan)[3] <- "midwin"
names(jiga_pany_fst_scan)[4] <- "nsites"
names(jiga_pany_fst_scan)[5] <- "fst"

Fst.18 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr18", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=1), fill = "gray20", alpha = 0.5) +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr18"), aes(x=(midwin), y=fst), colour = "black") + 
  ylab("Fst") + 
  labs(title="Fst chr.18", x ="position [Bp]", y = "Fst")

Fst.24 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr24", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=1), fill = "gray20", alpha = 0.5) +
  geom_point(data = filter(jiga_pany_fst_scan, chromo == "Mme_chr24"), aes(x=(midwin), y=fst), colour = "black") + 
  ylab("Fst") + 
  labs(title="Fst chr.24", x ="position [Bp]", y = "Fst")



#Plot chr18 alone:
pany.sf2.data2.plot = as.data.frame(pany.sf2.data[,c(4,1,2,3)])
pany.sf2.data2.plot.chr18 = filter(pany.sf2.data2.plot, CHR == 18)
clr.18 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr18"), aes(xmin=(start), xmax=(end), ymin=0, ymax=1100), fill = "gray20", alpha = 0.5) +
  geom_point(data = pany.sf2.data2.plot.chr18, aes(x=(location), y=LR, colour = 1/alpha)) + 
  ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.18", x ="position [Mb]", y = "CLR")



#Plot chr24 alone:
pany.sf2.data2.plot = as.data.frame(pany.sf2.data[,c(4,1,2,3)])
pany.sf2.data2.plot.chr24 = filter(pany.sf2.data2.plot, CHR == 24)
clr.24 = ggplot() +
  geom_rect(data = filter(SVs.plot, CHR == "Mme_chr24", inv.type == "major"), aes(xmin=(start), xmax=(end), ymin=0, ymax=2000), fill = "gray20", alpha = 0.5) +
  geom_point(data = pany.sf2.data2.plot.chr24, aes(x=(location), y=LR, colour = 1/alpha)) + 
  ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.24", x ="position [Mb]", y = "CLR")


#Combine all plots
clr.18.marey = marey.plot.chr18 / Fst.18 / clr.18
clr.24.marey = marey.plot.chr24 / Fst.24 / clr.24

clr.chr = clr.18.marey | clr.24.marey
```

```{r, eval=F}
oinc.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr01_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr02_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr03_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr04_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr05_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr06_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr07_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr08_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr09_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr10_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr11_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr12_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr13_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr14_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr15_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr16_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr17_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr18_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr19_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr20_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr21_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr22_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr23_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr24_sf2_output.txt",delim = "\t")

oinc.sf2_1kb.chr01$CHR=1
oinc.sf2_1kb.chr02$CHR=2
oinc.sf2_1kb.chr03$CHR=3
oinc.sf2_1kb.chr04$CHR=4
oinc.sf2_1kb.chr05$CHR=5
oinc.sf2_1kb.chr06$CHR=6
oinc.sf2_1kb.chr07$CHR=7
oinc.sf2_1kb.chr08$CHR=8
oinc.sf2_1kb.chr09$CHR=9
oinc.sf2_1kb.chr10$CHR=10
oinc.sf2_1kb.chr11$CHR=11
oinc.sf2_1kb.chr12$CHR=12
oinc.sf2_1kb.chr13$CHR=13
oinc.sf2_1kb.chr14$CHR=14
oinc.sf2_1kb.chr15$CHR=15
oinc.sf2_1kb.chr16$CHR=16
oinc.sf2_1kb.chr17$CHR=17
oinc.sf2_1kb.chr18$CHR=18
oinc.sf2_1kb.chr19$CHR=19
oinc.sf2_1kb.chr20$CHR=20
oinc.sf2_1kb.chr21$CHR=21
oinc.sf2_1kb.chr22$CHR=22
oinc.sf2_1kb.chr23$CHR=23
oinc.sf2_1kb.chr24$CHR=24

oinc.sf2.res=rbind(oinc.sf2_1kb.chr01,
oinc.sf2_1kb.chr02,
oinc.sf2_1kb.chr03,
oinc.sf2_1kb.chr04,
oinc.sf2_1kb.chr05,
oinc.sf2_1kb.chr06,
oinc.sf2_1kb.chr07,
oinc.sf2_1kb.chr08,
oinc.sf2_1kb.chr09,
oinc.sf2_1kb.chr10,
oinc.sf2_1kb.chr11,
oinc.sf2_1kb.chr12,
oinc.sf2_1kb.chr13,
oinc.sf2_1kb.chr14,
oinc.sf2_1kb.chr15,
oinc.sf2_1kb.chr16,
oinc.sf2_1kb.chr17,
oinc.sf2_1kb.chr18,
oinc.sf2_1kb.chr19,
oinc.sf2_1kb.chr20,
oinc.sf2_1kb.chr21,
oinc.sf2_1kb.chr22,
oinc.sf2_1kb.chr23,
oinc.sf2_1kb.chr24)

#Custom plot
oinc.sf2.data <- oinc.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(oinc.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = oinc.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#Estimate threshold excluding chromosomes containing major inversions:
quantile(oinc.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE)
oinc.sf2.data.noinv = filter(oinc.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(oinc.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
oinc.sf2.data.top=filter(oinc.sf2.data, LR > quantile(oinc.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
oinc.sf2.data.top2=filter(oinc.sf2.data, LR > quantile(oinc.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))

ggplot(oinc.sf2.data, aes(x=BPcum, y= LR)) +
  geom_point(aes(color=as.factor(CHR)), alpha=1, size=1, shape = 20) +
  scale_color_manual(values = rep(c("grey30","grey60"), 100)) +
  scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) + 
  ylab("CLR") +
  xlab("Chromosome 1 - 24") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(oinc.sf2.data, aes(x=BPcum, y= LR)) +
  geom_point(aes(color=as.factor(CHR)), alpha=1, size=1, shape = 20) +
  geom_point(data = oinc.sf2.data.top, aes(x=BPcum, y= LR), alpha=1, size=1, shape = 20, colour = "steelblue") +
    geom_point(data = oinc.sf2.data.top2, aes(x=BPcum, y= LR), alpha=1, size=1, shape = 20, colour = "firebrick") +
  scale_color_manual(values = rep(c("grey30","grey60"), 100)) +
  scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) + 
  ylab("CLR") +
  xlab("Chromosome 1 - 24") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1))

#CM manhattan plot
oinc.sf2.data2 <- unite(oinc.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
oinc.sf2.data2.plot <- as.data.frame(oinc.sf2.data2[,c(1,5,2,3)])
names(oinc.sf2.data2.plot) <- c("SNP","chr","pos","LR")
CMplot(oinc.sf2.data2.plot, type="h",plot.type="m",col=c("grey30","grey60"), band=0.1, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", )



#Plot individual chromosomes:
#Plot chr08 alone:
oinc.sf2.data2.plot = as.data.frame(oinc.sf2.data[,c(4,1,2,3)])
oinc.sf2.data2.plot.chr8 = filter(oinc.sf2.data2.plot, CHR == 8)
clr.08 = ggplot(data = oinc.sf2.data2.plot.chr8, aes(x=(location/1000000), y=LR, colour = 1/alpha)) + geom_point() + ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.08", x ="position [Mb]", y = "CLR")

#Plot chr11 alone:
oinc.sf2.data2.plot = as.data.frame(oinc.sf2.data[,c(4,1,2,3)])
oinc.sf2.data2.plot.chr18 = filter(oinc.sf2.data2.plot, CHR == 18)
clr.18 = ggplot(data = oinc.sf2.data2.plot.chr18, aes(x=(location/1000000), y=LR, colour = 1/alpha)) + geom_point() + ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.18", x ="position [Mb]", y = "CLR")

#Plot chr24 alone:
oinc.sf2.data2.plot = as.data.frame(oinc.sf2.data[,c(4,1,2,3)])
oinc.sf2.data2.plot.chr24 = filter(oinc.sf2.data2.plot, CHR == 24)
clr.24 = ggplot(data = oinc.sf2.data2.plot.chr24, aes(x=(location/1000000), y=LR, colour = 1/alpha)) + geom_point() + ylab("CLR") + scale_color_viridis_c() +labs(title="Sweep chr.24", x ="position [Mb]", y = "CLR")


clr.chr = clr.08 | clr.18 | clr.24

oinc.sf2.data2 <- unite(oinc.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
oinc.sf2.data2.plot <- as.data.frame(oinc.sf2.data2[,c(1,5,2,3)])
names(oinc.sf2.data2.plot) <- c("SNP","chr","pos","LR")
CMplot(oinc.sf2.data2.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", threshold = c(277.2311, 163.2037 ), threshold.col = c("firebrick","steelblue"), cex = 0.25, height = 2, width = 12, amplify = T,signal.cex = 0.25, signal.col = c("firebrick","steelblue"), file.name = "oinc_clr")
```



```{r, eval=F}
jiga.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr01_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr02_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr03_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr04_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr05_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr06_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr07_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr08_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr09_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr10_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr11_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr12_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr13_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr14_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr15_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr16_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr17_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr18_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr19_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr20_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr21_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr22_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr23_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr24_sf2_output.txt",delim = "\t")

jiga.sf2_1kb.chr01$CHR=1
jiga.sf2_1kb.chr02$CHR=2
jiga.sf2_1kb.chr03$CHR=3
jiga.sf2_1kb.chr04$CHR=4
jiga.sf2_1kb.chr05$CHR=5
jiga.sf2_1kb.chr06$CHR=6
jiga.sf2_1kb.chr07$CHR=7
jiga.sf2_1kb.chr08$CHR=8
jiga.sf2_1kb.chr09$CHR=9
jiga.sf2_1kb.chr10$CHR=10
jiga.sf2_1kb.chr11$CHR=11
jiga.sf2_1kb.chr12$CHR=12
jiga.sf2_1kb.chr13$CHR=13
jiga.sf2_1kb.chr14$CHR=14
jiga.sf2_1kb.chr15$CHR=15
jiga.sf2_1kb.chr16$CHR=16
jiga.sf2_1kb.chr17$CHR=17
jiga.sf2_1kb.chr18$CHR=18
jiga.sf2_1kb.chr19$CHR=19
jiga.sf2_1kb.chr20$CHR=20
jiga.sf2_1kb.chr21$CHR=21
jiga.sf2_1kb.chr22$CHR=22
jiga.sf2_1kb.chr23$CHR=23
jiga.sf2_1kb.chr24$CHR=24

jiga.sf2.res=rbind(jiga.sf2_1kb.chr01,
jiga.sf2_1kb.chr02,
jiga.sf2_1kb.chr03,
jiga.sf2_1kb.chr04,
jiga.sf2_1kb.chr05,
jiga.sf2_1kb.chr06,
jiga.sf2_1kb.chr07,
jiga.sf2_1kb.chr08,
jiga.sf2_1kb.chr09,
jiga.sf2_1kb.chr10,
jiga.sf2_1kb.chr11,
jiga.sf2_1kb.chr12,
jiga.sf2_1kb.chr13,
jiga.sf2_1kb.chr14,
jiga.sf2_1kb.chr15,
jiga.sf2_1kb.chr16,
jiga.sf2_1kb.chr17,
jiga.sf2_1kb.chr18,
jiga.sf2_1kb.chr19,
jiga.sf2_1kb.chr20,
jiga.sf2_1kb.chr21,
jiga.sf2_1kb.chr22,
jiga.sf2_1kb.chr23,
jiga.sf2_1kb.chr24)

#Custom plot
jiga.sf2.data <- jiga.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(jiga.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = jiga.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#Estimate threshold excluding chromosomes containing major inversions:
quantile(jiga.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE)
jiga.sf2.data.noinv = filter(jiga.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
jiga.sf2.data.top=filter(jiga.sf2.data, LR > quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
jiga.sf2.data.top2=filter(jiga.sf2.data, LR > quantile(jiga.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))

jiga.clr.plot = ggplot(jiga.sf2.data, aes(x=BPcum, y= LR)) +
  geom_point(aes(color=as.factor(CHR)), alpha=1, size=1, shape = 20) +
  geom_point(data = jiga.sf2.data.top2, aes(x=BPcum, y= LR), alpha=1, size=1, shape = 20, colour = "firebrick") +
  scale_color_manual(values = rep(c("grey30","grey60"), 100)) +
  scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) + 
  ylab("CLR") +
  xlab("Chromosome 1 - 24") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())

ggsave2(plot = jiga.clr.plot, filename = "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/output_plots/jiga_sf2_manhattan.png", height=2, width=12)

jiga.sf2.data2 <- unite(jiga.sf2.data, "SNP", c("CHR","location"),sep = ":", remove = F)
jiga.sf2.data2.plot <- as.data.frame(jiga.sf2.data2[,c(1,5,2,3)])
names(jiga.sf2.data2.plot) <- c("SNP","chr","pos","LR")
CMplot(jiga.sf2.data2.plot, type="p",plot.type="m",col=c("grey30","grey60"), band=0.5, LOG10=FALSE, ylab="CLR", file.output = T, file = "pdf", threshold = c(1687.852 , 108.4838), threshold.col = c("firebrick","steelblue"), cex = 0.25, height = 2, width = 12, amplify = T,signal.cex = 0.25, signal.col = c("firebrick","steelblue"), file.name = "jiga_clr")

ggplot() +
  geom_point(data = jiga.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(sv.maria.clr, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 11100, ymax=11300), fill = "steelblue3", colour="grey20") +
  xlab("Chromosome 24 [Mb]") +
  ylab("CLR") +
  theme_cowplot()
```


Without recombination rate grid
```{r, eval=F}
jiga.sf2_1kb.chr01=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr01_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr02=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr02_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr03=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr03_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr04_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr05=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr05_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr06=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr06_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr07=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr07_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr08=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr08_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr09=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr09_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr10=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr10_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr11_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr12=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr12_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr13_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr14=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr14_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr15=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr15_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr16=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr16_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr17_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr18_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr19=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr19_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr20=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr20_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr21=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr21_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr22=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr22_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr23=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr23_sf2_norec_output.txt",delim = "\t")
jiga.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_out_norec/jiga_chr24_sf2_norec_output.txt",delim = "\t")

jiga.sf2_1kb.chr01$CHR=1
jiga.sf2_1kb.chr02$CHR=2
jiga.sf2_1kb.chr03$CHR=3
jiga.sf2_1kb.chr04$CHR=4
jiga.sf2_1kb.chr05$CHR=5
jiga.sf2_1kb.chr06$CHR=6
jiga.sf2_1kb.chr07$CHR=7
jiga.sf2_1kb.chr08$CHR=8
jiga.sf2_1kb.chr09$CHR=9
jiga.sf2_1kb.chr10$CHR=10
jiga.sf2_1kb.chr11$CHR=11
jiga.sf2_1kb.chr12$CHR=12
jiga.sf2_1kb.chr13$CHR=13
jiga.sf2_1kb.chr14$CHR=14
jiga.sf2_1kb.chr15$CHR=15
jiga.sf2_1kb.chr16$CHR=16
jiga.sf2_1kb.chr17$CHR=17
jiga.sf2_1kb.chr18$CHR=18
jiga.sf2_1kb.chr19$CHR=19
jiga.sf2_1kb.chr20$CHR=20
jiga.sf2_1kb.chr21$CHR=21
jiga.sf2_1kb.chr22$CHR=22
jiga.sf2_1kb.chr23$CHR=23
jiga.sf2_1kb.chr24$CHR=24

jiga.sf2.res=rbind(jiga.sf2_1kb.chr01,
jiga.sf2_1kb.chr02,
jiga.sf2_1kb.chr03,
jiga.sf2_1kb.chr04,
jiga.sf2_1kb.chr05,
jiga.sf2_1kb.chr06,
jiga.sf2_1kb.chr07,
jiga.sf2_1kb.chr08,
jiga.sf2_1kb.chr09,
jiga.sf2_1kb.chr10,
jiga.sf2_1kb.chr11,
jiga.sf2_1kb.chr12,
jiga.sf2_1kb.chr13,
jiga.sf2_1kb.chr14,
jiga.sf2_1kb.chr15,
jiga.sf2_1kb.chr16,
jiga.sf2_1kb.chr17,
jiga.sf2_1kb.chr18,
jiga.sf2_1kb.chr19,
jiga.sf2_1kb.chr20,
jiga.sf2_1kb.chr21,
jiga.sf2_1kb.chr22,
jiga.sf2_1kb.chr23,
jiga.sf2_1kb.chr24)

#Custom plot
jiga.sf2.data <- jiga.sf2.res %>% 
  # Compute chromosome size
  group_by(CHR) %>% 
  summarise(chr_len=max(location)) %>% 
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  # Add this info to the initial dataset
  left_join(jiga.sf2.res, ., by=c("CHR"="CHR")) %>%
  # Add a cumulative position of each SNP
  arrange(CHR, location) %>%
  mutate(BPcum=location+tot)
axisdf = jiga.sf2.data %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#Estimate threshold excluding chromosomes containing major inversions:
jiga.sf2.data.noinv = filter(jiga.sf2.data, CHR != 4, CHR != 7, CHR != 8, CHR != 11, CHR != 18, CHR != 19, CHR != 24)
quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE)
jiga.sf2.data.top=filter(jiga.sf2.data, LR > quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE))
jiga.sf2.data.top2=filter(jiga.sf2.data, LR > quantile(jiga.sf2.data$LR, prob = 1 - 1/100, na.rm = TRUE))

jiga.clr.plot = ggplot(jiga.sf2.data, aes(x=BPcum, y= LR)) +
  geom_point(aes(color=as.factor(CHR)), alpha=1, size=1, shape = 20) +
  geom_point(data = jiga.sf2.data.top2, aes(x=BPcum, y= LR), alpha=1, size=1, shape = 20, colour = "firebrick") +
  scale_color_manual(values = rep(c("grey30","grey60"), 100)) +
  scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +
  scale_y_continuous(expand = c(0, 0) ) + 
  ylab("CLR") +
  xlab("Chromosome 1 - 24") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_line(colour = "black", size = 0.5),
    axis.ticks.x = element_line(colour = "black", size = 0.5),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())

ggsave(plot = jiga.clr.plot, filename = "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/output_plots/jiga_sf2_manhattan_norec.png", height=2, width=12)


#Plot CLR for major QTL
clr.jiga.chr11 = ggplot() +
  geom_point(data = jiga.sf2_1kb.chr11, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 150, ymax=160), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -20, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, , ymin = 130, ymax=140), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.jiga.chr13 = ggplot() +
  geom_point(data = jiga.sf2_1kb.chr13, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 200, ymax=250), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -50, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.jiga.chr17 = ggplot() +
  geom_point(data = jiga.sf2_1kb.chr17, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 300, ymax=350), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -50, ymax=-10, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_hline(yintercept = quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.jiga.chr18 = ggplot() +
  geom_point(data = jiga.sf2_1kb.chr18, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB, ymin = 15000, ymax=15500), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -1500, ymax=-400, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, , ymin = 14500, ymax=15000), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()

clr.jiga.chr24 = ggplot() +
  geom_point(data = jiga.sf2_1kb.chr24, aes(x=location/1000000 , y=LR, colour=(1/alpha))) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 14000, ymax=14500), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, Chr == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -1000, ymax=-100, fill = Trait), alpha = 0.5, colour = "gray20") +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, , ymin = 13000, ymax=13500), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = quantile(jiga.sf2.data.noinv$LR, prob = 1 - 1/100, na.rm = TRUE), colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("CLR") +
  theme_cowplot()
```



## Plot CLR in QTL regions:

```{r, eval=F}
#chr11
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=6000), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(jiga.sf2.res, CHR == "11"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#4F000B") +
  geom_smooth(data = filter(oinc.sf2.res, CHR == "11"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#EE7600") +
  geom_smooth(data = filter(pany.sf2.res, CHR == "11"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#FFC100") +
  geom_smooth(data = filter(mdme.sf2.res, CHR == "11"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#0F85A0") +
  geom_smooth(data = filter(mbns.sf2.res, CHR == "11"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#05668D") +
  geom_smooth(data = filter(maqu.sf2.res, CHR == "11"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#3D5A80") +
  ylab("CLR") +
  xlab("position [Mb]") +
  theme_cowplot()


#chr13
ggplot() +
  geom_smooth(data = filter(jiga.sf2.res, CHR == "13"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#4F000B") +
  geom_smooth(data = filter(oinc.sf2.res, CHR == "13"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#EE7600") +
  geom_smooth(data = filter(pany.sf2.res, CHR == "13"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#FFC100") +
  geom_smooth(data = filter(mdme.sf2.res, CHR == "13"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#0F85A0") +
  geom_smooth(data = filter(mbns.sf2.res, CHR == "13"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#05668D") +
  geom_smooth(data = filter(maqu.sf2.res, CHR == "13"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#3D5A80") +
  ylab("CLR") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr17
ggplot() +
  geom_smooth(data = filter(jiga.sf2.res, CHR == "17"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#4F000B") +
  geom_smooth(data = filter(oinc.sf2.res, CHR == "17"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#EE7600") +
  geom_smooth(data = filter(pany.sf2.res, CHR == "17"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#FFC100") +
  geom_smooth(data = filter(mdme.sf2.res, CHR == "17"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#0F85A0") +
  geom_smooth(data = filter(mbns.sf2.res, CHR == "17"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#05668D") +
  geom_smooth(data = filter(maqu.sf2.res, CHR == "17"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#3D5A80") +
  ylab("CLR") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr18
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=12000), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(jiga.sf2.res, CHR == "18"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#4F000B") +
  geom_smooth(data = filter(oinc.sf2.res, CHR == "18"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#EE7600") +
  geom_smooth(data = filter(pany.sf2.res, CHR == "18"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#FFC100") +
  geom_smooth(data = filter(mdme.sf2.res, CHR == "18"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#0F85A0") +
  geom_smooth(data = filter(mbns.sf2.res, CHR == "18"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#05668D") +
  geom_smooth(data = filter(maqu.sf2.res, CHR == "18"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#3D5A80") +
  ylab("CLR") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr24
ggplot() +
    geom_rect(data = filter(sv.maria, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=5000), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(jiga.sf2.res, CHR == "24"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#4F000B") +
  geom_smooth(data = filter(oinc.sf2.res, CHR == "24"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#EE7600") +
  geom_smooth(data = filter(pany.sf2.res, CHR == "24"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#FFC100") +
  geom_smooth(data = filter(mdme.sf2.res, CHR == "24"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#0F85A0") +
  geom_smooth(data = filter(mbns.sf2.res, CHR == "24"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#05668D") +
  geom_smooth(data = filter(maqu.sf2.res, CHR == "24"), aes(x=(location/1000000), y=LR), method = "loess", se = F, span=0.05, colour = "#3D5A80") +
  ylab("CLR") +
  xlab("position [Mb]") +
  theme_cowplot()
```

Chromosome 4
```{r, eval=F}
qtl = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint.csv")
qtl.04 = filter(qtl, chr == 4)

jiga.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr04_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr04$Chr = 04

oinc.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr04_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr04$Chr = 04

pany.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr04_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr04$Chr = 04

mdme.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr04_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr04$Chr = 04

mbns.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr04_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr04$Chr = 04

maqu.sf2_1kb.chr04=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr04_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr04$Chr = 04


jiga.clr.04 = ggplot() + geom_point(data = jiga.sf2_1kb.chr04, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1687.852 , colour = "firebrick") + geom_hline(yintercept = 108.04838 , colour = "steelblue")
oinc.clr.04 = ggplot() + geom_point(data = oinc.sf2_1kb.chr04, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 277.2311, colour = "firebrick") + geom_hline(yintercept = 163.2037, colour = "steelblue")
pany.clr.04 = ggplot() + geom_point(data = pany.sf2_1kb.chr04, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 0429.9045 , colour = "firebrick") + geom_hline(yintercept = 79.82179 , colour = "steelblue")
mdme.clr.04 = ggplot() + geom_point(data = mdme.sf2_1kb.chr04, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 987.3352  , colour = "firebrick") + geom_hline(yintercept = 104.8876  , colour = "steelblue")
mbns.clr.04 = ggplot() + geom_point(data = mbns.sf2_1kb.chr04, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1041.04989, colour = "firebrick") + geom_hline(yintercept = 90.9078, colour = "steelblue")
maqu.clr.04 = ggplot() + geom_point(data = maqu.sf2_1kb.chr04, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 952.8804, colour = "firebrick") + geom_hline(yintercept = 179.90405, colour = "steelblue")
qtl.04.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr04, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.04, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.04 = qtl.04.plot / jiga.clr.04 / oinc.clr.04 / pany.clr.04 / mdme.clr.04 /mbns.clr.04 / maqu.clr.04

jiga.clr.04 = ggplot() + geom_point(data = jiga.sf2_1kb.chr04, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
oinc.clr.04 = ggplot() + geom_point(data = oinc.sf2_1kb.chr04, aes(x=(location/1000000), y=LR, colour=1/alpha))
pany.clr.04 = ggplot() + geom_point(data = pany.sf2_1kb.chr04, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mdme.clr.04 = ggplot() + geom_point(data = mdme.sf2_1kb.chr04, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mbns.clr.04 = ggplot() + geom_point(data = mbns.sf2_1kb.chr04, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
maqu.clr.04 = ggplot() + geom_point(data = maqu.sf2_1kb.chr04, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
qtl.04.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr04, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.04, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.04 = qtl.04.plot / jiga.clr.04 / oinc.clr.04 / pany.clr.04 / mdme.clr.04 /mbns.clr.04 / maqu.clr.04
```

Chromosome11
```{r, eval=F}
qtl = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint.csv")
qtl.11 = filter(qtl, chr == 11)

jiga.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr11_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr11$Chr = 11

oinc.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr11_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr11$Chr = 11

pany.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr11_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr11$Chr = 11

mdme.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr11_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr11$Chr = 11

mbns.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr11_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr11$Chr = 11

maqu.sf2_1kb.chr11=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr11_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr11$Chr = 11


jiga.clr.11 = ggplot() + geom_point(data = jiga.sf2_1kb.chr11, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1687.852 , colour = "firebrick") + geom_hline(yintercept = 108.11838 , colour = "steelblue")
oinc.clr.11 = ggplot() + geom_point(data = oinc.sf2_1kb.chr11, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 277.2311, colour = "firebrick") + geom_hline(yintercept = 163.2037, colour = "steelblue")
pany.clr.11 = ggplot() + geom_point(data = pany.sf2_1kb.chr11, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1129.9115 , colour = "firebrick") + geom_hline(yintercept = 79.82179 , colour = "steelblue")
mdme.clr.11 = ggplot() + geom_point(data = mdme.sf2_1kb.chr11, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 987.3352  , colour = "firebrick") + geom_hline(yintercept = 111.8876  , colour = "steelblue")
mbns.clr.11 = ggplot() + geom_point(data = mbns.sf2_1kb.chr11, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1111.11989, colour = "firebrick") + geom_hline(yintercept = 90.9078, colour = "steelblue")
maqu.clr.11 = ggplot() + geom_point(data = maqu.sf2_1kb.chr11, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 952.8811, colour = "firebrick") + geom_hline(yintercept = 179.91105, colour = "steelblue")
qtl.11.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr11, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.11, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.11 = qtl.11.plot / jiga.clr.11 / oinc.clr.11 / pany.clr.11 / mdme.clr.11 /mbns.clr.11 / maqu.clr.11

jiga.clr.11 = ggplot() + geom_point(data = jiga.sf2_1kb.chr11, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
oinc.clr.11 = ggplot() + geom_point(data = oinc.sf2_1kb.chr11, aes(x=(location/1000000), y=LR, colour=1/alpha))
pany.clr.11 = ggplot() + geom_point(data = pany.sf2_1kb.chr11, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mdme.clr.11 = ggplot() + geom_point(data = mdme.sf2_1kb.chr11, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mbns.clr.11 = ggplot() + geom_point(data = mbns.sf2_1kb.chr11, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
maqu.clr.11 = ggplot() + geom_point(data = maqu.sf2_1kb.chr11, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
qtl.11.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr11, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.11, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.11 = qtl.11.plot / jiga.clr.11 / oinc.clr.11 / pany.clr.11 / mdme.clr.11 /mbns.clr.11 / maqu.clr.11
```

Chromosome 13
```{r, eval=F}
qtl = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint.csv")
qtl.13 = filter(qtl, chr == 13)

jiga.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr13_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr13$Chr = 13

oinc.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr13_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr13$Chr = 13

pany.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr13_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr13$Chr = 13

mdme.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr13_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr13$Chr = 13

mbns.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr13_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr13$Chr = 13

maqu.sf2_1kb.chr13=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr13_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr13$Chr = 13


jiga.clr.13 = ggplot() + geom_point(data = jiga.sf2_1kb.chr13, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1687.852 , colour = "firebrick") + geom_hline(yintercept = 108.13838 , colour = "steelblue")
oinc.clr.13 = ggplot() + geom_point(data = oinc.sf2_1kb.chr13, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 277.2313, colour = "firebrick") + geom_hline(yintercept = 163.2037, colour = "steelblue")
pany.clr.13 = ggplot() + geom_point(data = pany.sf2_1kb.chr13, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1329.9135 , colour = "firebrick") + geom_hline(yintercept = 79.82179 , colour = "steelblue")
mdme.clr.13 = ggplot() + geom_point(data = mdme.sf2_1kb.chr13, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 987.3352  , colour = "firebrick") + geom_hline(yintercept = 131.8876  , colour = "steelblue")
mbns.clr.13 = ggplot() + geom_point(data = mbns.sf2_1kb.chr13, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1313.13989, colour = "firebrick") + geom_hline(yintercept = 90.9078, colour = "steelblue")
maqu.clr.13 = ggplot() + geom_point(data = maqu.sf2_1kb.chr13, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 952.8813, colour = "firebrick") + geom_hline(yintercept = 179.91305, colour = "steelblue")
qtl.13.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr13, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.13, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.13 = qtl.13.plot / jiga.clr.13 / oinc.clr.13 / pany.clr.13 / mdme.clr.13 /mbns.clr.13 / maqu.clr.13

jiga.clr.13 = ggplot() + geom_point(data = jiga.sf2_1kb.chr13, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
oinc.clr.13 = ggplot() + geom_point(data = oinc.sf2_1kb.chr13, aes(x=(location/1000000), y=LR, colour=1/alpha))
pany.clr.13 = ggplot() + geom_point(data = pany.sf2_1kb.chr13, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mdme.clr.13 = ggplot() + geom_point(data = mdme.sf2_1kb.chr13, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mbns.clr.13 = ggplot() + geom_point(data = mbns.sf2_1kb.chr13, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
maqu.clr.13 = ggplot() + geom_point(data = maqu.sf2_1kb.chr13, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
qtl.13.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr13, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.13, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.13 = qtl.13.plot / jiga.clr.13 / oinc.clr.13 / pany.clr.13 / mdme.clr.13 /mbns.clr.13 / maqu.clr.13
```

Chromosome 17
```{r, eval=F}
qtl = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint.csv")
qtl.17 = filter(qtl, chr == 17)

jiga.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr17_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr17$Chr = 17

oinc.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr17_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr17$Chr = 17

pany.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr17_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr17$Chr = 17

mdme.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr17_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr17$Chr = 17

mbns.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr17_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr17$Chr = 17

maqu.sf2_1kb.chr17=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr17_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr17$Chr = 17


jiga.clr.17 = ggplot() + geom_point(data = jiga.sf2_1kb.chr17, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1687.852 , colour = "firebrick") + geom_hline(yintercept = 108.17838 , colour = "steelblue")
oinc.clr.17 = ggplot() + geom_point(data = oinc.sf2_1kb.chr17, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 277.2317, colour = "firebrick") + geom_hline(yintercept = 163.2037, colour = "steelblue")
pany.clr.17 = ggplot() + geom_point(data = pany.sf2_1kb.chr17, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1729.9175 , colour = "firebrick") + geom_hline(yintercept = 79.82179 , colour = "steelblue")
mdme.clr.17 = ggplot() + geom_point(data = mdme.sf2_1kb.chr17, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 987.3352  , colour = "firebrick") + geom_hline(yintercept = 171.8876  , colour = "steelblue")
mbns.clr.17 = ggplot() + geom_point(data = mbns.sf2_1kb.chr17, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1717.17989, colour = "firebrick") + geom_hline(yintercept = 90.9078, colour = "steelblue")
maqu.clr.17 = ggplot() + geom_point(data = maqu.sf2_1kb.chr17, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 952.8817, colour = "firebrick") + geom_hline(yintercept = 179.91705, colour = "steelblue")
qtl.17.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr17, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.17, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.17 = qtl.17.plot / jiga.clr.17 / oinc.clr.17 / pany.clr.17 / mdme.clr.17 /mbns.clr.17 / maqu.clr.17

jiga.clr.17 = ggplot() + geom_point(data = jiga.sf2_1kb.chr17, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
oinc.clr.17 = ggplot() + geom_point(data = oinc.sf2_1kb.chr17, aes(x=(location/1000000), y=LR, colour=1/alpha))
pany.clr.17 = ggplot() + geom_point(data = pany.sf2_1kb.chr17, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mdme.clr.17 = ggplot() + geom_point(data = mdme.sf2_1kb.chr17, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mbns.clr.17 = ggplot() + geom_point(data = mbns.sf2_1kb.chr17, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
maqu.clr.17 = ggplot() + geom_point(data = maqu.sf2_1kb.chr17, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
qtl.17.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr17, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.17, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.17 = qtl.17.plot / jiga.clr.17 / oinc.clr.17 / pany.clr.17 / mdme.clr.17 /mbns.clr.17 / maqu.clr.17
```

Chromosome 18
```{r, eval=F}
qtl = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint.csv")
qtl.18 = filter(qtl, chr == 18)

jiga.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr18_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr18$Chr = 18

oinc.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr18_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr18$Chr = 18

pany.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr18_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr18$Chr = 18

mdme.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr18_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr18$Chr = 18

mbns.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr18_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr18$Chr = 18

maqu.sf2_1kb.chr18=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr18_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr18$Chr = 18


jiga.clr.18 = ggplot() + geom_point(data = jiga.sf2_1kb.chr18, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1687.852 , colour = "firebrick") + geom_hline(yintercept = 108.18838 , colour = "steelblue")
oinc.clr.18 = ggplot() + geom_point(data = oinc.sf2_1kb.chr18, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 277.2318, colour = "firebrick") + geom_hline(yintercept = 163.2037, colour = "steelblue")
pany.clr.18 = ggplot() + geom_point(data = pany.sf2_1kb.chr18, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1829.9185 , colour = "firebrick") + geom_hline(yintercept = 79.82179 , colour = "steelblue")
mdme.clr.18 = ggplot() + geom_point(data = mdme.sf2_1kb.chr18, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 987.3352  , colour = "firebrick") + geom_hline(yintercept = 181.8876  , colour = "steelblue")
mbns.clr.18 = ggplot() + geom_point(data = mbns.sf2_1kb.chr18, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1818.18989, colour = "firebrick") + geom_hline(yintercept = 90.9078, colour = "steelblue")
maqu.clr.18 = ggplot() + geom_point(data = maqu.sf2_1kb.chr18, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 952.8818, colour = "firebrick") + geom_hline(yintercept = 179.91805, colour = "steelblue")
qtl.18.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr18, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.18, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.18 = qtl.18.plot / jiga.clr.18 / oinc.clr.18 / pany.clr.18 / mdme.clr.18 /mbns.clr.18 / maqu.clr.18

jiga.clr.18 = ggplot() + geom_point(data = jiga.sf2_1kb.chr18, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
oinc.clr.18 = ggplot() + geom_point(data = oinc.sf2_1kb.chr18, aes(x=(location/1000000), y=LR, colour=1/alpha))
pany.clr.18 = ggplot() + geom_point(data = pany.sf2_1kb.chr18, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mdme.clr.18 = ggplot() + geom_point(data = mdme.sf2_1kb.chr18, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mbns.clr.18 = ggplot() + geom_point(data = mbns.sf2_1kb.chr18, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
maqu.clr.18 = ggplot() + geom_point(data = maqu.sf2_1kb.chr18, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
qtl.18.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr18, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.18, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.18 = qtl.18.plot / jiga.clr.18 / oinc.clr.18 / pany.clr.18 / mdme.clr.18 /mbns.clr.18 / maqu.clr.18
```

Chromosome 24
```{r, eval=F}
qtl = read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint.csv")
qtl.24 = filter(qtl, chr == 24)

jiga.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/jiga_sf2_input/jiga_chr24_sf2_output.txt",delim = "\t")
jiga.sf2_1kb.chr24$Chr = 24

oinc.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/oinc_sf2_input/oinc_chr24_sf2_output.txt",delim = "\t")
oinc.sf2_1kb.chr24$Chr = 24

pany.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/pany_sf2_input/pany_chr24_sf2_output.txt",delim = "\t")
pany.sf2_1kb.chr24$Chr = 24

mdme.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mdme_sf2_input/mdme_chr24_sf2_output.txt",delim = "\t")
mdme.sf2_1kb.chr24$Chr = 24

mbns.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/mbns_sf2_input/mbns_chr24_sf2_output.txt",delim = "\t")
mbns.sf2_1kb.chr24$Chr = 24

maqu.sf2_1kb.chr24=read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/sweepfinder2/maqu_sf2_input/maqu_chr24_sf2_output.txt",delim = "\t")
maqu.sf2_1kb.chr24$Chr = 24


jiga.clr.24 = ggplot() + geom_point(data = jiga.sf2_1kb.chr24, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 1687.852 , colour = "firebrick") + geom_hline(yintercept = 108.24838 , colour = "steelblue")
oinc.clr.24 = ggplot() + geom_point(data = oinc.sf2_1kb.chr24, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 277.2324, colour = "firebrick") + geom_hline(yintercept = 163.2037, colour = "steelblue")
pany.clr.24 = ggplot() + geom_point(data = pany.sf2_1kb.chr24, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 2429.9245 , colour = "firebrick") + geom_hline(yintercept = 79.82179 , colour = "steelblue")
mdme.clr.24 = ggplot() + geom_point(data = mdme.sf2_1kb.chr24, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 987.3352  , colour = "firebrick") + geom_hline(yintercept = 241.8876  , colour = "steelblue")
mbns.clr.24 = ggplot() + geom_point(data = mbns.sf2_1kb.chr24, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 2424.24989, colour = "firebrick") + geom_hline(yintercept = 90.9078, colour = "steelblue")
maqu.clr.24 = ggplot() + geom_point(data = maqu.sf2_1kb.chr24, aes(x=(location/1000000), y=LR)) + geom_hline(yintercept = 952.8824, colour = "firebrick") + geom_hline(yintercept = 179.92405, colour = "steelblue")
qtl.24.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr24, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.24, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.24 = qtl.24.plot / jiga.clr.24 / oinc.clr.24 / pany.clr.24 / mdme.clr.24 /mbns.clr.24 / maqu.clr.24

jiga.clr.24 = ggplot() + geom_point(data = jiga.sf2_1kb.chr24, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
oinc.clr.24 = ggplot() + geom_point(data = oinc.sf2_1kb.chr24, aes(x=(location/1000000), y=LR, colour=1/alpha))
pany.clr.24 = ggplot() + geom_point(data = pany.sf2_1kb.chr24, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mdme.clr.24 = ggplot() + geom_point(data = mdme.sf2_1kb.chr24, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
mbns.clr.24 = ggplot() + geom_point(data = mbns.sf2_1kb.chr24, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
maqu.clr.24 = ggplot() + geom_point(data = maqu.sf2_1kb.chr24, aes(x=(location/1000000), y=LR, colour=1/alpha)) 
qtl.24.plot = ggplot() + geom_point(data = maqu.sf2_1kb.chr24, aes(x=(location/1000000), y=LR), colour = "white") + 
  geom_rect(data = qtl.24, aes(xmin=start, xmax=end, ymin=100, ymax=1000, fill = Trait), alpha = 0.5)

qtl.clr.24 = qtl.24.plot / jiga.clr.24 / oinc.clr.24 / pany.clr.24 / mdme.clr.24 /mbns.clr.24 / maqu.clr.24
```


***************************************************************************************************************



# Perform genotype-environment association for all 12 populations:

1) With all individual SNPs
2) With inversions treated as single loci (re-estimate omega matrix without SNPs inside inversions)


## Create allele frequency input file for all SNPs:
1 row per SNP and 2 columns per SNP (minor allele count, major allele count)

Load MAF data for each population and estimate minor and major allele counts, and finally concatenate all counts into a single table
```{r, eval=F}
JIGA.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_JIGA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
LRSC.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_LRSC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
CTVA.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_CTVA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
OINC.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_OINC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCNC.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCNC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
PANY.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_PANY_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCCT.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCCT_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
DXMA.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_DXMA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MDME.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MDME_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
HXNS.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_HXNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MBNS.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MBNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MAQU.maf <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MAQU_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))

#Estimate minor allele count (x) and major allele count (y) for each population
JIGA.maf$x=round((JIGA.maf$knownEM)*(2*(JIGA.maf$nInd)), digits = 0)
JIGA.maf$y=(2*(JIGA.maf$nInd))-JIGA.maf$x

LRSC.maf$x=round((LRSC.maf$knownEM)*(2*(LRSC.maf$nInd)), digits = 0)
LRSC.maf$y=(2*(LRSC.maf$nInd))-LRSC.maf$x

MCNC.maf$x=round((MCNC.maf$knownEM)*(2*(MCNC.maf$nInd)), digits = 0)
MCNC.maf$y=(2*(MCNC.maf$nInd))-MCNC.maf$x

OINC.maf$x=round((OINC.maf$knownEM)*(2*(OINC.maf$nInd)), digits = 0)
OINC.maf$y=(2*(OINC.maf$nInd))-OINC.maf$x

CTVA.maf$x=round((CTVA.maf$knownEM)*(2*(CTVA.maf$nInd)), digits = 0)
CTVA.maf$y=(2*(CTVA.maf$nInd))-CTVA.maf$x

PANY.maf$x=round((PANY.maf$knownEM)*(2*(PANY.maf$nInd)), digits = 0)
PANY.maf$y=(2*(PANY.maf$nInd))-PANY.maf$x

MCCT.maf$x=round((MCCT.maf$knownEM)*(2*(MCCT.maf$nInd)), digits = 0)
MCCT.maf$y=(2*(MCCT.maf$nInd))-MCCT.maf$x

DXMA.maf$x=round((DXMA.maf$knownEM)*(2*(DXMA.maf$nInd)), digits = 0)
DXMA.maf$y=(2*(DXMA.maf$nInd))-DXMA.maf$x

MDME.maf$x=round((MDME.maf$knownEM)*(2*(MDME.maf$nInd)), digits = 0)
MDME.maf$y=(2*(MDME.maf$nInd))-MDME.maf$x

MBNS.maf$x=round((MBNS.maf$knownEM)*(2*(MBNS.maf$nInd)), digits = 0)
MBNS.maf$y=(2*(MBNS.maf$nInd))-MBNS.maf$x

HXNS.maf$x=round((HXNS.maf$knownEM)*(2*(HXNS.maf$nInd)), digits = 0)
HXNS.maf$y=(2*(HXNS.maf$nInd))-HXNS.maf$x

MAQU.maf$x=round((MAQU.maf$knownEM)*(2*(MAQU.maf$nInd)), digits = 0)
MAQU.maf$y=(2*(MAQU.maf$nInd))-MAQU.maf$x

allele_count.df = cbind(JIGA.maf[,c(1,2,8,9)],LRSC.maf[,c(8,9)],MCNC.maf[,c(8,9)],OINC.maf[,c(8,9)]
                        ,CTVA.maf[,c(8,9)],PANY.maf[,c(8,9)],MCCT.maf[,c(8,9)],DXMA.maf[,c(8,9)]
                        ,MDME.maf[,c(8,9)],MBNS.maf[,c(8,9)],HXNS.maf[,c(8,9)],MAQU.maf[,c(8,9)])

allele_count.snpids = allele_count.df[,1:2]
allele_counts = allele_count.df[,3:26]
allele_counts[is.na(allele_counts)] <- 0

write_delim(allele_counts, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_filtsnps_allelecount_df_baypass.geno", delim = " ", col_names = F)
write_delim(allele_count.snpids, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_filtsnps_allele_snpids_df_baypass.txt", delim = "\t", col_names = T)
```


## BayPass selection and association analysis with all SNPS:

Estimate omega covariance matrix using full SNP dataset and the core model:
```{bash, eval=F}
nohup /workdir/arne/programs/baypass_2.2/sources/g_baypass -gfile mme_filtsnps_allelecount_df_baypass.geno -outprefix mme_12pop_coremodel -nthreads 8 > mme_12pop_coremodel_baypass_log.file &
```

Plot covariance matrix
```{r, eval=F}
library(corrplot)
library(ape)
#source the baypass R functions (check PATH)
source("baypass_utils.R")
#upload estimate of omega
omega=as.matrix(read.table("~/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_12pop_coremodel_mat_omega.out"))
pop.names=c("JIGA","LRSC","MCNC","OINC","CTVA","PANY","MCCT","DXMA",
            "MDME","MBNS","HXNS","MAQU")

dimnames(omega)=list(pop.names,pop.names)
#Compute and visualize the correlation matrix
cor.mat=cov2cor(omega)
corrplot(cor.mat,method="color",mar=c(2,1,2,2)+0.1,
         main=expression("Correlation map based on"~hat(Omega)))
```


Run standard covariance model to test association with PC1 and PC2 (one covariate file)
```{bash, eval=F}
nohup /workdir/arne/programs/baypass_2.2/sources/g_baypass -gfile mme_filtsnps_allelecount_df_baypass.geno -efile mme_12pop_envPC1_envPC2_baypass.env -nthreads 8 -omegafile mme_12pop_coremodel_mat_omega.out -scalecov -outprefix mme_envPCAs_assoc_stdcovmod > mme_12pop_stdcovmodel_baypass_envPCA_assoc_log.file &
```

Select significant SNPs
```{R, eval=F}
sv <- read.csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
names(svs.major)[2] = "seqnames"

sv.maria <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
names(sv.maria)[1] = "seqnames"

baypass.snp.info = read.table("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_filtsnps_allele_snpids_df_baypass.txt", header = T)
baypass.snp.info$MRK = c(1:length(baypass.snp.info$position))

#PC1
baypass.pc1 = read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_betai_reg_PC1_significant.out")
baypass.envPC.res.pc1.id = left_join(baypass.pc1, baypass.snp.info, by="MRK")
baypass.envPC.res.pc1.id.chr <-  as_tibble(dplyr::filter(baypass.envPC.res.pc1.id, grepl('Mme_', chromo)))
baypass.envPC.res.pc1.id.chr <- baypass.envPC.res.pc1.id.chr[,c(9,10,10,1:8)]
names(baypass.envPC.res.pc1.id.chr) = c("seqnames","start","end","covariable","MRK","M_Pearson","SD_Pearson","BFdb","Beta_is","SD_Beta_is","eBPis")
baypass.envPC.res.pc1.id.chr.ranges =  baypass.envPC.res.pc1.id.chr %>% as_granges()
baypass.envPC.res.pc1.id.chr.regions  = mergeWindows(baypass.envPC.res.pc1.id.chr.ranges, tol=1000000)
baypass.envPC.res.pc1.id.chr.regions.df = as_tibble(baypass.envPC.res.pc1.id.chr.regions$regions)
names(baypass.envPC.res.pc1.id.chr.regions.df) = c("chromo","start","end","width","strand")

ggplot() +
  geom_point(data = baypass.envPC.res.pc1.id.chr, aes(x=(start)/1000000, y=BFdb)) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin=0, ymax=10)) +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin=-10, ymax=0), fill="orange") +
  geom_rect(data = baypass.envPC.res.pc1.id.chr.regions.df, aes(xmin=(start)/1000000, xmax=(end)/1000000, ymin=10, ymax=20)) +
  facet_wrap(~seqnames, nrow = 4)

chr1.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr01")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr4.pc1.top1 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr04", start < 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr4.pc1.top2 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr04", start > 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr6.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr06")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr7.pc1.top1 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr07", start < 5000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr7.pc1.top2 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr07", start > 5000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr8.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr08")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr11.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr11")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr12.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr12")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr13.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr13")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr14.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr14")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr16.pc1.top1 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr16", start < 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr16.pc1.top2 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr16", start > 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr17.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr17")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr18.pc1.top1 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr18", start < 5000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr18.pc1.top2 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr18", start > 5000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr19.pc1.top1 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr19", start < 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr19.pc1.top2 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr19", start > 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr20.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr20")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr21.pc1.top = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr21")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr24.pc1.top1 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr24", start < 2000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr24.pc1.top2 = baypass.envPC.res.pc1.id.chr %>% filter(seqnames == "Mme_chr24", start > 5000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]

gea.pc1.topSNPs = rbind(chr1.pc1.top,
chr4.pc1.top1,
chr4.pc1.top2,
chr6.pc1.top ,
chr7.pc1.top1,
chr7.pc1.top2,
chr8.pc1.top ,
chr11.pc1.top,
chr12.pc1.top,
chr13.pc1.top,
chr14.pc1.top,
chr16.pc1.top1,
chr16.pc1.top2,
chr17.pc1.top ,
chr18.pc1.top1,
chr18.pc1.top2,
chr19.pc1.top1,
chr19.pc1.top2,
chr20.pc1.top ,
chr21.pc1.top ,
chr24.pc1.top1,
chr24.pc1.top2)


#PC2
baypass.pc2 = read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_betai_reg_pc2_significant.out")
baypass.envPC.res.pc2.id = left_join(baypass.pc2, baypass.snp.info, by="MRK")
baypass.envPC.res.pc2.id.chr <-  as_tibble(dplyr::filter(baypass.envPC.res.pc2.id, grepl('Mme_', chromo)))
baypass.envPC.res.pc2.id.chr <- baypass.envPC.res.pc2.id.chr[,c(9,10,10,1:8)]
names(baypass.envPC.res.pc2.id.chr) = c("seqnames","start","end","covariable","MRK","M_Pearson","SD_Pearson","BFdb","Beta_is","SD_Beta_is","eBPis")
baypass.envPC.res.pc2.id.chr.ranges =  baypass.envPC.res.pc2.id.chr %>% as_granges()
baypass.envPC.res.pc2.id.chr.regions  = mergeWindows(baypass.envPC.res.pc2.id.chr.ranges, tol=1000000)
baypass.envPC.res.pc2.id.chr.regions.df = as_tibble(baypass.envPC.res.pc2.id.chr.regions$regions)
names(baypass.envPC.res.pc2.id.chr.regions.df) = c("chromo","start","end","width","strand")

ggplot() +
  geom_point(data = baypass.envPC.res.pc2.id.chr, aes(x=(start)/1000000, y=BFdb)) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin=0, ymax=10)) +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin=-10, ymax=0), fill="orange") +
  geom_rect(data = baypass.envPC.res.pc2.id.chr.regions.df, aes(xmin=(start)/1000000, xmax=(end)/1000000, ymin=10, ymax=20)) +
  facet_wrap(~seqnames, nrow = 4)

chr1.pc2.top1 = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr01", start < 5000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr1.pc2.top2 = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr01", start > 5000000, start < 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr1.pc2.top3 = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr01", start > 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr2.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr02")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr3.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr03")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr4.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr04")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr5.pc2.top1 = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr05", start < 15000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr5.pc2.top2 = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr05", start > 15000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr6.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr06")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr7.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr07")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr9.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr09")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr10.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr10")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr11.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr11")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr12.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr12")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr14.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr14")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr15.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr15")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr16.pc2.top1 = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr16", start < 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr16.pc2.top2 = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr16", start > 10000000)  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr18.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr18")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr19.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr19")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr21.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr21")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr22.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr22")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]
chr24.pc2.top = baypass.envPC.res.pc2.id.chr %>% filter(seqnames == "Mme_chr24")  %>% .[with(.,order(-BFdb)),]  %>% .[1:5,]

gea.pc2.topSNPs = rbind(chr1.pc2.top1, 
chr1.pc2.top2,
chr1.pc2.top3,
chr2.pc2.top ,
chr3.pc2.top ,
chr4.pc2.top ,
chr5.pc2.top1,
chr5.pc2.top2,
chr6.pc2.top ,
chr7.pc2.top ,
chr9.pc2.top ,
chr10.pc2.top,
chr11.pc2.top,
chr12.pc2.top,
chr14.pc2.top,
chr15.pc2.top,
chr16.pc2.top1,
chr16.pc2.top2,
chr18.pc2.top ,
chr19.pc2.top ,
chr21.pc2.top ,
chr22.pc2.top ,
chr24.pc2.top)

gea.topSNPs = rbind(gea.pc1.topSNPs, gea.pc2.topSNPs)
write_csv(gea.topSNPs, "~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/baypass_gea_pc1_pc2_topSNPs_GTseq_panel.csv")
```


Plot selection scan and GEA with inversions and QTL

```{r, eval=F}
qtl <- read_csv("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")

sv.maria <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4)]
names(sv.maria2) = c("seqnames","start","end")
sv.maria.chr.ranges =  sv.maria2 %>% as_granges()

#with SV regions highlighted
sv <- read.csv("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
svs.major2 = svs.major[,c(2,10,11)]
names(svs.major2) = c("seqnames","start","end")
svs.major.ranges =  svs.major2 %>% as_granges()

all.sv = rbind(sv.maria2, svs.major2)
all.sv.ranges =  all.sv %>% as_granges()

names(qtl)[2] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[2] = "chromo"
```

```{r, eval=F}
#source the baypass R functions (check PATH)
source("~/Downloads/software/baypass_2.31/utils/baypass_utils.R")

xtx.df = read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_pi_xtx.out")
head(xtx.df)
names(xtx.df) = c("MRK","M_P","SD_P","M_XtX","SD_XtX","XtXst","log10pval")
baypass.snp.info = read.table("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_filtsnps_allele_snpids_df_baypass.txt", header = T)
baypass.snp.info$MRK = c(1:length(baypass.snp.info$position))
xtx.snp.df = left_join(baypass.snp.info, xtx.df, by="MRK")
xtx.snp.df.chr <-  as_tibble(dplyr::filter(xtx.snp.df, grepl('Mme_', chromo)))

hist(xtx.snp.df.chr$M_XtX)
hist(xtx.snp.df.chr$XtXst)
hist(xtx.snp.df.chr$log10pval)


#Simulate POD for XtX as output
omega=as.matrix(read.table("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_12pop_coremodel_mat_omega.out"))
geno = "/media/arne/Expansion/projects/silverside_genomics/baypass/mme_filtsnps_allelecount_df_baypass.geno"
geno.df = geno2YN(geno)

pod = simulate.baypass(omega, nsnp=1000,sample.size=geno.df$NN)
write.table(pod$Y.sim, "/media/arne/Expansion/projects/silverside_genomics/baypass/mme_simpod_baypass.geno", quote = F, append = F, row.names = F)
```

Run baypass with simulated pod genotypes and real omega
```{bash, eval=F}
~/Downloads/software/baypass_2.31/sources/g_baypass -gfile /media/arne/Expansion/projects/silverside_genomics/baypass/mme_simpod_baypass.geno -omegafile ~/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_12pop_coremodel_mat_omega.out -outprefix ~/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/pod_mmecov
```

Analyse POD
```{r, eval=F}
xtx.pod.df = read.table("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/pod_mmecov_summary_pi_xtx.out", header = T)
head(xtx.pod.df)

hist(xtx.pod.df$M_XtX)
hist(xtx.pod.df$XtXst)
hist(xtx.pod.df$log10pval)

pod.thresh.Mxtx = quantile(xtx.pod.df$M_XtX, probs=0.99)
pod.thresh.XtXst = quantile(xtx.pod.df$XtXst, probs=0.99)


xtx.snp.df.chr.sig = filter(xtx.snp.df.chr, XtXst > 20.84117)
write_delim(xtx.snp.df.chr.sig, "~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_pi_xtx_sigXtXst.txt")

xtx.snp.df.chr.nonsig = filter(xtx.snp.df.chr, XtXst < 5)
write_delim(xtx.snp.df.chr.nonsig, "~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_pi_xtx_non_sigXtXst.txt")

```


```{r, eval=F}
#XtX for major QTL
XtX.chr11 = ggplot() +
  geom_point(data = filter(xtx.snp.df.chr, chromo == "Mme_chr11"), aes(x=position/1000000 , y=XtXst)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 110, ymax=115), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 105, ymax=110), fill = "steelblue3", colour="grey20", alpha=0.5) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr13 = ggplot() +
  geom_point(data = filter(xtx.snp.df.chr, chromo == "Mme_chr13"), aes(x=position/1000000 , y=XtXst)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 105, ymax=110), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr17 = ggplot() +
  geom_point(data = filter(xtx.snp.df.chr, chromo == "Mme_chr17"), aes(x=position/1000000 , y=XtXst)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 105, ymax=110), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr18 = ggplot() +
  geom_point(data = filter(xtx.snp.df.chr, chromo == "Mme_chr18"), aes(x=position/1000000 , y=XtXst)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB,  ymin = 110, ymax=115), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 105, ymax=110), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr24 = ggplot() +
  geom_point(data = filter(xtx.snp.df.chr, chromo == "Mme_chr24"), aes(x=position/1000000 , y=XtXst)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 110, ymax=115), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 105, ymax=110), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()



#XtX in windows
xtx.snp.df.chr_win <- winScan(x = xtx.snp.df.chr, 
                           groups = "chromo", 
                           position = "position", 
                           values = c("M_XtX","XtXst","log10pval"), 
                           win_size = 10000,
                           win_step = 10000,
                           funs = c("mean"))
write_csv(xtx.snp.df.chr_win, "~/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/baypass_XtX_10kbwin.csv")

xtx.snp.df.chr_win <- read_csv("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/baypass_XtX_10kbwin.csv")

XtX.data <- xtx.snp.df.chr_win %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(win_mid)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(xtx.snp.df.chr_win, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, win_mid) %>%
  mutate(BPcum=win_mid+tot)
axisdf = XtX.data %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted

ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=80), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 81, ymax=85), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -5, ymax=-0.25), fill = "gray30", colour = "gray20") +
  geom_point(data = XtX.data, aes(x=(win_mid/1000000), y= XtXst_mean), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  ylab("XtXst") +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")


#windowed XtX for major QTL
XtX.chr11 = ggplot() +
  geom_point(data = filter(XtX.data, chromo == "Mme_chr11"), aes(x=win_mid/1000000 , y=XtXst_mean)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr11"), aes(xmin=startMB, xmax=endMB, ymin = 110, ymax=115), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr11"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 105, ymax=110), fill = "steelblue3", colour="grey20", alpha=0.5) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr13 = ggplot() +
  geom_point(data = filter(XtX.data, chromo == "Mme_chr13"), aes(x=win_mid/1000000 , y=XtXst_mean)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr13"), aes(xmin=startMB, xmax=endMB, ymin = 105, ymax=110), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr13"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr17 = ggplot() +
  geom_point(data = filter(XtX.data, chromo == "Mme_chr17"), aes(x=win_mid/1000000 , y=XtXst_mean)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr17"), aes(xmin=startMB, xmax=endMB, ymin = 105, ymax=110), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr17"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr18 = ggplot() +
  geom_point(data = filter(XtX.data, chromo == "Mme_chr18"), aes(x=win_mid/1000000 , y=XtXst_mean)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr18"), aes(xmin=startMB, xmax=endMB,  ymin = 110, ymax=115), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr18"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 105, ymax=110), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()

XtX.chr24 = ggplot() +
  geom_point(data = filter(XtX.data, chromo == "Mme_chr24"), aes(x=win_mid/1000000 , y=XtXst_mean)) + 
  geom_rect(data = filter(svs.major, chromo == "Mme_chr24"), aes(xmin=startMB, xmax=endMB, ymin = 110, ymax=115), fill = "firebrick", colour="grey20") +
  geom_rect(data = filter(qtl, chromo == "Mme_chr24"), aes(xmin=start, xmax=end, ymin = -5, ymax=-0.5, fill = Trait), colour = "gray20", alpha=0.5) +
  geom_rect(data = filter(sv.maria, chromo == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 105, ymax=110), fill = "steelblue3", colour="grey20") +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  xlab("Chromosome position [Mb]") +
  ylab("XtX") +
  theme_cowplot()


```


GEA PC1 and PC2
```{r, eval=F}
baypass.snp.info = read.table("~/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_filtsnps_allele_snpids_df_baypass.txt", header = T)
baypass.snp.info$MRK = c(1:length(baypass.snp.info$position))

#PC1
baypass.pc1 = read.table("~/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_betai_reg_PC1.out", header = F)
head(baypass.pc1)
names(baypass.pc1)[2] = "MRK"
baypass.envPC.res.pc1.id = left_join(baypass.pc1, baypass.snp.info, by="MRK")
baypass.envPC.res.pc1.id.chr <-  as_tibble(dplyr::filter(baypass.envPC.res.pc1.id, grepl('Mme_', chromo)))
baypass.envPC.res.pc1.id.chr$startMb = (baypass.envPC.res.pc1.id.chr$position/1000000)
baypass.envPC.res.pc1.id.chr$endMb = (baypass.envPC.res.pc1.id.chr$position/1000000)
baypass.envPC.res.pc1.id.chr <- baypass.envPC.res.pc1.id.chr[,c(9,10,2:8)]
names(baypass.envPC.res.pc1.id.chr) = c("chromo","pos","MRK","M_Pearson","SD_Pearson","BFdb","Beta_is","SD_Beta_is","eBPis")

hist(baypass.envPC.res.pc1.id.chr$BFdb)

#envPC1.sig = filter(baypass.envPC.res.pc1.id.chr, BFdb >= 20)
#envPC1.nonsig = filter(baypass.envPC.res.pc1.id.chr, BFdb < 20)
#envPC1.nonsig_win <- winScan(x = envPC1.nonsig, 
#                           groups = "chromo", 
#                           position = "pos", 
#                           values = c("BFdb","eBPis"), 
#                           win_size = 10000,
#                           win_step = 10000,
#                           funs = c("mean"))
#envPC1.nonsig_win.df = filter(envPC1.nonsig_win, BFdb_n > 2)
#
#envPC1.df = rbind(envPC1.nonsig.subset, envPC1.sig)
#envPC1.df = arrange(envPC1.df, MRK)

envPC1.data <- baypass.envPC.res.pc1.id.chr %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(pos)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(baypass.envPC.res.pc1.id.chr, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, pos) %>%
  mutate(BPcum=pos+tot)
axisdf = envPC1.data %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
baypass.pc1.plot = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=105), fill = "steelblue") +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 105, ymax=110), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -20, ymax=-10, fill = Trait), colour = "gray20") +
  geom_point_rast(data = envPC1.data, aes(x=(pos/1000000), y= BFdb), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = 20, colour="firebrick", linetype="dashed") +
  ylab("envPC1 BF [db]") +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")

#pc2
baypass.pc2 = read.table("~/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_betai_reg_PC2.out", header = F)
head(baypass.pc2)
names(baypass.pc2)[2] = "MRK"
baypass.envPC.res.pc2.id = left_join(baypass.pc2, baypass.snp.info, by="MRK")
baypass.envPC.res.pc2.id.chr <-  as_tibble(dplyr::filter(baypass.envPC.res.pc2.id, grepl('Mme_', chromo)))
baypass.envPC.res.pc2.id.chr$startMb = (baypass.envPC.res.pc2.id.chr$position/1000000)
baypass.envPC.res.pc2.id.chr$endMb = (baypass.envPC.res.pc2.id.chr$position/1000000)
baypass.envPC.res.pc2.id.chr <- baypass.envPC.res.pc2.id.chr[,c(9,10,2:8)]
names(baypass.envPC.res.pc2.id.chr) = c("chromo","pos","MRK","M_Pearson","SD_Pearson","BFdb","Beta_is","SD_Beta_is","eBPis")

hist(baypass.envPC.res.pc2.id.chr$BFdb)

envpc2.data <- baypass.envPC.res.pc2.id.chr %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(pos)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(baypass.envPC.res.pc2.id.chr, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, pos) %>%
  mutate(BPcum=pos+tot)
axisdf = envpc2.data %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

#with SV regions + QTL highlighted
baypass.pc2.plot = ggplot() +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 0, ymax=105), fill = "steelblue") +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 105, ymax=110), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -20, ymax=-10, fill = Trait), colour = "gray20") +
  geom_point_rast(data = envpc2.data, aes(x=(pos/1000000), y= BFdb), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = 20, colour="firebrick", linetype="dashed") +
  ylab("envPC2 BF [db]") +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")


baypass.gea.plots = baypass.pc1.plot / baypass.pc2.plot
ggsave2(plot=baypass.gea.plots, path="~/Dropbox/Cornell_Postdoc/manuscripts/mme_qtl_popgen_ms/figures/raw/6. GEA/", filename = "baypass_gea_plot_patch.pdf", width = 1000, height = 300, units = "mm")
```


```{r, eval=F}
#chr11
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr11"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.0, ymax=0.025), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(baypass.envPC.res.pc1.id.chr, Chr == "Mme_chr11"), aes(x=(start/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  ylab("BF [db] envPC1") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr11

#chr13
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr13"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("") +
  xlab("position [Mb]") +
  theme_cowplot()

#chr17
ggplot() +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr17"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr17


#chr18
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr18"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.0, ymax=0.025), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr18"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr18


#chr24
ggplot() +
  geom_rect(data = filter(sv.maria, chr == "Mme_chr24"), aes(xmin=inv_start, xmax=inv_end, ymin = 0.0, ymax=0.025), fill = "steelblue", alpha = 0.2) +
  geom_smooth(data = filter(tajD.jiga, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#4F000B") +
  geom_smooth(data = filter(tajD.oinc, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#EE7600") +
  geom_smooth(data = filter(tajD.pany, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#FFC100") +
  geom_smooth(data = filter(tajD.mdme, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#0F85A0") +
  geom_smooth(data = filter(tajD.mbns, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#05668D") +
  geom_smooth(data = filter(tajD.maqu, Chr == "Mme_chr24"), aes(x=(WinCenter/1000000), y=(tP/nSites)), method = "loess", se = F, span=0.1, colour = "#3D5A80") +
  ylab("") +
  xlab("position [Mb]") +
  theme_cowplot()
Pi_smoothed_5pops_chr24
```





Overlap of significant GEA SNPs with SVs and QTL:
```{R, eval=F}
sv <- read.csv("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major" | inv.type == "minor")
svs.major = svs.major[,c(2,10,11)]
names(svs.major) = c("seqnames","start","end")
svs.major.ranges =  svs.major %>% as_granges()

sv.maria <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria2 <- sv.maria[,c(1,3,4)]
names(sv.maria2) = c("seqnames","start","end")
sv.maria.chr.ranges =  sv.maria2 %>% as_granges()

allSVs = rbind(svs.major, sv.maria2)
allSVs.ranges =  allSVs %>% as_granges()

qtl <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
qtl <- qtl[,c(2,3,4,1,c(5:11))]
names(qtl) = c("seqnames","start","end","trait","mid_pos","LOD","model","pve","cM","cM_CI","chromo")
qtl.ranges =  qtl %>% as_granges()

baypass.snp.info = read.table("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_filtsnps_allele_snpids_df_baypass.txt", header = T)
baypass.snp.info$MRK = c(1:length(baypass.snp.info$position))

#PC1
baypass.pc1 = read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_betai_reg_PC1_significant.out")
baypass.envPC.res.pc1.id = left_join(baypass.pc1, baypass.snp.info, by="MRK")
baypass.envPC.res.pc1.id.chr <-  as_tibble(dplyr::filter(baypass.envPC.res.pc1.id, grepl('Mme_', chromo)))
baypass.envPC.res.pc1.id.chr$startMb = (baypass.envPC.res.pc1.id.chr$position/1000000)
baypass.envPC.res.pc1.id.chr$endMb = (baypass.envPC.res.pc1.id.chr$position/1000000)
baypass.envPC.res.pc1.id.chr <- baypass.envPC.res.pc1.id.chr[,c(9,11,12,1:8)]
names(baypass.envPC.res.pc1.id.chr) = c("seqnames","start","end","covariable","MRK","M_Pearson","SD_Pearson","BFdb","Beta_is","SD_Beta_is","eBPis")
baypass.envPC.res.pc1.id.chr.ranges =  baypass.envPC.res.pc1.id.chr %>% as_granges()
baypass.envPC.res.pc1.id.chr.regions  = mergeWindows(baypass.envPC.res.pc1.id.chr.ranges, tol=0.01)
baypass.envPC.res.pc1.id.chr.regions.df = as_tibble(baypass.envPC.res.pc1.id.chr.regions$regions)
names(baypass.envPC.res.pc1.id.chr.regions.df) = c("chromo","start","end","width","strand")
median(baypass.envPC.res.pc1.id.chr.regions.df$width)
hist(baypass.envPC.res.pc1.id.chr.regions.df$width)

#PC2
baypass.pc2 = read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_betai_reg_pc2_significant.out")
baypass.envPC.res.pc2.id = left_join(baypass.pc2, baypass.snp.info, by="MRK")
baypass.envPC.res.pc2.id.chr <-  as_tibble(dplyr::filter(baypass.envPC.res.pc2.id, grepl('Mme_', chromo)))
baypass.envPC.res.pc2.id.chr$startMb = (baypass.envPC.res.pc2.id.chr$position/1000000)
baypass.envPC.res.pc2.id.chr$endMb = (baypass.envPC.res.pc2.id.chr$position/1000000)
baypass.envPC.res.pc2.id.chr <- baypass.envPC.res.pc2.id.chr[,c(9,11,12,1:8)]
names(baypass.envPC.res.pc2.id.chr) = c("seqnames","start","end","covariable","MRK","M_Pearson","SD_Pearson","BFdb","Beta_is","SD_Beta_is","eBPis")
baypass.envPC.res.pc2.id.chr.ranges =  baypass.envPC.res.pc2.id.chr %>% as_granges()
baypass.envPC.res.pc2.id.chr.regions  = mergeWindows(baypass.envPC.res.pc2.id.chr.ranges, tol=0.01)
baypass.envPC.res.pc2.id.chr.regions.df = as_tibble(baypass.envPC.res.pc2.id.chr.regions$regions)
names(baypass.envPC.res.pc2.id.chr.regions.df) = c("chromo","start","end","width","strand")
mean(baypass.envPC.res.pc2.id.chr.regions.df$width)
hist(baypass.envPC.res.pc2.id.chr.regions.df$width)


wilcox.test(baypass.envPC.res.pc1.id.chr.regions.df$width, baypass.envPC.res.pc2.id.chr.regions.df$width)

#Overlap INV
SV.overlap = plyranges::find_overlaps(sv.maria.chr.ranges, svs.major.ranges)
SV.overlap.df = as_tibble(SV.overlap)
SV.overlap2 = plyranges::find_overlaps(svs.major.ranges, sv.maria.chr.ranges)
SV.overlap2.df = as_tibble(SV.overlap2)

#Overlap QTL vs INV
SV.qtl.overlap = plyranges::find_overlaps(sv.maria.chr.ranges, qtl.ranges)
SV.qtl.overlap.df = as_tibble(SV.qtl.overlap)
qtl.SV.overlap = plyranges::find_overlaps(qtl.ranges, sv.maria.chr.ranges)
qtl.SV.overlap.df = as_tibble(qtl.SV.overlap)

SVanna.qtl.overlap = plyranges::find_overlaps(svs.major.ranges, qtl.ranges)
SVanna.qtl.overlap.df = as_tibble(SVanna.qtl.overlap)
qtl.SVanna.overlap = plyranges::find_overlaps(qtl.ranges, svs.major.ranges)
qtl.SVanna.overlap.df = as_tibble(qtl.SV.overlap)


#Overlap GEA vs SVmaria
envPC1.SV.overlap = plyranges::find_overlaps(sv.maria.chr.ranges, baypass.envPC.res.pc1.id.chr.ranges)
envPC1.SV.overlap.df = as_tibble(envPC1.SV.overlap)
unique(envPC1.SV.overlap.df$seqnames)
envPC1reg.SV.overlap = plyranges::find_overlaps(sv.maria.chr.ranges, baypass.envPC.res.pc1.id.chr.regions$regions)
envPC1reg.SV.overlap.df = as_tibble(envPC1reg.SV.overlap)
length(unique(envPC1reg.SV.overlap.df$seqnames))

envPC2.SV.overlap = plyranges::find_overlaps(sv.maria.chr.ranges, baypass.envPC.res.pc2.id.chr.ranges)
envPC2.SV.overlap.df = as_tibble(envPC2.SV.overlap)
unique(envPC2.SV.overlap.df$seqnames)
envPC2reg.SV.overlap = plyranges::find_overlaps(sv.maria.chr.ranges, baypass.envPC.res.pc2.id.chr.regions$regions)
envPC2reg.SV.overlap.df = as_tibble(envPC2reg.SV.overlap)
length(unique(envPC2reg.SV.overlap.df$seqnames))

#Overlap GEA vs SVanna
envPC1.SVanna.overlap = plyranges::find_overlaps(svs.major.ranges, baypass.envPC.res.pc1.id.chr.ranges)
envPC1.SVanna.overlap.df = as_tibble(envPC1.SVanna.overlap)
unique(envPC1.SVanna.overlap.df$seqnames)
envPC1reg.SVanna.overlap = plyranges::find_overlaps(baypass.envPC.res.pc1.id.chr.regions$regions, svs.major.ranges)
envPC1reg.SVanna.overlap.df = as_tibble(envPC1reg.SVanna.overlap)
length(unique(envPC1reg.SVanna.overlap.df$seqnames))

envPC2.SVanna.overlap = plyranges::find_overlaps(svs.major.ranges, baypass.envPC.res.pc2.id.chr.ranges)
envPC2.SVanna.overlap.df = as_tibble(envPC2.SVanna.overlap)
unique(envPC2.SVanna.overlap.df$seqnames)
envPC2reg.SVanna.overlap = plyranges::find_overlaps(svs.major.ranges, baypass.envPC.res.pc2.id.chr.regions$regions)
envPC2reg.SVanna.overlap.df = as_tibble(envPC2reg.SVanna.overlap)
length(unique(envPC2reg.SVanna.overlap.df$seqnames))

#Overlap GEA vs allSVs
envPC1reg.allSVs.overlap = plyranges::find_overlaps(baypass.envPC.res.pc1.id.chr.regions$regions, allSVs.ranges)
envPC1reg.allSVs.overlap.df = as_tibble(envPC1reg.allSVs.overlap)
length(unique(envPC1reg.allSVs.overlap.df$seqnames))

envPC2reg.allSVs.overlap = plyranges::find_overlaps(baypass.envPC.res.pc2.id.chr.regions$regions, allSVs.ranges)
envPC2reg.allSVs.overlap.df = as_tibble(envPC2reg.allSVs.overlap)
length(unique(envPC2reg.allSVs.overlap.df$seqnames))


#Overlap GEA-PC1 vs QTL
envPC1.QTL.overlap = plyranges::find_overlaps(qtl.ranges, baypass.envPC.res.pc1.id.chr.ranges)
envPC1.QTL.overlap.df = as_tibble(envPC1.QTL.overlap)
unique(envPC1.QTL.overlap.df$seqnames)
envPC1reg.QTL.overlap = plyranges::find_overlaps(baypass.envPC.res.pc1.id.chr.regions$regions, qtl.ranges)
envPC1reg.QTL.overlap.df = as_tibble(envPC1reg.QTL.overlap)
length(unique(envPC1reg.QTL.overlap.df$mid_pos))

#Overlap GEA-PC2 vs QTL
envPC2.QTL.overlap = plyranges::find_overlaps(qtl.ranges, baypass.envPC.res.pc2.id.chr.ranges)
envPC2.QTL.overlap.df = as_tibble(envPC2.QTL.overlap)
unique(envPC2.QTL.overlap.df$seqnames)
envPC2reg.QTL.overlap = plyranges::find_overlaps(baypass.envPC.res.pc2.id.chr.regions$regions, qtl.ranges)
envPC2reg.QTL.overlap.df = as_tibble(envPC2reg.QTL.overlap)
length(unique(envPC2reg.QTL.overlap.df$seqnames))

envPC1_vs_PC2.reg.QTL.overlap.df = inner_join(envPC1reg.QTL.overlap.df, envPC2reg.QTL.overlap.df, by=c("seqnames","start","end"))
envPC1_vs_PC2.reg.QTL.overlap.df2 = unique(inner_join(envPC1reg.QTL.overlap.df[,c(6:10)], envPC2reg.QTL.overlap.df[,c(6:10)], by=c("trait","mid_pos","LOD","model","pve")))
```



Plot GEA results over Fst plot highlighting GEA-associated genomic regions, SVs and QTL:

GEA-envPC1
```{r, eval=F}
#Load genome-scan results:
jiga_pany_fst_scan <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/summary_statistics/pairwise_fst/JIGA_PANY_fst_filtsnps.10000kb_win_fst.txt", delim = "\t", col_names = F, skip = 1)

names(jiga_pany_fst_scan)[1] <- "region"
names(jiga_pany_fst_scan)[2] <- "chromo"
names(jiga_pany_fst_scan)[3] <- "midwin"
names(jiga_pany_fst_scan)[4] <- "nsites"
names(jiga_pany_fst_scan)[5] <- "fst"

jiga_pany_fst_scan <- dplyr::filter(jiga_pany_fst_scan, grepl('Mme_', chromo)) 
hist(jiga_pany_fst_scan$nsites, breaks = 100)
jiga_pany_fst_scan <- filter(jiga_pany_fst_scan, nsites >= 30)


#plot with cmplot
jiga_pany_fst_scan <- unite(jiga_pany_fst_scan, "SNP", c("chromo","midwin"),sep = ":", remove = F)
jiga_pany_fst_scan.plot <- as.data.frame(jiga_pany_fst_scan[,c(2,3,4,6)])
names(jiga_pany_fst_scan.plot) <- c("SNP","chr","pos","Fst")
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr01"] <- 1
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr06"] <- 6
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr02"] <- 2
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr03"] <- 3
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr04"] <- 4
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr05"] <- 5
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr07"] <- 7
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr08"] <- 8
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr09"] <- 9
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr10"] <- 10
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr11"] <- 11
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr12"] <- 12
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr13"] <- 13
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr14"] <- 14
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr15"] <- 15
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr16"] <- 16
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr17"] <- 17
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr18"] <- 18
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr19"] <- 19
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr20"] <- 20
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr21"] <- 21
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr22"] <- 22
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr23"] <- 23
jiga_pany_fst_scan.plot[jiga_pany_fst_scan.plot == "Mme_chr24"] <- 24

#Plot results with highlighted outliers and relativ positions using ggplot:
Fst.data <- jiga_pany_fst_scan %>% 
  group_by(chromo) %>% 
  summarise(chr_len=max(midwin)) %>% 
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%
  left_join(jiga_pany_fst_scan, ., by=c("chromo"="chromo")) %>%
  arrange(chromo, midwin) %>%
  mutate(BPcum=midwin+tot)
axisdf = Fst.data %>% group_by(chromo) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )


#with SV regions + QTL + GEA highlighted 
names(qtl)[1] = "chromo"
names(sv.maria)[1] = "chromo"
names(svs.major)[1] = "chromo"
names(svs.major)[1] = "chromo"
svs.major = as_tibble(svs.major)
names(sv.maria)[3] = "start"
names(sv.maria)[4] = "end"


ggplot() +
  geom_rect(data = baypass.envPC.res.pc1.id.chr.regions.df, aes(xmin=start, xmax=end, ymin = 1, ymax=1.1), fill = "royalblue", colour = "black", alpha=0.75) +
  geom_rect(data = baypass.envPC.res.pc2.id.chr.regions.df, aes(xmin=start, xmax=end, ymin = 1.1, ymax=1.2), fill = "gray", colour = "black", alpha=0.75) +
  geom_rect(data = sv.maria, aes(xmin=start, xmax=end, ymin = 0, ymax=1), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=start, xmax=end, ymin = -0.1, ymax=0), fill = "firebrick") +
  geom_point(data = Fst.data, aes(x=(midwin/1000000), y= fst), alpha=1, size=1, shape = 20) +
  ylab("Fst GA-NY") +
  ylim(-0.1, 1.5) +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```






***************************************************************************************************************

# Overlap candidate regions (SNPs under selection, inversions, GEA snps, GLM snps) with QTL regions

Randomly inversions regions across genome, count overlap with QTL regions, and repeat 10,000 times for null distribution of overlap. Then compare observed to expected overlap. This can also be done to compare other sets

QTL vs SV
```{r, eval=F}
library(regioneR)
library(plyranges)
library(bedtoolsr)

qtl = read.csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/mme_qtl_confint_revised.csv")
qtl.df=qtl[,c(2,3:7,1)]
names(qtl.df) <- c("seqnames", "start","end","lod","model","pve","trait")
qtl.ranges =  toGRanges(qtl.df)

sv <- read_csv("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/anna_structural_variant_positions_CTmatepairs/anna_structural_variant_positions_CTmatepairs_anchoredassembly_coord_chr.csv")
sv$startMB = sv$start/1000000
sv$endMB = sv$end/1000000
svs.major = filter(sv, inv.type == "major")
svs.major.df=svs.major[,c(1,10:11,6:9)]
names(svs.major.df) <- c("seqnames", "start","end","startBP","endBP","size","type")
svs.major.ranges =  toGRanges(svs.major.df)
sv.maria <- read_delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/mainv.txt")
sv.maria$start = sv.maria$inv_start * 1000000
sv.maria$end = sv.maria$inv_end * 1000000

sv.df=sv[,c(1,10:11,6:9)]
names(sv.df) <- c("seqnames","start","end","startBP","endBP","size","type")
sv.ranges =  toGRanges(sv.df)

numOverlaps(qtl.ranges, svs.major.ranges)
numOverlaps(qtl.ranges, sv.ranges)

pt <- permTest(A=qtl.ranges, ntimes=1000, randomize.function=resampleRegions, universe=sv.ranges,
  evaluate.function=numOverlaps, B=svs.major.ranges, verbose=FALSE)

summary(pt)
plot(pt)

#Custom script with bedtools. Shuffle QTL in genome and then intersect with SVs. Repeat 1000x and compare to real distribution.
mme_genome = read.delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/results/anchoring_assembly/Reference_assembly_20190129/mme_refgenome_renamedchr_genomefile.bed", header = FALSE)
names(mme_genome) <- NULL 
qtl.bed = read.delim("~/Library/Cloudstorage/Dropbox/Cornell_Postdoc/analysis/qtl_intervals.bed", header = F)
names(qtl.bed) <- NULL 
qtl.bed = as.integer(qtl.bed)
svs.major.bed = sv[,c(2,6,7)]
sv.maria.bed = as.data.frame(sv.maria[,c(1,5,6)])
names(sv.maria.bed)[1] = "mme_chr"
sv.all = rbind(svs.major.bed, sv.maria.bed)
names(sv.all) <- NULL 

x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=qtl.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, sv.all)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_qtl_allSVs_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}

int.real.qtl.allSVs = bedtoolsr::bt.intersect(qtl.bed, sv.all)
count.real.qtl.allSVs = length(int.real.qtl.allSVs$V1)

perm.allSVs.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_qtl_allSVs_intersect_out.txt") 
hist(perm.allSVs.count$V1)
abline(v=count.real.qtl.allSVs, col="firebrick")
wilcox.test(count.real.qtl.allSVs, perm.allSVs.count$V1, alternative = "greater")


sv.maria.bed2 = sv.maria.bed
sv.maria.bed2[,c(2,3)] = round(sv.maria.bed2[,c(2,3)], digits = 0)
names(sv.maria.bed2) <- NULL 
x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=qtl.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, sv.maria.bed2)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_qtl_linkagemapSVs_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}

int.real.qtl.lmSVs = bedtoolsr::bt.intersect(qtl.bed, sv.maria.bed2)
count.real.qtl.lmSVs = length(int.real.qtl.lmSVs$V1)
perm.linkmapSVs.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_qtl_linkagemapSVs_intersect_out.txt") 
hist(x = perm.linkmapSVs.count$V1)
m, xlim = 100, ylim = 200)
abline(v=count.real.qtl.lmSVs, col="firebrick")
wilcox.test(count.real.qtl.lmSVs, perm.linkmapSVs.count$V1, alternative = "greater")
```

GEA regions vs SVs
```{r, eval=F}
baypass.envPC.res.pc1.id.chr.regions.df$start2 = baypass.envPC.res.pc1.id.chr.regions.df$start * 1000000
baypass.envPC.res.pc1.id.chr.regions.df$end2 = baypass.envPC.res.pc1.id.chr.regions.df$end * 1000000
baypass.envPC.res.pc1.id.chr.regions.df.bed = as.data.frame(baypass.envPC.res.pc1.id.chr.regions.df[,c(1,6,7)])
names(baypass.envPC.res.pc1.id.chr.regions.df.bed) <- NULL 

baypass.envPC.res.pc2.id.chr.regions.df$start2 = baypass.envPC.res.pc2.id.chr.regions.df$start * 1000000
baypass.envPC.res.pc2.id.chr.regions.df$end2 = baypass.envPC.res.pc2.id.chr.regions.df$end * 1000000
baypass.envPC.res.pc2.id.chr.regions.df.bed = as.data.frame(baypass.envPC.res.pc2.id.chr.regions.df[,c(1,6,7)])
names(baypass.envPC.res.pc2.id.chr.regions.df.bed) <- NULL 

#GEA-PC1 vs allSVs/mariaSV overlap
x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=baypass.envPC.res.pc1.id.chr.regions.df.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, sv.all)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_geaPC1_allSVs_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}
##Actual overlap
int.real.GEApc1.allSVs = bedtoolsr::bt.intersect(baypass.envPC.res.pc1.id.chr.regions.df.bed, sv.all)
count.real.GEApc1.allSVs = length(int.real.GEApc1.allSVs$V1)
##Compare actual and permuted overlap
perm.geaPC1.allSVs.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_geaPC1_allSVs_intersect_out.txt") 
hist(x = perm.geaPC1.allSVs.count$V1)
abline(v=count.real.GEApc1.allSVs, col="firebrick")
wilcox.test(count.real.GEApc1.allSVs, perm.geaPC1.allSVs.count$V1, alternative = "greater")

x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=baypass.envPC.res.pc1.id.chr.regions.df.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, sv.maria.bed2)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_geaPC1_linkagemapSVs_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}
##Actual overlap
int.real.GEApc1.lmSVs = bedtoolsr::bt.intersect(baypass.envPC.res.pc1.id.chr.regions.df.bed, sv.maria.bed2)
count.real.GEApc1.lmSVs = length(int.real.GEApc1.lmSVs$V1)
##Compare actual and permuted overlap
perm.geaPC1.lmSVs.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_geaPC1_linkagemapSVs_intersect_out.txt") 
hist(x = perm.geaPC1.lmSVs.count$V1)
abline(v=count.real.GEApc1.lmSVs, col="firebrick")
wilcox.test(count.real.GEApc1.lmSVs, perm.geaPC1.lmSVs.count$V1, alternative = "greater")

#GEA-pc2 vs allSVs/mariaSV overlap
x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=baypass.envPC.res.pc2.id.chr.regions.df.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, sv.all)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_geapc2_allSVs_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}
##Actual overlap
int.real.GEApc2.allSVs = bedtoolsr::bt.intersect(baypass.envPC.res.pc2.id.chr.regions.df.bed, sv.all)
count.real.GEApc2.allSVs = length(int.real.GEApc2.allSVs$V1)
##Compare actual and permuted overlap
perm.geapc2.allSVs.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_geapc2_allSVs_intersect_out.txt") 
hist(x = perm.geapc2.allSVs.count$V1)
abline(v=count.real.GEApc2.allSVs, col="firebrick")
wilcox.test(count.real.GEApc2.allSVs, perm.geapc2.allSVs.count$V1, alternative = "greater")

x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=baypass.envPC.res.pc2.id.chr.regions.df.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, sv.maria.bed2)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_geapc2_linkagemapSVs_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}
##Actual overlap
int.real.GEApc2.lmSVs = bedtoolsr::bt.intersect(baypass.envPC.res.pc2.id.chr.regions.df.bed, sv.maria.bed2)
count.real.GEApc2.lmSVs = length(int.real.GEApc2.lmSVs$V1)
##Compare actual and permuted overlap
perm.geapc2.lmSVs.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_geapc2_linkagemapSVs_intersect_out.txt") 
hist(x = perm.geapc2.lmSVs.count$V1)
abline(v=count.real.GEApc2.lmSVs, col="firebrick")
wilcox.test(count.real.GEApc2.lmSVs, perm.geapc2.lmSVs.count$V1, alternative = "greater")

#NOT MORE OVERLAP THAN EXPECTED BETWEEN GEA-PC1/PC2 AND SVS (ALL OR LM ONLY)
```


GEA regions vs QTL
```{r, eval=F}
baypass.envPC.res.pc1.id.chr.regions.df$start2 = baypass.envPC.res.pc1.id.chr.regions.df$start * 1000000
baypass.envPC.res.pc1.id.chr.regions.df$end2 = baypass.envPC.res.pc1.id.chr.regions.df$end * 1000000
baypass.envPC.res.pc1.id.chr.regions.df.bed = as.data.frame(baypass.envPC.res.pc1.id.chr.regions.df[,c(1,6,7)])
names(baypass.envPC.res.pc1.id.chr.regions.df.bed) <- NULL 

baypass.envPC.res.pc2.id.chr.regions.df$start2 = baypass.envPC.res.pc2.id.chr.regions.df$start * 1000000
baypass.envPC.res.pc2.id.chr.regions.df$end2 = baypass.envPC.res.pc2.id.chr.regions.df$end * 1000000
baypass.envPC.res.pc2.id.chr.regions.df.bed = as.data.frame(baypass.envPC.res.pc2.id.chr.regions.df[,c(1,6,7)])
names(baypass.envPC.res.pc2.id.chr.regions.df.bed) <- NULL 

#GEA-PC1 vs QTL
x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=baypass.envPC.res.pc1.id.chr.regions.df.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, qtl.bed)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_geaPC1_qtl_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}
##Actual overlap
int.real.GEApc1.qtl = bedtoolsr::bt.intersect(baypass.envPC.res.pc1.id.chr.regions.df.bed, qtl.bed)
count.real.GEApc1.qtl = length(int.real.GEApc1.qtl$V1)
##Compare actual and permuted overlap
perm.geaPC1.qtl.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_geaPC1_qtl_intersect_out.txt") 
hist(x = perm.geaPC1.qtl.count$V1)
abline(v=count.real.GEApc1.qtl, col="firebrick")
wilcox.test(count.real.GEApc1.qtl, perm.geaPC1.qtl.count$V1, alternative = "greater")


#GEA-pc2 vs QTL
x <- 1 #start permutation
z <- 1000 #number of permutations
repeat {
  out = bedtoolsr::bt.shuffle(i=baypass.envPC.res.pc2.id.chr.regions.df.bed, g=mme_genome)
  int = bedtoolsr::bt.intersect(out, qtl.bed)
  count = length(int$V1)
  write(paste(count), "~/Dropbox/Cornell_Postdoc/analysis/results/perm_geapc2_qtl_intersect_out.txt", append=T, )
  x = x+1 #add next round
  if (x == z){ #stop if reaches number of permutations (z)
    break
  }
}
##Actual overlap
int.real.GEApc2.qtl = bedtoolsr::bt.intersect(baypass.envPC.res.pc2.id.chr.regions.df.bed, qtl.bed)
count.real.GEApc2.qtl = length(int.real.GEApc2.qtl$V1)
##Compare actual and permuted overlap
perm.geapc2.qtl.count <- read.table("~/Dropbox/Cornell_Postdoc/analysis/results/perm_geapc2_qtl_intersect_out.txt") 
hist(x = perm.geapc2.qtl.count$V1)
abline(v=count.real.GEApc2.qtl, col="firebrick")
wilcox.test(count.real.GEApc2.qtl, perm.geapc2.qtl.count$V1, alternative = "greater")

#NOT MORE OVERLAP THAN EXPECTED BETWEEN GEA-PC1/PC2 AND QTL
```





### Latitudinal clines of SNPs under selection (XtX) compared to neutral SNPs


MAF table
```{r, eval=F}
JIGA.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_JIGA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
LRSC.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_LRSC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
CTVA.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_CTVA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
OINC.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_OINC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCNC.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCNC_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
PANY.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_PANY_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MCCT.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MCCT_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
DXMA.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_DXMA_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MDME.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MDME_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
HXNS.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_HXNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MBNS.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MBNS_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))
MAQU.maf <- read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/allele_frequency_files/mme_MAQU_filtsnps_maf.mafs.gz",delim = "\t", col_names = T, col_types = list(chromo = col_character() ))

maf.df = cbind(JIGA.maf[,c(1,2,6)],LRSC.maf[,c(6)],MCNC.maf[,c(6)],OINC.maf[,c(6)]
                        ,CTVA.maf[,c(6)],PANY.maf[,c(6)],MCCT.maf[,c(6)],DXMA.maf[,c(6)]
                        ,MDME.maf[,c(6)],MBNS.maf[,c(6)])

names(maf.df) = c("chromo","position","JIGA","LRSC","MCNC","OINC","CTVA","PANY","MCCT","DXMA","MDME","MBNS")
```

Subset of selected and non-selected SNPs
```{r, eval=F}
dist <- read_csv("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/geographic_environmental_data/leastcost_geographic_geodist_fromJIGA.csv")
dist = filter(dist, Pop != "HXNS", Pop != "MAQU")

xtx.snp.df.chr.sig = read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_pi_xtx_sigXtXst.txt")
xtx.snp.df.chr.sig.maf = inner_join(xtx.snp.df.chr.sig, maf.df, by=c("chromo","position"))
xtx.snp.df.chr.sig.maf.dt <- as.data.table(xtx.snp.df.chr.sig.maf)
xtx.snp.df.chr.sig.maf.dt.max = xtx.snp.df.chr.sig.maf.dt[xtx.snp.df.chr.sig.maf.dt[, .I[XtXst == max(XtXst)], by=chromo]$V1]
xtx.snp.df.chr.sig.maf.dt.max.df = as_tibble(xtx.snp.df.chr.sig.maf.dt.max)

xtx.snp.df.chr.sig.maf.dt.max.df <- xtx.snp.df.chr.sig.maf %>% group_by(chromo) %>% slice_sample(n=5)
xtx.snp.df.chr.sig.maf.t = as.matrix(t(xtx.snp.df.chr.sig.maf.dt.max.df[,c(10:19)]))
str(xtx.snp.df.chr.sig.maf.t)
xtx.snp.df.chr.sig.maf.t.dist = as_tibble(cbind(dist$Dist, xtx.snp.df.chr.sig.maf.t))
xtx.snp.df.chr.sig.maf.t.dist.tall <- melt(xtx.snp.df.chr.sig.maf.t.dist, id.vars="V1")

ggplot(data = xtx.snp.df.chr.sig.maf.t.dist.tall, aes(x=V1, y=value, colour = variable)) +
  geom_smooth(method = "loess", se=F) +
  geom_point() +
  facet_wrap(~variable) +
  theme(legend.position = "none") +
  xlab("Distance from JIGA") +
  ylab("MAF")




xtx.snp.df.chr.nonsig = read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/mme_envPCAs_assoc_stdcovmod_summary_pi_xtx_non_sigXtXst.txt")
xtx.snp.df.chr.nonsig.maf = inner_join(xtx.snp.df.chr.nonsig, maf.df, by=c("chromo","position"))
xtx.snp.df.chr.nonsig.maf.dt <- as.data.table(xtx.snp.df.chr.nonsig.maf)
xtx.snp.df.chr.nonsig.maf.dt.max = xtx.snp.df.chr.nonsig.maf.dt[xtx.snp.df.chr.nonsig.maf.dt[, .I[XtXst == min(XtXst)], by=chromo]$V1]
xtx.snp.df.chr.nonsig.maf.dt.max.df = as_tibble(xtx.snp.df.chr.nonsig.maf.dt.max)
#xtx.snp.df.chr.sig.maf.dt.max.df <- xtx.snp.df.chr.nonsig.maf %>% group_by(chromo) %>% slice_sample(n=5)
xtx.snp.df.chr.nonsig.maf.dt.max.df = as_tibble(xtx.snp.df.chr.nonsig.maf.dt.max)
xtx.snp.df.chr.nonsig.maf.t = as.matrix(t(xtx.snp.df.chr.nonsig.maf.dt.max.df[,c(10:19)]))
str(xtx.snp.df.chr.nonsig.maf.t)
xtx.snp.df.chr.nonsig.maf.t.dist = as_tibble(cbind(dist$Dist, xtx.snp.df.chr.nonsig.maf.t))
xtx.snp.df.chr.nonsig.maf.t.dist.tall <- melt(xtx.snp.df.chr.nonsig.maf.t.dist, id.vars="V1")

ggplot(data = xtx.snp.df.chr.nonsig.maf.t.dist.tall, aes(x=V1, y=value, colour = variable)) +
  geom_smooth(method = "loess", se=F) +
  geom_point() +
  facet_wrap(~variable) +
  theme(legend.position = "none") +
  xlab("Distance from JIGA") +
  ylab("MAF")


```








### Overlap of XtX peaks with centromeres:


```{r, eval=F}
xtx.snp.df.chr_win <- read_csv("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/results/selection_analyses/baypass/baypass_XtX_10kbwin.csv")

centro.pos = read_delim("~/Library/CloudStorage/Dropbox/Cornell_Postdoc/analysis/centro.bed", col_names = F)
names(centro.pos) = c("chromo","win_start","win_end")

ggplot() +
  geom_rect(data = centro.pos, aes(xmin=(win_start/1000000), xmax=(win_end/1000000), ymin = 0, ymax=80), fill = "grey", alpha=0.75) +
  geom_rect(data = sv.maria, aes(xmin=inv_start, xmax=inv_end, ymin = 40, ymax=60), fill = "steelblue", alpha=0.75) +
  geom_rect(data = svs.major, aes(xmin=startMB, xmax=endMB, ymin = 60, ymax=80), fill = "firebrick") +
  geom_rect(data = qtl, aes(xmin=start, xmax=end, ymin = -5, ymax=-0.25, fill = Trait), colour = "gray20") +
  geom_point(data = xtx.snp.df.chr_win, aes(x=(win_mid/1000000), y= XtXst_mean), alpha=1, size=1, shape = 20) +
  geom_point(data = xtx.snp.df.chr_win, aes(x=(win_mid), y= XtXst_mean), alpha=1, size=1, shape = 20) +
  geom_hline(yintercept = pod.thresh.XtXst, colour="firebrick", linetype="dashed") +
  ylab("XtXst") +
  scale_fill_viridis_d() +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.line.y = element_line(colour = "black", size = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_blank(),
    legend.position="none",
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_blank()) +
  facet_grid(~chromo, scales = "free_x")
```






